diff --git a/common/src/util/min-heap.ts b/common/src/util/min-heap.ts
index fbf88117f1..240e2e2542 100644
--- a/common/src/util/min-heap.ts
+++ b/common/src/util/min-heap.ts
@@ -5,14 +5,17 @@ export class MinHeap<T> {
   private heap: { item: T; score: number }[] = []
 
   private getParentIndex(index: number): number {
+    if (!Number.isFinite(index)) return 0
     return Math.floor((index - 1) / 2)
   }
 
   private getLeftChildIndex(index: number): number {
+    if (!Number.isFinite(index)) return 0
     return 2 * index + 1
   }
 
   private getRightChildIndex(index: number): number {
+    if (!Number.isFinite(index)) return 0
     return 2 * index + 2
   }
 
