diff --git a/npm-app/src/client.ts b/npm-app/src/client.ts
index 95fd8a17e..cfe5bdd86 100644
--- a/npm-app/src/client.ts
+++ b/npm-app/src/client.ts
@@ -683,6 +683,7 @@ export class Client {
     }
     const userInputId =
       `mc-input-` + Math.random().toString(36).substring(2, 15)
+    loggerContext.clientRequestId = userInputId
 
     const { responsePromise, stopResponse } = this.subscribeToResponse(
       (chunk) => {
