diff --git a/packages/internal/src/openrouter-ai-sdk/openrouter-chat-language-model.ts b/packages/internal/src/openrouter-ai-sdk/openrouter-chat-language-model.ts
index 46a5e1564..35c1f00e3 100644
--- a/packages/internal/src/openrouter-ai-sdk/openrouter-chat-language-model.ts
+++ b/packages/internal/src/openrouter-ai-sdk/openrouter-chat-language-model.ts
@@ -759,7 +759,7 @@ const OpenRouterChatCompletionBaseResponseSchema = z.object({
       cost: z.number().optional(),
       cost_details: z
         .object({
-          upstream_inference_cost: z.number(),
+          upstream_inference_cost: z.number().nullish(),
         })
         .nullish(),
     })
