diff --git a/scripts/fat-sdk-openrouter-example.ts b/scripts/fat-sdk-openrouter-example.ts
index f73c6cef1..7e2ba7413 100644
--- a/scripts/fat-sdk-openrouter-example.ts
+++ b/scripts/fat-sdk-openrouter-example.ts
@@ -5,7 +5,7 @@ import {
   VERSION,
 } from '@ai-sdk/openai-compatible'
 import { websiteUrl } from '@codebuff/npm-app/config'
-import { streamText } from 'ai'
+import { generateText } from 'ai'
 
 const apiKey = '12345'
 
@@ -21,7 +21,8 @@ const codebuffBackendModel = new OpenAICompatibleChatLanguageModel(
     }),
     metadataExtractor: {
       extractMetadata: async (...inputs) => {
-        console.log(inputs, 'extractMetadata')
+        console.dir({ extractMetadata: inputs }, { depth: null })
+
         return undefined
       },
       createStreamExtractor: () => ({
@@ -43,10 +44,9 @@ const codebuffBackendModel = new OpenAICompatibleChatLanguageModel(
   },
 )
 
-const response = streamText({
-  // const response = await streamText({
-  // const response = await generateText({
-  // model: codebuffBackendProvider('openai/gpt-5'),
+// const response = streamText({
+// const response = await generateObject({
+const response = await generateText({
   model: codebuffBackendModel,
   messages: [
     {
@@ -82,6 +82,7 @@ const response = streamText({
   },
 })
 
-for await (const chunk of response.fullStream) {
-  console.dir({ chunk }, { depth: null })
-}
+// for await (const chunk of response.fullStream) {
+//   console.dir({ chunk }, { depth: null })
+// }
+console.log(response.text)
diff --git a/sdk/src/impl/llm.ts b/sdk/src/impl/llm.ts
index 23fd7794a..22213b73a 100644
--- a/sdk/src/impl/llm.ts
+++ b/sdk/src/impl/llm.ts
@@ -66,9 +66,18 @@ function getAiSdkModel(params: {
       'user-agent': `ai-sdk/openai-compatible/${VERSION}/codebuff`,
     }),
     metadataExtractor: {
-      extractMetadata: async (...inputs) => {
-        console.log(inputs, 'extractMetadata')
-        return undefined
+      extractMetadata: async ({ parsedBody }: { parsedBody: any }) => {
+        if (typeof parsedBody?.usage?.cost === 'number') {
+          openrouterUsage.cost = parsedBody.usage.cost
+        }
+        if (
+          typeof parsedBody?.usage?.cost_details?.upstream_inference_cost ===
+          'number'
+        ) {
+          openrouterUsage.cost =
+            parsedBody.usage.cost_details.upstream_inference_cost
+        }
+        return { codebuff: { usage: openrouterUsage } }
       },
       createStreamExtractor: () => ({
         processChunk: (parsedChunk: any) => {
