diff --git a/env.ts b/env.ts
index 9c4377e02..51efd941a 100644
--- a/env.ts
+++ b/env.ts
@@ -54,6 +54,7 @@ export const env = createValidatedEnv({
   client: {
     NEXT_PUBLIC_CB_ENVIRONMENT: z.string().min(1),
     NEXT_PUBLIC_APP_URL: z.string().url().min(1),
+    NEXT_PUBLIC_BACKEND_URL: z.string().url().min(1),
     NEXT_PUBLIC_SUPPORT_EMAIL: z.string().email().min(1),
     NEXT_PUBLIC_POSTHOG_API_KEY: z.string().optional().default(''),
     NEXT_PUBLIC_POSTHOG_HOST_URL: z.string().url().optional(),
@@ -97,6 +98,7 @@ export const env = createValidatedEnv({
     // Client variables
     NEXT_PUBLIC_CB_ENVIRONMENT: process.env.NEXT_PUBLIC_CB_ENVIRONMENT,
     NEXT_PUBLIC_APP_URL: process.env.NEXT_PUBLIC_APP_URL,
+    NEXT_PUBLIC_BACKEND_URL: process.env.NEXT_PUBLIC_BACKEND_URL,
     NEXT_PUBLIC_SUPPORT_EMAIL: process.env.NEXT_PUBLIC_SUPPORT_EMAIL,
     NEXT_PUBLIC_POSTHOG_API_KEY: process.env.NEXT_PUBLIC_POSTHOG_API_KEY,
     NEXT_PUBLIC_POSTHOG_HOST_URL: process.env.NEXT_PUBLIC_POSTHOG_HOST_URL,
diff --git a/npm-app/package.json b/npm-app/package.json
index 9ea6a075d..64ab14d14 100644
--- a/npm-app/package.json
+++ b/npm-app/package.json
@@ -14,6 +14,7 @@
     "format": "prettier --write \"**/*.{ts,tsx,json,md}\"",
     "test": "jest",
     "start": "infisical run --log-level=warn -- ts-node -r tsconfig-paths/register src/index.ts ..",
+    "start-js": "nx clean npm-app && nx build npm-app && infisical run --log-level=warn -- sh -c 'TS_NODE_BASEURL=./dist node -r tsconfig-paths/register dist/index.js ..'",
     "start-max": "infisical run --log-level=warn -- ts-node -r tsconfig-paths/register src/index.ts .. --max",
     "start-lite": "infisical run --log-level=warn -- ts-node -r tsconfig-paths/register src/index.ts .. --lite",
     "start-ask": "infisical run --log-level=warn -- ts-node -r tsconfig-paths/register src/index.ts .. --ask",
