diff --git a/cli/release-staging/index.js b/cli/release-staging/index.js
index 429fb16b9..2faa6058b 100644
--- a/cli/release-staging/index.js
+++ b/cli/release-staging/index.js
@@ -137,9 +137,9 @@ function getCurrentVersion() {
           if (!child.killed) {
             child.kill('SIGKILL')
           }
-        }, 1000)
+        }, 4000)
         resolve('error')
-      }, 1000)
+      }, 4000)
 
       child.on('exit', (code) => {
         clearTimeout(timeout)
