diff --git a/backend/package.json b/backend/package.json
index 1c9c79462..3d8612ad8 100644
--- a/backend/package.json
+++ b/backend/package.json
@@ -23,7 +23,7 @@
     "express": "4.19.2",
     "lodash": "*",
     "openai": "^4.56.0",
-    "tiktoken": "1.0.16",
+    "tiktoken": "1.0.15",
     "ws": "8.18.0",
     "zod": "3.23.8"
   },
diff --git a/backend/src/util/token-counter.ts b/backend/src/util/token-counter.ts
index 9f4836864..b5db207c2 100644
--- a/backend/src/util/token-counter.ts
+++ b/backend/src/util/token-counter.ts
@@ -4,10 +4,15 @@ export function countTokens(
   text: string,
   model: TiktokenModel = 'gpt-4o'
 ): number {
-  const encoder = encoding_for_model(model)
-  const tokens = encoder.encode(text)
-  encoder.free()
-  return tokens.length
+  try {
+    const encoder = encoding_for_model(model)
+    const tokens = encoder.encode(text)
+    encoder.free()
+    return tokens.length
+  } catch (e) {
+    console.error('Error counting tokens', e)
+    return Math.ceil(text.length / 3)
+  }
 }
 
 export function countTokensForFiles(
