diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 467c4e705..980578804 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -31,7 +31,11 @@ import {
   checkConversationProgress,
   checkToAllowUnboundedIteration,
 } from './conversation-progress'
-import { getRelevantFilesForPlanning, planComplexChange } from './planning'
+import {
+  getRelevantFilesForPlanning,
+  loadFilesForPlanning,
+  planComplexChange,
+} from './planning'
 import { promptDeepseekStream } from './deepseek-api'
 import { messagesWithSystem } from '@/util/messages'
 
@@ -322,19 +326,16 @@ export async function mainPrompt(
         userId
       )
       const fetchFilesDuration = Date.now() - fetchFilesStart
-      const loadedFiles = await requestFiles(ws, filePaths)
-      const fileContents = Object.fromEntries(
-        Object.entries(loadedFiles).filter(
-          ([, content]) => content !== null
-        ) as [string, string][]
-      )
-
+      const fileContents = await loadFilesForPlanning(ws, filePaths)
       const existingFilePaths = Object.keys(fileContents)
+
       onResponseChunk(`\nRelevant files:\n${existingFilePaths.join(' ')}\n`)
       fullResponse += `\nRelevant files:\n${existingFilePaths.join('\n')}\n`
       onResponseChunk(`\nThinking deeply (can take a minute)...\n\n`)
+
       logger.debug({ prompt, filePaths, existingFilePaths }, 'Thinking deeply')
       const planningStart = Date.now()
+
       const { response, fileProcessingPromises: promises } =
         await planComplexChange(
           prompt,
diff --git a/backend/src/planning.ts b/backend/src/planning.ts
index fcc4099e1..148492ef6 100644
--- a/backend/src/planning.ts
+++ b/backend/src/planning.ts
@@ -1,6 +1,7 @@
+import { WebSocket } from 'ws'
 import { Message, FileChange } from 'common/actions'
 import { models, claudeModels, CostMode } from 'common/constants'
-import { countTokensJson } from './util/token-counter'
+import { countTokens, countTokensJson } from './util/token-counter'
 import {
   createFileBlock,
   createMarkdownFileBlock,
@@ -13,6 +14,7 @@ import { OpenAIMessage, promptOpenAI } from './openai-api'
 import { getSearchSystemPrompt } from './system-prompt'
 import { hasLazyEdit } from 'common/util/string'
 import { applyRemainingChanges } from './process-file-block'
+import { requestFiles } from './websockets/websocket-action'
 
 const systemPrompt = `
 You are a senior software engineer. You are given a request from a user and a set of files that are relevant to the request.
@@ -184,3 +186,32 @@ Only output the file paths, one per line, nothing else.`,
     .map((line) => line.trim())
     .filter((line) => isValidFilePath(line))
 }
+
+export async function loadFilesForPlanning(ws: WebSocket, filePaths: string[]) {
+  const loadedFiles = await requestFiles(ws, filePaths)
+  const fileContents = Object.fromEntries(
+    Object.entries(loadedFiles).filter(
+      ([, content]) =>
+        content !== null &&
+        content !== '[INVALID_FILE_PATH]' &&
+        countTokens(content) < 40_000
+    ) as [string, string][]
+  )
+
+  const maxFileTokens = 140_000
+
+  let totalTokens = 0
+  const filesToRemove: string[] = []
+  for (const [filePath, content] of Object.entries(fileContents)) {
+    if (totalTokens < maxFileTokens) {
+      totalTokens += countTokens(content)
+    }
+    if (totalTokens >= maxFileTokens) {
+      filesToRemove.push(filePath)
+    }
+  }
+  for (const filePath of filesToRemove) {
+    delete fileContents[filePath]
+  }
+  return fileContents
+}
diff --git a/backend/src/util/token-counter.ts b/backend/src/util/token-counter.ts
index 239932841..64d696ea3 100644
--- a/backend/src/util/token-counter.ts
+++ b/backend/src/util/token-counter.ts
@@ -3,7 +3,7 @@ import { LRUCache } from 'common/util/lru-cache'
 
 const ANTHROPIC_TOKEN_FUDGE_FACTOR = 1.35
 
-const TOKEN_COUNT_CACHE = new LRUCache<string, number>(10000) // Cache up to 10k strings
+const TOKEN_COUNT_CACHE = new LRUCache<string, number>(5000) // Cache up to 5k strings
 
 export function countTokens(text: string): number {
   try {
