diff --git a/web/src/app/api/v1/agents/[publisherId]/[agentId]/[version]/route.ts b/web/src/app/api/v1/agents/[publisherId]/[agentId]/[version]/route.ts
new file mode 100644
index 000000000..4c0ce4755
--- /dev/null
+++ b/web/src/app/api/v1/agents/[publisherId]/[agentId]/[version]/route.ts
@@ -0,0 +1,5 @@
+import { GET as wrapped } from '@/app/api/agents/[publisherId]/[agentId]/[version]/route'
+
+export function GET(...args: Parameters<typeof wrapped>) {
+  return wrapped(...args)
+}
diff --git a/web/src/app/api/v1/agents/[publisherId]/[agentId]/latest/route.ts b/web/src/app/api/v1/agents/[publisherId]/[agentId]/latest/route.ts
new file mode 100644
index 000000000..0c4fcfc6c
--- /dev/null
+++ b/web/src/app/api/v1/agents/[publisherId]/[agentId]/latest/route.ts
@@ -0,0 +1,5 @@
+import { GET as wrapped } from '@/app/api/agents/[publisherId]/[agentId]/latest/route'
+
+export function GET(...args: Parameters<typeof wrapped>) {
+  return wrapped(...args)
+}
