diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 4accf3558..27213419d 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -167,6 +167,9 @@ export class CLI {
         process.removeAllListeners('uncaughtException')
         Spinner.get().restoreCursor()
         await killAllBackgroundProcesses()
+
+        Client.getInstance().close()
+
         await flushAnalytics()
         process.exit(0)
       })
@@ -865,6 +868,8 @@ export class CLI {
 
     await killAllBackgroundProcesses()
 
+    Client.getInstance().close() // Close WebSocket
+
     const client = Client.getInstance()
     const logMessages = []
     const totalCreditsUsedThisSession = Object.values(client.creditsByPromptId)
diff --git a/npm-app/src/client.ts b/npm-app/src/client.ts
index 2be72dd7c..42c51b8d6 100644
--- a/npm-app/src/client.ts
+++ b/npm-app/src/client.ts
@@ -651,6 +651,10 @@ export class Client {
     loggerContext.costMode = this.costMode
   }
 
+  public close() {
+    this.webSocket.close()
+  }
+
   private setupSubscriptions() {
     this.webSocket.subscribe('action-error', (action) => {
       if (action.error === 'Insufficient credits') {
