diff --git a/web/src/content/advanced/troubleshooting.mdx b/web/src/content/advanced/troubleshooting.mdx
index 94b7f122b..97804edb2 100644
--- a/web/src/content/advanced/troubleshooting.mdx
+++ b/web/src/content/advanced/troubleshooting.mdx
@@ -9,6 +9,14 @@ order: 1
 
 [Open an issue](https://github.com/CodebuffAI/codebuff/issues) or check existing reports.
 
+## Install Node
+
+If you don't have `npm`, [get Node here](https://nodejs.org/en/download).
+
+## Delete local cache
+
+Codebuff stores files at `~/.config/manicode` including the binary `codebuff`. Sometimes deleting the binary and restarting codebuff will fix issues.
+
 ## Version
 
 The latest version of the Codebuff client can be found on the [npm page](https://www.npmjs.com/package/codebuff).
@@ -17,10 +25,6 @@ Run `codebuff --version` to make sure you're on the latest version.
 
 Codebuff should auto-update. If not, try the steps below.
 
-## Codebuff not updating
-
-If you see the npm install run but Codebuff still isn't at the latest version, this is most likely an issue with npm's cache. Try running `npm cache clean --force` and then `npm install -g codebuff`.
-
 ## Install failed
 
 If `npm install -g codebuff` gave you an error regarding permissions, try using setting the owner of that directory to your user.
@@ -39,27 +43,6 @@ Include recent chats if you ask for help.
 
 ## Common Issues
 
-### Connection Issues
-
-If you see a message like this:
-
-```bash
-/Users/<youruser>/Library/pnpm/global/5/.pnpm/codebuff@someversion/node_modules/codebuff/dist/common/websockets/websocket-client.js:158
-                    reject(new Error(`Websocket message with txid ${txid} timed out.`));                           ^
-
-Error: Websocket message with txid 32 timed out.
-    at Timeout._onTimeout (/Users/<youruser>/Library/pnpm/global/5/.pnpm/codebuff@someversion/node_modules/codebuff/dist/common/websockets/websocket-client.js:158:28)
-    at listOnTimeout (node:internal/timers:594:17)
-    at process.processTimers (node:internal/timers:529:7)
-
-Node.js v22.12.0
-```
-
-it means you're having connection issues. In that case, try:
-
-1. Check your internet connection.
-2. Try logging out and back in with `logout` followed by `login`.
-
 ### Endless Auto-Updates
 
 If Codebuff keeps trying to update itself in a loop, check the items below:
