diff --git a/web/src/app/referrals/[code]/page.tsx b/web/src/app/referrals/[code]/page.tsx
index 6b901f8e2..de41095d7 100644
--- a/web/src/app/referrals/[code]/page.tsx
+++ b/web/src/app/referrals/[code]/page.tsx
@@ -106,7 +106,7 @@ export default function RedeemPage({ params }: { params: { code: string } }) {
             </li>
             <li>
               Run Codebuff in Terminal
-              <InputWithCopyButton text={'manicode'} />
+              <InputWithCopyButton text={'codebuff'} />
             </li>
             {!data?.status.reason && (
               <li>
