diff --git a/npm-app/src/utils/__tests__/background-process-manager.test.ts b/npm-app/src/utils/__tests__/background-process-manager.test.ts
index 9ee1588d6..9a9456f79 100644
--- a/npm-app/src/utils/__tests__/background-process-manager.test.ts
+++ b/npm-app/src/utils/__tests__/background-process-manager.test.ts
@@ -7,6 +7,7 @@ import {
   mock,
   spyOn,
   test,
+  // @ts-ignore
 } from 'bun:test'
 
 import {
