diff --git a/backend/src/util/logger.ts b/backend/src/util/logger.ts
index a70ab042a..3d2039943 100644
--- a/backend/src/util/logger.ts
+++ b/backend/src/util/logger.ts
@@ -25,4 +25,10 @@ export const logger = pino({
   mixin() {
     return loggerAsyncStorage.getStore() ?? {}
   },
+  formatters: {
+    level: (label) => {
+      return { level: label.toUpperCase() }
+    },
+  },
+  timestamp: () => `,"timestamp":"${new Date(Date.now()).toISOString()}"`,
 })
