diff --git a/musings.knowledge.md b/musings.knowledge.md
index 0d832ff55..a3fc63a51 100644
--- a/musings.knowledge.md
+++ b/musings.knowledge.md
@@ -181,7 +181,7 @@ Problems
 
 - Dawson's case of wanting it to scrape documentation from a url and answer questions with it.
 - x% knowledge written in menu based on number of commits to knowledge files
-- How it could run bash:
+[x] How it could run bash:
 1. First, let's build the `common` package:
 
 ```bash
@@ -208,4 +208,7 @@ Notes from Janna & Stephen test
 [x] backspace on windows doesn't clear the char, just moves cursor
 [x] Tried to do left arrow and it didn't work
 [x] Failed to apply any edits. They all failed even after a second attempt. This was on Windows
-[x] CTRL-C doesn't work
\ No newline at end of file
+[x] CTRL-C doesn't work
+
+[] Kill if it continues too far without user prompt.
+[] Give it a new marker token that will await all file changes, so then it can run tsc or tests, etc.
diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 1a935eb1c..50cf8f584 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -25,10 +25,10 @@ export class CLI {
       historySize: 1000,
     })
 
-    this.rl.prompt()
-
     this.rl.on('line', (line) => {
-      this.handleUserInput(line.trim()).then(() => this.rl.prompt())
+      if (!this.isReceivingResponse) {
+        this.handleUserInput(line.trim())
+      }
     })
 
     this.rl.on('SIGINT', () => {
@@ -50,8 +50,9 @@ export class CLI {
     })
   }
 
-  private printInitialPrompt() {
+  public printInitialPrompt() {
     console.log('What would you like to do? Press ESC for menu.')
+    this.rl.prompt()
   }
 
   private handleUndo() {
@@ -99,7 +100,6 @@ export class CLI {
       this.isInMenu = false
       console.clear()
       this.printInitialPrompt()
-      this.rl.prompt()
     } else {
       displayMenu()
       this.isInMenu = true
@@ -160,7 +160,7 @@ export class CLI {
     }
     if (changesSuceeded.length > 0) {
       console.log('Complete!\n')
-    }
+    } else console.log()
 
     const assistantMessage: Message = {
       role: 'assistant',
@@ -173,6 +173,8 @@ export class CLI {
 
     const updatedFiles = getExistingFiles(filesChanged)
     this.chatStorage.addNewFileState(updatedFiles)
+
+    this.rl.prompt()
   }
 
   private async sendUserInputAndAwaitResponse() {
diff --git a/npm-app/src/index.ts b/npm-app/src/index.ts
index 902f298b9..b695d9988 100644
--- a/npm-app/src/index.ts
+++ b/npm-app/src/index.ts
@@ -7,8 +7,6 @@ import { CLI } from './cli'
 import { getProjectFileContext } from './project-files'
 
 async function manicode(userPrompt: string | undefined) {
-  console.log('What would you like to do? Press ESC for menu.')
-
   // Preload.
   getProjectFileContext()
 
@@ -21,6 +19,8 @@ async function manicode(userPrompt: string | undefined) {
 
   if (userPrompt) {
     await cli.handleUserInput(userPrompt)
+  } else {
+    cli.printInitialPrompt()
   }
 }
 
