diff --git a/backend/src/websockets/websocket-action.ts b/backend/src/websockets/websocket-action.ts
index 0f3e41bd8..26b5d19ac 100644
--- a/backend/src/websockets/websocket-action.ts
+++ b/backend/src/websockets/websocket-action.ts
@@ -148,6 +148,7 @@ const onPrompt = async (
 
       if (prompt) {
         logger.info({ prompt }, `USER INPUT: ${prompt.slice(0, 100)}`)
+        logger.info({ prompt }, `TEST! USER INPUT: ${prompt.slice(0, 100)}`)
         trackEvent(AnalyticsEvent.USER_INPUT, userId, {
           prompt,
           promptId,
