diff --git a/npm-app/src/tool-handlers.ts b/npm-app/src/tool-handlers.ts
index 0481556af..4b5849a46 100644
--- a/npm-app/src/tool-handlers.ts
+++ b/npm-app/src/tool-handlers.ts
@@ -119,10 +119,10 @@ export const handleCodeSearch: ToolHandler<{ pattern: string }> = async (
     let stderr = ''
 
     const basename = path.basename(projectPath)
-    const pattern = parameters.pattern.replace(/"/g, '')
-    const command = `${path.resolve(rgPath)} "${pattern}" .`
+    const pattern = JSON.stringify(parameters.pattern)
+    const command = `${path.resolve(rgPath)} ${pattern}.`
     console.log()
-    console.log(green(`Searching ${basename} for "${pattern}":`))
+    console.log(green(`Searching ${basename} for ${pattern}:`))
     const childProcess = spawn(command, {
       cwd: projectPath,
       shell: true,
