diff --git a/common/src/constants.ts b/common/src/constants.ts
index a25e70d2c..5f5a35dfd 100644
--- a/common/src/constants.ts
+++ b/common/src/constants.ts
@@ -37,6 +37,7 @@ export const SKIPPED_TERMINAL_COMMANDS = [
   'add',
   'hey',
   'diff',
+  'make',
 ]
 
 export const MAX_DATE = new Date(86399999999999)
diff --git a/common/src/project-file-tree.ts b/common/src/project-file-tree.ts
index ac22cf8ab..1696af7e2 100644
--- a/common/src/project-file-tree.ts
+++ b/common/src/project-file-tree.ts
@@ -12,6 +12,7 @@ export function getProjectFileTree(
   const defaultIgnore = ignore.default()
   defaultIgnore.add('.git')
   defaultIgnore.add('*.min.*')
+  defaultIgnore.add('node_modules')
 
   if (projectRoot === os.homedir()) {
     defaultIgnore.add('.*')
diff --git a/common/src/websockets/websocket-client.ts b/common/src/websockets/websocket-client.ts
index e4cfb70c5..07f862d66 100644
--- a/common/src/websockets/websocket-client.ts
+++ b/common/src/websockets/websocket-client.ts
@@ -12,7 +12,7 @@ import {
 const VERBOSE_LOGGING = false
 
 // mqp: no way should our server ever take 5 seconds to reply
-const TIMEOUT_MS = 5000
+const TIMEOUT_MS = 10000
 
 const RECONNECT_WAIT_MS = 5000
 
diff --git a/npm-app/src/project-files.ts b/npm-app/src/project-files.ts
index 9a309678f..190794af0 100644
--- a/npm-app/src/project-files.ts
+++ b/npm-app/src/project-files.ts
@@ -150,14 +150,15 @@ export function getChangesSinceLastFileVersion(
 
 export function getFiles(filePaths: string[]) {
   const result: Record<string, string | null> = {}
-  const MAX_FILE_SIZE = 5 * 1024 * 1024 // 5MB in bytes
+  const MAX_FILE_SIZE = 1024 * 1024 // 1MB in bytes
 
   for (const filePath of filePaths) {
     const fullPath = path.join(projectRoot, filePath)
     try {
       const stats = fs.statSync(fullPath)
       if (stats.size > MAX_FILE_SIZE) {
-        result[filePath] = `[FILE_TOO_LARGE: ${(stats.size / (1024 * 1024)).toFixed(2)}MB]`
+        result[filePath] =
+          `[FILE_TOO_LARGE: ${(stats.size / (1024 * 1024)).toFixed(2)}MB]`
       } else {
         const content = fs.readFileSync(fullPath, 'utf8')
         result[filePath] = content
