diff --git a/backend/src/claude.ts b/backend/src/claude.ts
index b9bc244a6..2793852c3 100644
--- a/backend/src/claude.ts
+++ b/backend/src/claude.ts
@@ -7,6 +7,7 @@ import { RATE_LIMIT_POLICY } from './constants'
 import { env } from './env.mjs'
 import { saveMessage } from './billing/message-cost-tracker'
 import { logger } from './util/logger'
+import { randBoolFromStr } from 'common/util/string'
 
 export type model_types = (typeof claudeModels)[keyof typeof claudeModels]
 
@@ -38,8 +39,10 @@ export const promptClaudeStream = async function* (
     ignoreHelicone = false,
   } = options
 
-  const apiKey =
-    Math.random() > 0.5 ? env.ANTHROPIC_API_KEY : env.ANTHROPIC_API_KEY2
+  // Randomly choose between the two API keys to improve rate limits.
+  const apiKey = randBoolFromStr(fingerprintId)
+    ? env.ANTHROPIC_API_KEY
+    : env.ANTHROPIC_API_KEY2
 
   if (!apiKey) {
     throw new Error('Missing ANTHROPIC_API_KEY')
diff --git a/common/src/util/string.ts b/common/src/util/string.ts
index 1a2f2d6a6..e7b245d55 100644
--- a/common/src/util/string.ts
+++ b/common/src/util/string.ts
@@ -1,3 +1,5 @@
+import { sumBy } from 'lodash'
+
 export const truncateString = (str: string, maxLength: number) => {
   if (str.length <= maxLength) {
     return str
@@ -15,7 +17,8 @@ export const replaceNonStandardPlaceholderComments = (
       placeholder: '// ... existing code ...',
     },
     {
-      regex: /\/\*\s*\.{3}\s*.*(?:rest|unchanged|keep|file).*(?:\s*\.{3})?\s*\*\//gi,
+      regex:
+        /\/\*\s*\.{3}\s*.*(?:rest|unchanged|keep|file).*(?:\s*\.{3})?\s*\*\//gi,
       placeholder: '/* ... existing code ... */',
     },
     // Python, Ruby, R comments
@@ -25,7 +28,8 @@ export const replaceNonStandardPlaceholderComments = (
     },
     // HTML-style comments
     {
-      regex: /<!--\s*\.{3}\s*.*(?:rest|unchanged|keep|file).*(?:\s*\.{3})?\s*-->/gi,
+      regex:
+        /<!--\s*\.{3}\s*.*(?:rest|unchanged|keep|file).*(?:\s*\.{3})?\s*-->/gi,
       placeholder: '<!-- ... existing code ... -->',
     },
     // SQL, Haskell, Lua comments
@@ -48,3 +52,7 @@ export const replaceNonStandardPlaceholderComments = (
 
   return updatedContent
 }
+
+export const randBoolFromStr = (str: string) => {
+  return sumBy(str.split(''), (char) => char.charCodeAt(0)) % 2 === 0
+}
