diff --git a/cli/scripts/build-binary.ts b/cli/scripts/build-binary.ts
index eb39d5bb7..7f941dfa3 100644
--- a/cli/scripts/build-binary.ts
+++ b/cli/scripts/build-binary.ts
@@ -156,6 +156,7 @@ async function main() {
     `--outfile=${outputFile}`,
     '--sourcemap=none',
     ...defineFlags.flatMap(([key, value]) => ['--define', `${key}=${value}`]),
+    '--env "NEXT_PUBLIC_*"', // Copies all current env vars in process.env to the compiled binary that match the pattern.
   ]
 
   log(
