diff --git a/common/src/env.mjs b/common/src/env.mjs
index 54c763195..64571c631 100644
--- a/common/src/env.mjs
+++ b/common/src/env.mjs
@@ -21,14 +21,12 @@ export const env = createEnv({
     NEXT_PUBLIC_ENVIRONMENT: z.string().min(1),
     DATABASE_URL: z.string().min(1),
     STRIPE_SECRET_KEY: z.string().min(1),
+    NEXT_PUBLIC_SUPPORT_EMAIL: z.string().min(1),
   },
   client: {
     NEXT_PUBLIC_ENVIRONMENT: z.string().min(1),
     NEXT_PUBLIC_APP_URL: z.string().min(1),
+    NEXT_PUBLIC_SUPPORT_EMAIL: z.string().min(1),
   },
-  runtimeEnv: {
-    ...process.env,
-    NEXT_PUBLIC_APP_URL: process.env.NEXT_PUBLIC_APP_URL,
-    NEXT_PUBLIC_ENVIRONMENT: process.env.NEXT_PUBLIC_ENVIRONMENT,
-  },
+  runtimeEnv: process.env,
 })
diff --git a/web/src/app/referrals/page.tsx b/web/src/app/referrals/page.tsx
index ff547223d..7537e72a7 100644
--- a/web/src/app/referrals/page.tsx
+++ b/web/src/app/referrals/page.tsx
@@ -135,8 +135,9 @@ const ReferralsPage = () => {
                   <div className="flex flex-col space-y-4">
                     {data.limitReached ? (
                       <p>
-                        You have reached your referral limit. Dang, you've got a
-                        lot of friends! Maybe introduce them to one another? 🌝
+                        You have reached your referral limit. Dang, you&apos;ve
+                        got a lot of friends! Maybe introduce them to one
+                        another? 🌝
                       </p>
                     ) : (
                       <>
