diff --git a/npm-app/src/cli-handlers/publish.ts b/npm-app/src/cli-handlers/publish.ts
index 647471ce9..ffd234e46 100644
--- a/npm-app/src/cli-handlers/publish.ts
+++ b/npm-app/src/cli-handlers/publish.ts
@@ -233,10 +233,10 @@ async function publishAgentTemplate(
       method: 'POST',
       headers: {
         'Content-Type': 'application/json',
-        Cookie: `next-auth.session-token=${authToken}`,
       },
       body: JSON.stringify({
         data,
+        authToken,
         ...(publisherId && { publisherId }),
       }),
     })
