diff --git a/web/src/app/publishers/[id]/agents/[agentId]/[version]/page.tsx b/web/src/app/publishers/[id]/agents/[agentId]/[version]/page.tsx
index 19885181f..e4db34366 100644
--- a/web/src/app/publishers/[id]/agents/[agentId]/[version]/page.tsx
+++ b/web/src/app/publishers/[id]/agents/[agentId]/[version]/page.tsx
@@ -179,10 +179,13 @@ const AgentDetailPage = async ({ params }: AgentDetailPageProps) => {
                     className="p-1 text-muted-foreground/60 hover:text-muted-foreground"
                   />
                 </div>
+                {/*
+                Hide download button for now. (It doesn't do anything)
                 <Button variant="outline" size="sm">
                   <Download className="h-4 w-4 mr-2" />
                   Download
                 </Button>
+                */}
               </div>
             </div>
           </CardHeader>
