diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index ec03eff1f..bfd33ff34 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -628,15 +628,11 @@ export class CLI {
       resetShell(getProjectRoot())
     }
 
-    if ('line' in this.rl) {
-      ;(this.rl as any).line = ''
-    }
-
     if (this.isReceivingResponse) {
       this.handleStopResponse()
     } else {
       const now = Date.now()
-      if (now - this.lastSigintTime < 5000) {
+      if (now - this.lastSigintTime < 5000 && !this.rl.line) {
         await this.handleExit()
       } else {
         this.lastSigintTime = now
