diff --git a/common/src/util/stripe.ts b/common/src/util/stripe.ts
index a7767251f..a918eec31 100644
--- a/common/src/util/stripe.ts
+++ b/common/src/util/stripe.ts
@@ -6,11 +6,11 @@ export const stripeServer = new Stripe(env.STRIPE_SECRET_KEY, {
   apiVersion: '2024-06-20',
 })
 
-stripeServer.on('request', (request) => {
-  console.log('Stripe API Request', request)
-})
+// stripeServer.on('request', (request) => {
+//   console.log('Stripe API Request', request)
+// })
 
-// Listen for response events
-stripeServer.on('response', (response) => {
-  console.log('Stripe API Response', response)
-})
+// // Listen for response events
+// stripeServer.on('response', (response) => {
+//   console.log('Stripe API Response', response)
+// })
