diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index aea1cafcb..66b089486 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -1,15 +1,15 @@
 import { uniqBy } from 'lodash'
 
 import { ChatStorage } from './chat-storage'
-import { ChatClient } from './chat-client'
+import { Client } from './client'
 import { Message, FileChanges } from 'common/actions'
 import { displayMenu } from './menu'
 import { applyChanges, getExistingFiles, setFiles } from './project-files'
 import { STOP_MARKER } from 'common/constants'
 
 export class CLI {
+  private client: Client
   private chatStorage: ChatStorage
-  private chatClient: ChatClient
   private inputBuffer: string = ''
   private history: string[] = []
   private historyIndex: number = -1
@@ -20,9 +20,9 @@ export class CLI {
   private isInMenu: boolean = false
   private savedInput: string = '' // New property to store the current input when navigating history
 
-  constructor(chatStorage: ChatStorage, wsClient: ChatClient) {
+  constructor(client: Client, chatStorage: ChatStorage) {
+    this.client = client
     this.chatStorage = chatStorage
-    this.chatClient = wsClient
   }
 
   start() {
@@ -247,7 +247,7 @@ export class CLI {
       response: string
       changes: FileChanges
     }>(async (resolve) => {
-      const { unsubscribe, result } = this.chatClient.subscribeToResponse(
+      const { unsubscribe, result } = this.client.subscribeToResponse(
         (chunk) => {
           if (this.stopResponseRequested) {
             unsubscribe()
@@ -267,7 +267,7 @@ export class CLI {
         }
       )
 
-      this.chatClient.sendUserInput([])
+      this.client.sendUserInput([])
 
       resolve(await result)
     })
diff --git a/npm-app/src/chat-client.ts b/npm-app/src/client.ts
similarity index 84%
rename from npm-app/src/chat-client.ts
rename to npm-app/src/client.ts
index d6359063e..5ba74af98 100644
--- a/npm-app/src/chat-client.ts
+++ b/npm-app/src/client.ts
@@ -4,22 +4,22 @@ import { ChatStorage } from './chat-storage'
 import { FileChanges, Message } from 'common/actions'
 import { toolHandlers } from './tool-handlers'
 
-export class ChatClient {
-  private ws: APIRealtimeClient
+export class Client {
+  private webSocket: APIRealtimeClient
   private chatStorage: ChatStorage
 
   constructor(websocketUrl: string, chatStorage: ChatStorage) {
-    this.ws = new APIRealtimeClient(websocketUrl)
+    this.webSocket = new APIRealtimeClient(websocketUrl)
     this.chatStorage = chatStorage
   }
 
   async connect() {
-    await this.ws.connect()
+    await this.webSocket.connect()
     this.setupSubscriptions()
   }
 
   private setupSubscriptions() {
-    this.ws.subscribe('tool-call', async (a) => {
+    this.webSocket.subscribe('tool-call', async (a) => {
       const { response, changes, data } = a
       const { id, name, input } = data
 
@@ -66,11 +66,11 @@ export class ChatClient {
       }
     })
 
-    this.ws.subscribe('read-files', (a) => {
+    this.webSocket.subscribe('read-files', (a) => {
       const { filePaths } = a
       const files = getFiles(filePaths)
 
-      this.ws.sendAction({
+      this.webSocket.sendAction({
         type: 'read-files-response',
         files,
       })
@@ -79,7 +79,7 @@ export class ChatClient {
 
   async sendUserInput(previousChanges: FileChanges) {
     const fileContext = await getProjectFileContext()
-    this.ws.sendAction({
+    this.webSocket.sendAction({
       type: 'user-input',
       messages: this.chatStorage.getCurrentChat().messages,
       fileContext,
@@ -97,12 +97,12 @@ export class ChatClient {
       response: string
       changes: FileChanges
     }>((resolve) => {
-      const unsubscribeChunks = this.ws.subscribe('response-chunk', (a) => {
+      const unsubscribeChunks = this.webSocket.subscribe('response-chunk', (a) => {
         const { chunk } = a
         onChunk(chunk)
       })
 
-      const unsubscribeComplete = this.ws.subscribe(
+      const unsubscribeComplete = this.webSocket.subscribe(
         'response-complete',
         (a) => {
           unsubscribeChunks()
diff --git a/npm-app/src/index.ts b/npm-app/src/index.ts
index 916685d2f..67cd5d2e0 100644
--- a/npm-app/src/index.ts
+++ b/npm-app/src/index.ts
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 
 import { websocketUrl } from './config'
-import { ChatClient } from './chat-client'
+import { Client } from './client'
 import { ChatStorage } from './chat-storage'
 import { CLI } from './cli'
 import { getProjectFileContext } from './project-files'
@@ -13,9 +13,9 @@ async function manicode(userPrompt: string | undefined) {
   getProjectFileContext()
 
   const chatStorage = new ChatStorage()
-  const client = new ChatClient(websocketUrl, chatStorage)
+  const client = new Client(websocketUrl, chatStorage)
 
-  const cli = new CLI(chatStorage, client)
+  const cli = new CLI(client, chatStorage)
   cli.start()
 
   await client.connect()
