diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 10026ae10..6b31737ec 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -165,7 +165,7 @@ export const mainPrompt = async (
 
     'When editing an existing file, write just the parts of the file that have changed. Do not start writing the first line of the file. Instead, use comments surrounding your edits like "// ... existing code ..." (or "# ... existing code ..." or "/* ... existing code ... */" or "<!-- ... existing code ... -->", whichever is appropriate for the language) plus a few lines of context from the original file.',
 
-    'When using tools, make sure to NOT use XML attributes. The format should contain nested XML tags. For example, when using write_file, the format should be <write_file><path>...</path><content>...</content></write_file>',
+    'If you are trying to kill background processes, make sure to kill the entire process GROUP (or tree in Windows), and always prefer SIGTERM signals. If you restart the process, make sure to do so with process_type=BACKGROUND',
 
     `Only use the tools listed, (i.e. ${TOOL_LIST.join(', ')}). If you use tools not listed, nothing will happen, but the user will get some unintended display issues.`,
 
diff --git a/backend/src/system-prompt/prompts.ts b/backend/src/system-prompt/prompts.ts
index 63798a252..edfc67b72 100644
--- a/backend/src/system-prompt/prompts.ts
+++ b/backend/src/system-prompt/prompts.ts
@@ -11,11 +11,21 @@ import { truncateString } from 'common/util/string'
 import { truncateFileTreeBasedOnTokenBudget } from './truncate-file-tree'
 
 export const configSchemaPrompt = `
-  # Codebuff Configuration (${codebuffConfigFile})
+# Codebuff Configuration (${codebuffConfigFile})
+
+## Schema
 
 The following describes the structure of the \`./${codebuffConfigFile}\` configuration file that users might have in their project root. You can use this to understand user settings if they mention them.
 
 ${stringifySchema(CodebuffConfigSchema)}
+
+## Background Processes
+
+The user does not have access to these outputs. Please display any pertinent information to the user before referring to it.
+
+To stop a background process, attempt to close the process using the appropriate command. If you deem that command to be \`kill\`, **make sure** to kill the **ENTIRE PROCESS GROUP** (Mac/Linux) or tree (Windows).
+
+When you want to restart a background process, make sure to run the terminal command in the background.
 `.trim()
 
 export const knowledgeFilesPrompt = `
