diff --git a/backend/src/util/logger.ts b/backend/src/util/logger.ts
index 3d2039943..cb1a1a984 100644
--- a/backend/src/util/logger.ts
+++ b/backend/src/util/logger.ts
@@ -23,7 +23,7 @@ export const logger = pino({
           },
         },
   mixin() {
-    return loggerAsyncStorage.getStore() ?? {}
+    return { ...loggerAsyncStorage.getStore() }
   },
   formatters: {
     level: (label) => {
