diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 54033a773..ab1ab8bdf 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -462,7 +462,12 @@ export const mainPrompt = async (
     },
 
     prompt && [
-      cwd && { role: 'user' as const, content: asSystemMessage(`cwd: ${cwd}`) },
+      cwd && {
+        role: 'user' as const,
+        content: asSystemMessage(
+          `**user** cwd: ${cwd} (assistant cwd is project root)`
+        ),
+      },
       {
         role: 'user' as const,
         content: prompt,
