diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 00c858186..6ee635a06 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -181,9 +181,11 @@ ${STOP_MARKER}
           toolCall.input['prompt'],
           userId
         )
-        const responseChunk = '\n' + getRelevantFileInfoMessage(relevantFiles)
-        onResponseChunk(responseChunk)
-        fullResponse += responseChunk
+        if (relevantFiles !== null && relevantFiles.length > 0) {
+          const responseChunk = '\n' + getRelevantFileInfoMessage(relevantFiles)
+          onResponseChunk(responseChunk)
+          fullResponse += responseChunk
+        }
       }
       isComplete = true
     } else {
@@ -261,7 +263,7 @@ async function updateFileContext(
     userId
   )
 
-  if (relevantFiles.length === 0) {
+  if (relevantFiles === null || relevantFiles.length === 0) {
     return null
   }
 
diff --git a/backend/src/request-files-prompt.ts b/backend/src/request-files-prompt.ts
index 40fdba30d..78d09f318 100644
--- a/backend/src/request-files-prompt.ts
+++ b/backend/src/request-files-prompt.ts
@@ -21,7 +21,7 @@ export async function requestRelevantFiles(
   fileContext: ProjectFileContext,
   assistantPrompt: string | null,
   userId: string
-): Promise<string[]> {
+) {
   const previousFiles = Object.keys(fileContext.files)
   const countPerRequest = assistantPrompt ? 8 : 4
 
@@ -59,7 +59,7 @@ export async function requestRelevantFiles(
 
   if (!newFilesNecessary) {
     debugLog('No new files necessary, keeping current files')
-    return previousFiles
+    return null
   }
 
   const newFiles = await fileRequestsPromise
