diff --git a/common/src/websockets/websocket-client.ts b/common/src/websockets/websocket-client.ts
index f84b206fd..b3ccc363c 100644
--- a/common/src/websockets/websocket-client.ts
+++ b/common/src/websockets/websocket-client.ts
@@ -1,7 +1,6 @@
 import { WebSocket } from 'ws'
-import { green } from 'picocolors'
 
-import { ServerAction, ClientAction, SERVER_ACTION_SCHEMA } from '../actions'
+import { ServerAction, ClientAction } from '../actions'
 import {
   ClientMessage,
   ClientMessageType,
@@ -62,13 +61,15 @@ export class APIRealtimeClient {
   heartbeat?: any
   hadError = false
   onError: () => void
+  onReconnect: () => void
 
-  constructor(url: string, onError: () => void) {
+  constructor(url: string, onError: () => void, onReconnect: () => void) {
     this.url = url
     this.txid = 0
     this.txns = new Map()
     this.subscribers = new Map()
     this.onError = onError
+    this.onReconnect = onReconnect
   }
 
   get state() {
@@ -87,7 +88,7 @@ export class APIRealtimeClient {
     this.ws.onmessage = (ev) => {
       if (this.hadError) {
         this.hadError = false
-        console.log(green('Reconnected!'))
+        this.onReconnect()
       }
       this.receiveMessage(JSON.parse(ev.data as any))
     }
diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 9a1c2a877..7373be01a 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -93,18 +93,13 @@ export class CLI {
         return [matchingTokens, lastWord]
       },
     })
+
     this.client = new Client(
       websocketUrl,
       this.chatStorage,
       this.onWebSocketError.bind(this),
-      () => {
-        this.rl.prompt()
-        this.isReceivingResponse = false
-        if (this.stopResponse) {
-          this.stopResponse()
-        }
-        this.stopLoadingAnimation()
-      },
+      this.onWebSocketReconnect.bind(this),
+      this.returnControlToUser.bind(this),
       this.costMode
     )
 
@@ -156,6 +151,15 @@ export class CLI {
     })
   }
 
+  private returnControlToUser() {
+    this.rl.prompt()
+    this.isReceivingResponse = false
+    if (this.stopResponse) {
+      this.stopResponse()
+    }
+    this.stopLoadingAnimation()
+  }
+
   private onWebSocketError() {
     this.stopLoadingAnimation()
     this.isReceivingResponse = false
@@ -166,6 +170,11 @@ export class CLI {
     console.error(yellow('\nCould not connect. Retrying...'))
   }
 
+  private onWebSocketReconnect() {
+    console.log(green('\nReconnected!'))
+    this.returnControlToUser()
+  }
+
   private detectPasting() {
     const currentTime = Date.now()
     const timeDiff = currentTime - this.lastInputTime
diff --git a/npm-app/src/client.ts b/npm-app/src/client.ts
index 2d317309d..e7a14daa5 100644
--- a/npm-app/src/client.ts
+++ b/npm-app/src/client.ts
@@ -57,11 +57,16 @@ export class Client {
     websocketUrl: string,
     chatStorage: ChatStorage,
     onWebSocketError: () => void,
+    onWebSocketReconnect: () => void,
     returnControlToUser: () => void,
     costMode: CostMode
   ) {
     this.costMode = costMode
-    this.webSocket = new APIRealtimeClient(websocketUrl, onWebSocketError)
+    this.webSocket = new APIRealtimeClient(
+      websocketUrl,
+      onWebSocketError,
+      onWebSocketReconnect
+    )
     this.chatStorage = chatStorage
     this.user = this.getUser()
     this.getFingerprintId()
