diff --git a/scripts/fat-sdk-openrouter-example.ts b/scripts/fat-sdk-openrouter-example.ts
index 5f120e5f9..4c9a18dd9 100644
--- a/scripts/fat-sdk-openrouter-example.ts
+++ b/scripts/fat-sdk-openrouter-example.ts
@@ -1,16 +1,22 @@
 import { createOpenAICompatible } from '@ai-sdk/openai-compatible'
 import { websiteUrl } from '@codebuff/npm-app/config'
-import { generateText } from 'ai'
+import { generateObject } from 'ai'
+import z from 'zod/v4'
 
 const codebuffBackendProvider = createOpenAICompatible({
   name: 'codebuff',
   apiKey: '12345',
   baseURL: websiteUrl + '/api/v1',
+  // apiKey: process.env.OPEN_ROUTER_API_KEY,
+  // baseURL: 'https://openrouter.ai/api/v1',
+  supportsStructuredOutputs: true,
 })
 
-// const response = await streamText({
-const response = await generateText({
-  model: codebuffBackendProvider('anthropic/claude-sonnet-4.5'),
+const response = await generateObject({
+  schema: z.object({ greeting: z.string() }),
+  // const response = await streamText({
+  // const response = await generateText({
+  model: codebuffBackendProvider('openai/gpt-5'),
   messages: [
     {
       role: 'user',
