diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 479b3eb1c..df486e44c 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -1,7 +1,7 @@
 import { uniq } from 'lodash'
 import { applyChanges } from 'common/util/changes'
 import * as readline from 'readline'
-import { green, yellow } from 'picocolors'
+import { green, yellow, underline } from 'picocolors'
 import { parse } from 'path'
 
 import { websocketUrl } from './config'
@@ -322,7 +322,7 @@ export class CLI {
       userInputId,
       () => {
         this.stopLoadingAnimation()
-        console.log()
+        process.stdout.write(underline(green('\nManicode')) + ' ')
       }
     )
 
