diff --git a/packages/bigquery/src/client.ts b/packages/bigquery/src/client.ts
index 8bb4522bb..12878f4ee 100644
--- a/packages/bigquery/src/client.ts
+++ b/packages/bigquery/src/client.ts
@@ -331,7 +331,7 @@ export async function getTracesAndRelabelsForUser(
   })
 }
 
-interface TraceBundle {
+export interface TraceBundle {
   trace: GetRelevantFilesTrace // the base row
   relatedTraces: BaseTrace[] // the extras (runtime-typed)
   relabels: Relabel[]
diff --git a/scripts/ft-file-selection/README.md b/scripts/ft-file-selection/README.md
index cddfa8c45..5a466494d 100644
--- a/scripts/ft-file-selection/README.md
+++ b/scripts/ft-file-selection/README.md
@@ -17,7 +17,7 @@
 - Gemini Endpoint ID: 8493203957034778624
 - Finetune date: Apr 23, 2025
 - Finetune file: gemini-tune-data-007.jsonl, available in the GCS bucket
-- Finetune samples: 1934 model messages
+- Finetune samples: 1934 messages
 - Tokens: 74.6M
 - Epochs: 2
 - Base model: gemini-2.0-flash-001
@@ -35,7 +35,7 @@
 - Gemini Endpoint ID: 2589952415784501248
 - Finetune date: May 29, 2025
 - Finetune file: gemini-tune-data-027.jsonl, available in the GCS bucket
-- Finetune samples: 1077 model messages
+- Finetune samples: 1077 messages
 - Tokens: 44.7M
 - Epochs: 1
 - Base model: gemini-2.0-flash-001
@@ -46,7 +46,26 @@
   - Only used 'Key' traces AFAIK - TODO: Will this worsen non-obvious, or encourage 'Key' to pick too many?
 - Est costs:
   - ~$150 for finetuning ($3.00/1M tokens * 1 epoch * 44.7M tokens)
-  - >$660 for generating relabel inputs (at $15.00/1M input Claude tokens + some additional file context tokens)
+  - over $660 for generating relabel inputs (at $15.00/1M input Claude tokens + some additional file context tokens)
+
+`ft_filepicker_topk_001`
+- Gemini Endpoint ID: 
+- Finetune date: May 30, 2025
+- Finetune file: gemini-tune-data-039-top2.jsonl, available in the GCS bucket
+- Validation file: gemini-tune-data-039-top2-validation.jsonl, available in the GCS bucket
+- Finetune samples: 1432 messages
+- Validation samples: 119 messages
+- Tokens: 60.5M
+- Epochs: 2
+- Base model: gemini-2.0-flash-001
+- Distilled model: claude-opus-4-20250514 
+- Notes: 
+  - Trained only on the top-2 file predictions from claude-opus-4
+    - Otherwise, identical to the `ft_filepicker_007` finetune
+  - Forced maximum adapter size, and 2 epochs
+- Est costs:
+  - ~$300 for finetuning ($3.00/1M tokens * 1 epoch * 44.7M tokens)
+  - Mostly reuses relabel inputs from other Claude runs - no additional cost anticipated.
 
 
 ## Scripts
@@ -59,5 +78,7 @@ Contains a variety of scripts for inspecting and processing finetuning data from
 
 `relabel-traces.ts [--prod]` - Relabels traces in BigQuery, ie: generates new outputs using more powerful models for real production inputs.
 
+`relabel-traces-with-context.ts [--prod]` - Relabels traces using the uploaded "full file context"
+
 `collect-tuning-data.ts <model> [--prod]` - Downloads tuning data from BigQuery, using real inputs + relabeled outputs. The `model` parameter controls which relabeler model is used. 
 
diff --git a/scripts/ft-file-selection/collect-tuning-data.ts b/scripts/ft-file-selection/collect-tuning-data.ts
index 19a64e948..3f9a60776 100644
--- a/scripts/ft-file-selection/collect-tuning-data.ts
+++ b/scripts/ft-file-selection/collect-tuning-data.ts
@@ -1,6 +1,12 @@
+import { createHash } from 'crypto'
 import { existsSync, readdirSync, writeFileSync } from 'fs'
 
-import { getTracesWithRelabels, setupBigQuery } from '@codebuff/bigquery'
+import {
+  GetRelevantFilesTrace,
+  getTracesWithRelabels,
+  Relabel,
+  setupBigQuery,
+} from '@codebuff/bigquery'
 import { Message } from 'common/types/message'
 
 // Get model from command line args
@@ -9,6 +15,9 @@ const isProd = process.argv.includes('--prod')
 const DATASET = isProd ? 'codebuff_data' : 'codebuff_data_dev'
 const MAX_LENGTH_CHARS = 500_000
 
+const VALIDATION_SAMPLING_RATE = 0.1
+const SAVE_TOP_FEW_DATA = true
+
 if (!model) {
   console.log('Missing model argument')
   console.log(
@@ -48,6 +57,13 @@ function getNextAvailableFilename(
   return `${dir}/${baseFilename}-${paddedNum}.${extension}`
 }
 
+function getDeterministicSample(traceId: string): number {
+  const hash = createHash('sha256').update(traceId).digest('hex')
+  // Convert the first 8 characters of the hash to a number between 0 and 1
+  const numericalValue = parseInt(hash.substring(0, 8), 16)
+  return numericalValue / 0xffffffff
+}
+
 interface SystemMessage {
   text: string
   type: 'text'
@@ -82,6 +98,36 @@ function convertRole(role: string): 'user' | 'model' | 'system' {
   return 'user'
 }
 
+const TOP_N_FILES = 2
+function convertToTopFewTrainingExample(
+  example: GeminiTuningExample
+): GeminiTuningExample {
+  // This function should:
+  // a) in the very last message, keep the only the top 2 files, ie: the first 2 "lines" in the output
+  // b) in all other messages, find a note of format: Remember to focus on the most important files and limit your selection to {count} files
+  // c) replace {count} with 2
+  // d) return the new example
+
+  const lastMessage = example.contents[example.contents.length - 1]
+  const lastMessageContent = lastMessage.parts[0].text
+  const topNFiles = lastMessageContent.split('\n').slice(0, TOP_N_FILES)
+  lastMessage.parts[0].text = topNFiles.join('\n')
+
+  example.contents.forEach((msg) => {
+    msg.parts.forEach((part) => {
+      if (typeof part.text === 'string') {
+        const msgContentWithTopN = part.text.replace(
+          /Remember to focus on the most important files and limit your selection to (\d+)/,
+          `Remember to focus on the most important files and limit your selection to ${TOP_N_FILES}`
+        )
+        part.text = msgContentWithTopN
+      }
+    })
+  })
+
+  return example
+}
+
 function convertToGeminiFormat(
   system: SystemMessage[],
   messages: Message[],
@@ -197,81 +243,150 @@ function convertToOpenAIFormat(
   }
 }
 
-async function main() {
-  try {
-    await setupBigQuery(DATASET)
-    console.log(`Using dataset: ${DATASET}`)
+function writeTracesAsOpenAIData(
+  traces: {
+    trace: GetRelevantFilesTrace
+    relabel: Relabel
+  }[]
+) {
+  // Convert to OpenAI format
+  const openaiTuningData = traces
+    .map(({ trace, relabel }) => {
+      try {
+        return convertToOpenAIFormat(
+          trace.payload.system as SystemMessage[],
+          trace.payload.messages as Message[],
+          relabel.payload.output
+        )
+      } catch (error) {
+        console.error('Error processing trace for OpenAI:', error)
+        return null
+      }
+    })
+    .filter(Boolean)
 
-    // Get traces for the specified model from BigQuery
-    const traces = await getTracesWithRelabels(model, 10000, DATASET)
-    console.log(`Found ${traces.length} traces for model ${model}`)
+  // OpenAI gets mad if we have <10 examples, lets repeat the last example until we have 10
+  // Terrible terrible idea, but good for testing.
+  while (openaiTuningData.length < 10 && openaiTuningData.length > 0) {
+    openaiTuningData.push(openaiTuningData[openaiTuningData.length - 1])
+  }
 
-    // Process traces and convert to Gemini format
-    const tuningData = traces
-      .map(({ trace, relabel }) => {
-        try {
-          return convertToGeminiFormat(
-            trace.payload.system as SystemMessage[],
-            trace.payload.messages as Message[],
-            relabel.payload.output
-          )
-        } catch (error) {
-          console.error('Error processing trace:', error)
-          return null
-        }
-      })
-      .filter(Boolean)
+  // Save as JSONL with auto-incrementing filename
+  const openaiJsonlContentFiltered = openaiTuningData
+    .map((example) => JSON.stringify(example))
+    .filter((example) => example.length < MAX_LENGTH_CHARS)
 
-    // Save as JSONL with auto-incrementing filename
-    const jsonlContentFiltered = tuningData
-      .map((example) => JSON.stringify(example))
-      // toss messages longer than 500k chars
-      .filter((example) => example.length < MAX_LENGTH_CHARS)
+  const openaiJsonlContent = openaiJsonlContentFiltered.join('\n')
 
-    const jsonlContent = jsonlContentFiltered.join('\n')
+  const openaiPath = getNextAvailableFilename('openai-tune-data', 'jsonl')
+  writeFileSync(openaiPath, openaiJsonlContent)
 
-    const geminiPath = getNextAvailableFilename('gemini-tune-data', 'jsonl')
-    writeFileSync(geminiPath, jsonlContent)
+  console.log(
+    `Successfully saved ${openaiJsonlContentFiltered.length} examples to ${openaiPath} (filtered ${openaiTuningData.length - openaiJsonlContentFiltered.length} examples due to length)`
+  )
+}
 
-    console.log(
-      `Successfully saved ${jsonlContentFiltered.length} examples to ${geminiPath} (filtered ${tuningData.length - jsonlContentFiltered.length} examples due to length)`
-    )
+function writeGeminiTrainingAndValidationData(
+  filename: string,
+  examples: {
+    example: GeminiTuningExample
+    deterministicSample: number
+  }[]
+) {
+  const trainingData = examples.filter(
+    ({ deterministicSample }) => deterministicSample > VALIDATION_SAMPLING_RATE
+  )
+  const validationData = examples.filter(
+    ({ deterministicSample }) => deterministicSample <= VALIDATION_SAMPLING_RATE
+  )
 
-    // Convert to OpenAI format
-    const openaiTuningData = traces
-      .map(({ trace, relabel }) => {
-        try {
-          return convertToOpenAIFormat(
+  // Save as JSONL with auto-incrementing filename
+  const trainingJsonlContentFiltered = trainingData
+    .map((example) => JSON.stringify(example.example))
+    // toss messages longer than 500k chars
+    .filter((example) => example.length < MAX_LENGTH_CHARS)
+
+  const trainingJsonlContent = trainingJsonlContentFiltered.join('\n')
+
+  const validationJsonlContentFiltered = validationData
+    .map((example) => JSON.stringify(example.example))
+    .filter((example) => example.length < MAX_LENGTH_CHARS)
+
+  const validationJsonlContent = validationJsonlContentFiltered.join('\n')
+  writeFileSync(filename, trainingJsonlContent)
+  writeFileSync(
+    filename.replace('.jsonl', '-validation.jsonl'),
+    validationJsonlContent
+  )
+
+  console.log(
+    `Successfully saved ${trainingJsonlContentFiltered.length} training examples and ${validationJsonlContentFiltered.length} validation examples to ${filename}`
+  )
+}
+
+function writeTracesAsGeminiData(
+  traces: {
+    trace: GetRelevantFilesTrace
+    relabel: Relabel
+  }[]
+) {
+  const tuningData = traces
+    .map(({ trace, relabel }) => {
+      try {
+        return {
+          example: convertToGeminiFormat(
             trace.payload.system as SystemMessage[],
             trace.payload.messages as Message[],
             relabel.payload.output
-          )
-        } catch (error) {
-          console.error('Error processing trace for OpenAI:', error)
-          return null
+          ),
+          deterministicSample: getDeterministicSample(trace.id),
         }
-      })
-      .filter(Boolean)
-
-    // OpenAI gets mad if we have <10 examples, lets repeat the last example until we have 10
-    // Terrible terrible idea, but good for testing.
-    while (openaiTuningData.length < 10 && openaiTuningData.length > 0) {
-      openaiTuningData.push(openaiTuningData[openaiTuningData.length - 1])
-    }
+      } catch (error) {
+        console.error('Error processing trace:', error)
+        return null
+      }
+    })
+    .filter(Boolean) as {
+    example: GeminiTuningExample
+    deterministicSample: number
+  }[]
+
+  const geminiPath = getNextAvailableFilename('gemini-tune-data', 'jsonl')
+  writeGeminiTrainingAndValidationData(geminiPath, tuningData)
+
+  if (SAVE_TOP_FEW_DATA) {
+    // Convert to top few training examples
+    const topFewTrainingData = tuningData.map(
+      ({ example, deterministicSample }) => {
+        const topFewExample = convertToTopFewTrainingExample(example)
+        return {
+          example: topFewExample,
+          deterministicSample,
+        }
+      }
+    )
+    const topFewTrainingDataPath = geminiPath.replace('.jsonl', '-top2.jsonl')
+    writeGeminiTrainingAndValidationData(
+      topFewTrainingDataPath,
+      topFewTrainingData
+    )
+  }
+}
 
-    // Save as JSONL with auto-incrementing filename
-    const openaiJsonlContentFiltered = openaiTuningData
-      .map((example) => JSON.stringify(example))
-      .filter((example) => example.length < MAX_LENGTH_CHARS)
+async function main() {
+  try {
+    await setupBigQuery(DATASET)
+    console.log(`Using dataset: ${DATASET}`)
 
-    const openaiJsonlContent = openaiJsonlContentFiltered.join('\n')
+    // Get traces for the specified model from BigQuery
+    const traces = await getTracesWithRelabels(model, 10000, DATASET)
+    console.log(`Found ${traces.length} traces for model ${model}`)
 
-    const openaiPath = getNextAvailableFilename('openai-tune-data', 'jsonl')
-    writeFileSync(openaiPath, openaiJsonlContent)
+    // Process traces and convert to Gemini format
+    writeTracesAsGeminiData(traces)
 
-    console.log(
-      `Successfully saved ${openaiJsonlContentFiltered.length} examples to ${openaiPath} (filtered ${openaiTuningData.length - openaiJsonlContentFiltered.length} examples due to length)`
-    )
+    // write traces as OpenAI data
+    // writeTracesAsOpenAIData(traces)
   } catch (error) {
     console.error('Error:', error)
     process.exit(1)
diff --git a/scripts/ft-file-selection/relabel-traces-with-context.ts b/scripts/ft-file-selection/relabel-traces-with-context.ts
index e6a3849a5..3fa05eb05 100644
--- a/scripts/ft-file-selection/relabel-traces-with-context.ts
+++ b/scripts/ft-file-selection/relabel-traces-with-context.ts
@@ -13,8 +13,9 @@ const MODEL_TO_TEST = claudeModels.opus4
 const isProd = process.argv.includes('--prod')
 const DATASET = isProd ? 'codebuff_data' : 'codebuff_data_dev'
 const MAX_PARALLEL = 2 // Maximum number of traces to process in parallel
-const LIMIT = 2000
-const PAGE_SIZE = 50
+const LIMIT = 3000
+const PAGE_SIZE = 30
+const START_CURSOR = '2025-05-27T00:00:00.000Z' // only helpful if you quit midway thru
 
 async function runTraces() {
   await setupBigQuery(DATASET)
@@ -22,7 +23,7 @@ async function runTraces() {
     console.log(
       `\nProcessing traces for model ${MODEL_TO_TEST} with full file context...`
     )
-    let cursor: string | undefined = undefined
+    let cursor: string | undefined = START_CURSOR
 
     for (let i = 0; i < LIMIT; i += PAGE_SIZE) {
       // Get traces and all related data
