diff --git a/backend/src/run-agent-step.ts b/backend/src/run-agent-step.ts
index 843e59589..886b9395a 100644
--- a/backend/src/run-agent-step.ts
+++ b/backend/src/run-agent-step.ts
@@ -366,6 +366,7 @@ export const runAgentStep = async (
       shouldEndTurn,
       duration: Date.now() - startTime,
       fullResponse,
+      fullResponseChunks,
       finalMessageHistoryWithToolResults: agentState.messageHistory,
       toolCalls,
       toolResults,
