diff --git a/agents/researcher/researcher-docs.ts b/agents/researcher/researcher-docs.ts
index 659db4703..89b817d4f 100644
--- a/agents/researcher/researcher-docs.ts
+++ b/agents/researcher/researcher-docs.ts
@@ -4,7 +4,7 @@ import { publisher } from '../constants'
 const definition: SecretAgentDefinition = {
   id: 'researcher-docs',
   publisher,
-  model: 'x-ai/grok-4-fast',
+  model: 'x-ai/grok-4.1-fast',
   displayName: 'Doc',
   spawnerPrompt: `Expert at reading technical documentation of major public libraries and frameworks to find relevant information. (e.g. React, MongoDB, Postgres, etc.)`,
   inputSchema: {
diff --git a/agents/researcher/researcher-web.ts b/agents/researcher/researcher-web.ts
index 2705d24c9..08f53d830 100644
--- a/agents/researcher/researcher-web.ts
+++ b/agents/researcher/researcher-web.ts
@@ -5,7 +5,7 @@ import { publisher } from '../constants'
 const definition: SecretAgentDefinition = {
   id: 'researcher-web',
   publisher,
-  model: 'x-ai/grok-4-fast',
+  model: 'x-ai/grok-4.1-fast',
   displayName: 'Weeb',
   spawnerPrompt: `Browses the web to find relevant information.`,
   inputSchema: {
