diff --git a/web/src/app/page.tsx b/web/src/app/page.tsx
index 60701c28d..65cc8c746 100644
--- a/web/src/app/page.tsx
+++ b/web/src/app/page.tsx
@@ -9,6 +9,7 @@ import {
   Play,
   ChevronDown,
   ChevronUp,
+  ExternalLink,
 } from 'lucide-react'
 import Link from 'next/link'
 import { useTheme } from 'next-themes'
@@ -16,13 +17,54 @@ import { Button } from '@/components/ui/button'
 import { Dialog, DialogContent } from '@/components/ui/dialog'
 import { BackgroundBeams } from '@/components/ui/background-beams'
 import { useToast } from '@/components/ui/use-toast'
-import { Card, CardContent } from '@/components/ui/card'
 import Marquee from '@/components/ui/marquee'
-import { testimonials } from '@/lib/testimonials'
+import { Testimonial, testimonials } from '@/lib/testimonials'
 import { faqs } from '@/lib/faq'
 import { cn } from '@/lib/utils'
 import { storeSearchParams } from '@/lib/trackConversions'
 
+const ReviewCard = ({ t }: { t: Testimonial }) => {
+  return (
+    <figure
+      className={cn(
+        'relative w-64 lg:w-80 cursor-pointer overflow-hidden rounded-xl border p-6',
+        // light styles
+        'border-gray-950/[.1] bg-gray-950/[.01] hover:bg-gray-950/[.05]',
+        // dark styles
+        'dark:border-gray-50/[.1] dark:bg-gray-50/[.10] dark:hover:bg-gray-50/[.15]'
+      )}
+    >
+      <div className="flex justify-between">
+        <div className="flex flex-row items-center gap-2">
+          <img
+            className="rounded-full"
+            width="32"
+            height="32"
+            alt=""
+            src={
+              t.avatar ??
+              `https://avatar.vercel.sh/${t.author.split(' ').join('-').toLowerCase()}?size=32`
+            }
+          />
+          <div className="flex flex-col">
+            <figcaption className="text-sm font-medium dark:text-white">
+              {t.author}
+            </figcaption>
+            <p className="text-xs font-medium dark:text-white/40">{t.title}</p>
+          </div>
+        </div>
+        <div className="flex items-center gap-2">
+          <ExternalLink
+            className="h-4 w-4"
+            onClick={() => window.open(t.link)}
+          />
+        </div>
+      </div>
+      <blockquote className="mt-4 text-sm lg:text-base">{t.quote}</blockquote>
+    </figure>
+  )
+}
+
 const Home = () => {
   const { theme } = useTheme()
   const { toast } = useToast()
@@ -198,34 +240,7 @@ const Home = () => {
             >
               <div className="flex gap-4">
                 {row.map((testimonial, i) => (
-                  <Card
-                    key={i}
-                    className={cn(
-                      'w-[350px] shrink-0',
-                      testimonial.link &&
-                        'cursor-pointer hover:scale-105 transition-transform'
-                    )}
-                    onClick={() =>
-                      testimonial.link &&
-                      window.open(testimonial.link, '_blank')
-                    }
-                  >
-                    <CardContent className="pt-6">
-                      <p className="mb-4">{testimonial.quote}</p>
-                      <div className="flex items-center gap-2">
-                        {testimonial.avatar && (
-                          <img
-                            src={testimonial.avatar}
-                            alt={testimonial.author}
-                            className="w-8 h-8 rounded-full object-cover"
-                          />
-                        )}
-                        <p className="font-semibold">
-                          - {testimonial.author}, {testimonial.title}
-                        </p>
-                      </div>
-                    </CardContent>
-                  </Card>
+                  <ReviewCard key={i} t={testimonial} />
                 ))}
               </div>
             </Marquee>
diff --git a/web/src/lib/testimonials.ts b/web/src/lib/testimonials.ts
index 2c02b9a2e..a7c659273 100644
--- a/web/src/lib/testimonials.ts
+++ b/web/src/lib/testimonials.ts
@@ -8,17 +8,10 @@ export type Testimonial = {
 
 export const testimonials: Testimonial[][] = [
   [
-    {
-      quote:
-        "I played around with Codebuff and added some features to something I was working on. It really does have a different feeling than any other AI tools I've used; feels much more right, and I'm impressed by how you managed to land on that when nobody else did.",
-      author: 'JJ Fliegelman',
-      title: 'Founder',
-      link: '/testimonials/proof/jj-fliegelman.png',
-    },
     {
       quote:
         "As a Ruby on Rails dev this was a challenge as client want the solution in Python and Manicode has saved my skin big time as l've never used Python. So kudos.",
-      author: 'Anonymous',
+      author: 'anonymous',
       title: 'Developer',
       link: '/testimonials/proof/ruby-on-rails.png',
     },
@@ -51,6 +44,15 @@ export const testimonials: Testimonial[][] = [
       avatar: '/testimonials/stevo.png',
       link: '/testimonials/proof/stevo.png',
     },
+    {
+      quote:
+        "For those who use (n)vim or other terminal-based editors, some friends of mine built @ManicodeAI and I've really been enjoying it. I can give it coding tasks right from my terminal and it just takes care of them for me...",
+      author: 'Dennis Beatty',
+      title: 'Founder & CEO',
+      avatar:
+        'https://pbs.twimg.com/profile_images/943341063502286848/2h_xKTs9_400x400.jpg',
+      link: 'https://x.com/dnsbty/status/1849612125972594929',
+    },
     {
       quote:
         'Just did a complete structural refactoring that would have took 4-8 hours by a human in 30 minutes using Claude (Web) to drive Codebuff to finish line. I think research in AI+AI pair programming is a must. ',
@@ -61,6 +63,13 @@ export const testimonials: Testimonial[][] = [
     },
   ],
   [
+    {
+      quote:
+        "I played around with Codebuff and added some features to something I was working on. It really does have a different feeling than any other AI tools I've used; feels much more right, and I'm impressed by how you managed to land on that when nobody else did.",
+      author: 'JJ Fliegelman',
+      title: 'Founder',
+      link: '/testimonials/proof/jj-fliegelman.png',
+    },
     {
       quote: "I finally tried composer. It's ass compared to manicode",
       author: 'anonymous',
@@ -99,15 +108,6 @@ export const testimonials: Testimonial[][] = [
       avatar: '/testimonials/shardool-patel.jpg',
       link: '/testimonials/proof/shardool-patel.png',
     },
-    {
-      quote:
-        "For those who use (n)vim or other terminal-based editors, some friends of mine built @ManicodeAI and I've really been enjoying it. I can give it coding tasks right from my terminal and it just takes care of them for me...",
-      author: 'Dennis Beatty',
-      title: 'Founder & CEO',
-      avatar:
-        'https://pbs.twimg.com/profile_images/943341063502286848/2h_xKTs9_400x400.jpg',
-      link: 'https://x.com/dnsbty/status/1849612125972594929',
-    },
     {
       quote:
         'when investors ask me about codebuff I tell them i use it 6 days a week',
