diff --git a/web/src/content/case-studies/benjamin-bryant.mdx b/web/src/content/case-studies/benjamin-bryant.mdx
index df8c7eecf..40de5b066 100644
--- a/web/src/content/case-studies/benjamin-bryant.mdx
+++ b/web/src/content/case-studies/benjamin-bryant.mdx
@@ -13,7 +13,7 @@ Benjamin Bryant runs an early-stage investment fund that focuses on Y Combinator
 In his day to day, he switches between different roles rapidly, going from coding to investor meetings and founder pitches. Before Codebuff, each new software feature would often take at least an afternoon to implement, with more complex one requiring several days.
 
 ## Enter Codebuff 
-Codebuff helped Ben reduce the implementation time down to minutes, if not hours.
+Codebuff helped Ben reduce implementation time down to minutes, not hours.
 
 > In 10 minutes, Codebuff will get me most of the way there... It's like having an extension of your brain.
 
