diff --git a/backend/src/find-files/request-files-prompt.ts b/backend/src/find-files/request-files-prompt.ts
index 8b7443cb3..288c08a5d 100644
--- a/backend/src/find-files/request-files-prompt.ts
+++ b/backend/src/find-files/request-files-prompt.ts
@@ -1,24 +1,20 @@
 import { dirname, isAbsolute, normalize } from 'path'
 
 import { TextBlockParam } from '@anthropic-ai/sdk/resources'
-import {
-  GetExpandedFileContextForTrainingTrace,
-  GetRelevantFilesTrace,
-  insertTrace,
-} from '@codebuff/bigquery'
+import { GetRelevantFilesTrace, insertTrace } from '@codebuff/bigquery'
 import { models, type CostMode } from 'common/constants'
 import { getAllFilePaths } from 'common/project-file-tree'
 import { Message } from 'common/types/message'
 import { filterDefined } from 'common/util/array'
 import {
+  ProjectFileContext,
   cleanMarkdownCodeBlock,
   createMarkdownFileBlock,
-  ProjectFileContext,
 } from 'common/util/file'
 import { range, shuffle, uniq } from 'lodash'
 import { WebSocket } from 'ws'
 
-import { promptClaude, System } from '../llm-apis/claude'
+import { System } from '../llm-apis/claude'
 import { logger } from '../util/logger'
 import { countTokens } from '../util/token-counter'
 import { requestFiles } from '../websockets/websocket-action'
@@ -168,62 +164,6 @@ export async function requestRelevantFiles(
   return firstPassFiles.slice(0, MAX_FILES_PER_REQUEST)
 }
 
-export async function requestRelevantFilesForTraining(
-  {
-    messages,
-    system,
-  }: {
-    messages: Message[]
-    system: string | Array<TextBlockParam>
-  },
-  fileContext: ProjectFileContext,
-  assistantPrompt: string | null,
-  agentStepId: string,
-  clientSessionId: string,
-  fingerprintId: string,
-  userInputId: string,
-  userId: string | undefined,
-  costMode: CostMode
-) {
-  const COUNT = 50
-
-  const lastMessage = messages[messages.length - 1]
-  const messagesExcludingLastIfByUser =
-    lastMessage.role === 'user' ? messages.slice(0, -1) : messages
-  const userPrompt =
-    lastMessage.role === 'user'
-      ? typeof lastMessage.content === 'string'
-        ? lastMessage.content
-        : JSON.stringify(lastMessage.content)
-      : ''
-
-  const nonObviousPrompt = generateNonObviousRequestFilesPrompt(
-    userPrompt,
-    assistantPrompt,
-    fileContext,
-    COUNT
-  )
-
-  const nonObviousFiles = await getRelevantFilesForTraining(
-    {
-      messages: messagesExcludingLastIfByUser,
-      system,
-    },
-    nonObviousPrompt,
-    'Non-Obvious',
-    agentStepId,
-    clientSessionId,
-    fingerprintId,
-    userInputId,
-    userId,
-    costMode
-  )
-
-  const candidateFiles = nonObviousFiles
-
-  return validateFilePaths(uniq(candidateFiles.files))
-}
-
 async function getRelevantFiles(
   {
     messages,
@@ -293,75 +233,6 @@ async function getRelevantFiles(
   return { files, duration, requestType, response }
 }
 
-async function getRelevantFilesForTraining(
-  {
-    messages,
-    system,
-  }: {
-    messages: Message[]
-    system: string | Array<TextBlockParam>
-  },
-  userPrompt: string,
-  requestType: string,
-  agentStepId: string,
-  clientSessionId: string,
-  fingerprintId: string,
-  userInputId: string,
-  userId: string | undefined,
-  costMode: CostMode
-) {
-  const bufferTokens = 100_000
-  const messagesWithPrompt = getMessagesSubset(
-    [
-      ...messages,
-      {
-        role: 'user' as const,
-        content: userPrompt,
-      },
-    ],
-    bufferTokens
-  )
-  const start = performance.now()
-  let response = await promptClaude(messagesWithPrompt, {
-    system,
-    clientSessionId,
-    fingerprintId,
-    userInputId,
-    model: models.sonnet,
-    userId,
-  })
-
-  const end = performance.now()
-  const duration = end - start
-
-  const files = validateFilePaths(response.split('\n'))
-
-  const trace: GetExpandedFileContextForTrainingTrace = {
-    id: crypto.randomUUID(),
-    agent_step_id: agentStepId,
-    user_id: userId ?? '',
-    created_at: new Date(),
-    type: 'get-expanded-file-context-for-training',
-    payload: {
-      messages: messagesWithPrompt,
-      system,
-      output: response,
-      request_type: requestType,
-      cost_mode: costMode,
-      user_input_id: userInputId,
-      client_session_id: clientSessionId,
-      fingerprint_id: fingerprintId,
-      model: models.ft_filepicker_005,
-    },
-  }
-
-  insertTrace(trace).catch((error: Error) => {
-    logger.error({ error }, 'Failed to insert trace')
-  })
-
-  return { files, duration, requestType, response }
-}
-
 function topLevelDirectories(fileContext: ProjectFileContext) {
   const { fileTree } = fileContext
   return fileTree
diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 395f82c22..57c36bfde 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -1,9 +1,5 @@
 import { TextBlockParam } from '@anthropic-ai/sdk/resources'
-import {
-  AgentResponseTrace,
-  GetExpandedFileContextForTrainingBlobTrace,
-  insertTrace,
-} from '@codebuff/bigquery'
+import { AgentResponseTrace, insertTrace } from '@codebuff/bigquery'
 import { ClientAction } from 'common/actions'
 import {
   HIDDEN_FILE_READ_STATUS,
@@ -24,10 +20,7 @@ import { difference, partition, uniq } from 'lodash'
 import { WebSocket } from 'ws'
 
 import { checkTerminalCommand } from './check-terminal-command'
-import {
-  requestRelevantFiles,
-  requestRelevantFilesForTraining,
-} from './find-files/request-files-prompt'
+import { requestRelevantFiles } from './find-files/request-files-prompt'
 import { getDocumentationForQuery } from './get-documentation-for-query'
 import { processFileBlock } from './process-file-block'
 import { processStrReplace } from './process-str-replace'
@@ -1024,24 +1017,6 @@ async function getFileReadingUpdates(
       )) ??
       []
 
-  uploadExpandedFileContextForTraining(
-    ws,
-    { messages, system },
-    fileContext,
-    prompt,
-    agentStepId,
-    clientSessionId,
-    fingerprintId,
-    userInputId,
-    userId,
-    costMode
-  ).catch((error) => {
-    logger.error(
-      { error },
-      'Error uploading expanded file context for training'
-    )
-  })
-
   const isFirstRead = previousFileList.length === 0
   const initialFiles = getInitialFiles(fileContext)
   const includedInitialFiles = isFirstRead
@@ -1164,65 +1139,3 @@ function getPrintedPaths(
       )
   )
 }
-
-async function uploadExpandedFileContextForTraining(
-  ws: WebSocket,
-  {
-    messages,
-    system,
-  }: {
-    messages: Message[]
-    system: string | Array<TextBlockParam>
-  },
-  fileContext: ProjectFileContext,
-  assistantPrompt: string | null,
-  agentStepId: string,
-  clientSessionId: string,
-  fingerprintId: string,
-  userInputId: string,
-  userId: string | undefined,
-  costMode: CostMode
-) {
-  const files = await requestRelevantFilesForTraining(
-    { messages, system },
-    fileContext,
-    assistantPrompt,
-    agentStepId,
-    clientSessionId,
-    fingerprintId,
-    userInputId,
-    userId,
-    costMode
-  )
-
-  const loadedFiles = await requestFiles(ws, files)
-
-  // Upload a map of:
-  // {file_path: {content, token_count}}
-  // up to 50k tokens
-  const filesToUpload: Record<string, { content: string; tokens: number }> = {}
-  for (const file of files) {
-    const tokens = countTokens(loadedFiles[file]!)
-    if (tokens > 50000) {
-      break
-    }
-    filesToUpload[file] = { content: loadedFiles[file]!, tokens }
-  }
-
-  const trace: GetExpandedFileContextForTrainingBlobTrace = {
-    type: 'get-expanded-file-context-for-training-blobs',
-    created_at: new Date(),
-    id: crypto.randomUUID(),
-    agent_step_id: agentStepId,
-    user_id: userId ?? '',
-    payload: {
-      files: filesToUpload,
-      user_input_id: userInputId,
-      client_session_id: clientSessionId,
-      fingerprint_id: fingerprintId,
-    },
-  }
-
-  // Upload the files to bigquery
-  await insertTrace(trace)
-}
diff --git a/packages/bigquery/src/schema.ts b/packages/bigquery/src/schema.ts
index e850b2240..70f09b02c 100644
--- a/packages/bigquery/src/schema.ts
+++ b/packages/bigquery/src/schema.ts
@@ -13,12 +13,7 @@ interface BasePayload {
 }
 
 // Define possible trace types
-export type TraceType =
-  | 'get-relevant-files' // Get-relevant-files call in production
-  | 'file-trees' // Uploads additional file trees of various sizes
-  | 'agent-response' // Response from the agent at the end
-  | 'get-expanded-file-context-for-training' // Additional capture of a stronger-model + higher-file limit call
-  | 'get-expanded-file-context-for-training-blobs' // Capture of full file context from get-relevant-files-for-training call
+export type TraceType = 'get-relevant-files' | 'file-trees' | 'agent-response'
 
 // Base trace interface
 export interface BaseTrace extends BaseEvent {
@@ -42,20 +37,6 @@ export interface GetRelevantFilesTrace extends BaseTrace {
   payload: GetRelevantFilesPayload
 }
 
-export interface GetExpandedFileContextForTrainingTrace extends BaseTrace {
-  type: 'get-expanded-file-context-for-training'
-  payload: GetRelevantFilesPayload
-}
-
-export interface FilesBlobPayload extends BasePayload {
-  files: Record<string, { content: string; tokens: number }>
-}
-
-export interface GetExpandedFileContextForTrainingBlobTrace extends BaseTrace {
-  type: 'get-expanded-file-context-for-training-blobs'
-  payload: FilesBlobPayload
-}
-
 interface FileTreePayload extends BasePayload {
   filetrees: Record<number, string>
 }
@@ -75,12 +56,7 @@ export interface AgentResponseTrace extends BaseTrace {
 }
 
 // Union type for all trace records
-export type Trace =
-  | GetRelevantFilesTrace
-  | FileTreeTrace
-  | AgentResponseTrace
-  | GetExpandedFileContextForTrainingBlobTrace
-  | GetExpandedFileContextForTrainingTrace
+export type Trace = GetRelevantFilesTrace | FileTreeTrace | AgentResponseTrace
 
 export const TRACES_SCHEMA: TableSchema = {
   fields: [
@@ -112,4 +88,4 @@ export const RELABELS_SCHEMA: TableSchema = {
     { name: 'model', type: 'STRING', mode: 'REQUIRED' },
     { name: 'payload', type: 'JSON', mode: 'REQUIRED' },
   ],
-}
+}
\ No newline at end of file
