diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index ab1ab8bdf..575eeb325 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -416,15 +416,15 @@ export const mainPrompt = async (
 
   const relevantDocumentation = await relevantDocumentationPromise
 
+  const hasAssistantMessage = messageHistory.some((m) => m.role === 'assistant')
   const messagesWithUserMessage = buildArray(
-    ...messageHistory.filter(
-      (m) =>
-        costMode !== 'experimental' ||
+    ...messageHistory.filter((m) => {
+      return (
         !prompt ||
         typeof m.content !== 'string' ||
         !isSystemInstruction(m.content)
-    ),
-
+      )
+    }),
     !prompt && {
       role: 'user' as const,
       content: asSystemInstruction(
@@ -437,6 +437,14 @@ export const mainPrompt = async (
       content: asSystemMessage(renderToolResults(toolResults)),
     },
 
+    false &&
+      hasAssistantMessage && {
+        role: 'user' as const,
+        content: asSystemInstruction(
+          "All <previous_assistant_message>messages</previous_assistant_message> were from some previous assistant. Your task is to identify any mistakes the previous assistant has made or if they have gone off track. Reroute the conversation back toward the user request, correct the previous assistant's mistakes, identify potential issues in the code, etc.\nSeamlessly continue the conversation as if you are the same assistant, because that is what the user sees. e.g. when correcting the previous assistant, use language as if you were correcting yourself.\nIf you cannot identify any mistakes, that's great! Continue the conversation as if you are the same assistant."
+        ),
+      },
+
     // Add in new copy of agent context.
     prompt &&
       agentContext && {
@@ -485,9 +493,7 @@ export const mainPrompt = async (
     ...readFileMessages
   )
 
-  const iterationNum = messagesWithUserMessage.filter(
-    (m) => m.role === 'assistant'
-  ).length
+  const iterationNum = messagesWithUserMessage.length
 
   const system = getAgentSystemPrompt(fileContext)
   const systemTokens = countTokensJson(system)
diff --git a/backend/src/util/messages.ts b/backend/src/util/messages.ts
index 8c11fd899..7847f8e06 100644
--- a/backend/src/util/messages.ts
+++ b/backend/src/util/messages.ts
@@ -47,6 +47,30 @@ export function getMessageText(message: Message): string | undefined {
   return message.content.map((c) => ('text' in c ? c.text : '')).join('\n')
 }
 
+export function castAssistantMessage(message: Message): Message {
+  if (message.role !== 'assistant') {
+    return message
+  }
+  if (typeof message.content === 'string') {
+    return {
+      content: `<previous_assistant_message>${message.content}</previous_assistant_message>`,
+      role: 'user' as const,
+    }
+  }
+  return {
+    role: 'user' as const,
+    content: message.content.map((m) => {
+      if (m.type === 'text') {
+        return {
+          ...m,
+          text: `<previous_assistant_message>${m.text}</previous_assistant_message>`,
+        }
+      }
+      return m
+    }),
+  }
+}
+
 // Number of terminal command outputs to keep in full form before simplifying
 const numTerminalCommandsToKeep = 5
 
@@ -169,9 +193,22 @@ export function getMessagesSubset(messages: Message[], otherTokens: number) {
     otherTokens
   )
 
-  // Remove cache_control from all messages
-  for (const message of messagesSubset) {
-    if (typeof message.content === 'object' && message.content.length > 0) {
+  // Remove cache_control from all messages except the last one
+  let lastCachedMessageIndex = -1
+  for (const [index, message] of messagesSubset.entries()) {
+    if (
+      typeof message.content === 'object' &&
+      message.content[message.content.length - 1]?.cache_control
+    ) {
+      lastCachedMessageIndex = index
+    }
+  }
+  for (const [index, message] of messagesSubset.entries()) {
+    if (
+      typeof message.content === 'object' &&
+      message.content.length > 0 &&
+      index !== lastCachedMessageIndex
+    ) {
       delete message.content[message.content.length - 1].cache_control
     }
   }
