diff --git a/npm-app/src/utils/tool-renderers.ts b/npm-app/src/utils/tool-renderers.ts
index ab2dc6b94..de900b522 100644
--- a/npm-app/src/utils/tool-renderers.ts
+++ b/npm-app/src/utils/tool-renderers.ts
@@ -60,12 +60,17 @@ export const toolRenderers: Record<ToolName, ToolCallRenderer> = {
   code_search: {
     // Don't render anything
   },
-  read_files: {
-    // Don't render anything
-  },
   end_turn: {
     // Don't render anything
   },
+  read_files: {
+    ...defaultToolCallRenderer,
+    onParamChunk: (content, paramName, toolName) => {
+      return null
+    },
+
+    onParamEnd: (paramName, toolName, content) => content.trim(),
+  },
   think_deeply: {
     ...defaultToolCallRenderer,
   },
