diff --git a/backend/src/claude.ts b/backend/src/claude.ts
index 1c0e8e37b..8a02497b2 100644
--- a/backend/src/claude.ts
+++ b/backend/src/claude.ts
@@ -133,7 +133,7 @@ export const promptClaudeStream = async function* (
     }
 
     // End of turn
-    if (type === 'message_delta' && chunk.delta.stop_reason === 'end_turn') {
+    if (type === 'message_delta' && 'usage' in chunk) {
       if (!messageId) {
         console.error('No messageId found')
         break
diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 73219159b..286ed71cb 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -163,7 +163,6 @@ ${STOP_MARKER}
           return `<edit_file path="${path}">`
         },
         onTagEnd: (fileContent, { path }) => {
-          console.log('onTagEnd', { path, fileContent })
           const filePathWithoutStartNewline = fileContent.startsWith('\n')
             ? fileContent.slice(1)
             : fileContent
diff --git a/backend/src/request-files-prompt.ts b/backend/src/request-files-prompt.ts
index ed3c1004e..7930d8690 100644
--- a/backend/src/request-files-prompt.ts
+++ b/backend/src/request-files-prompt.ts
@@ -400,7 +400,7 @@ export const warmCacheForRequestRelevantFiles = async (
   clientSessionId: string,
   fingerprintId: string,
   userInputId: string,
-  userId?: string
+  userId: string | undefined
 ) => {
   await promptClaude(
     [
