diff --git a/backend/src/websockets/middleware.ts b/backend/src/websockets/middleware.ts
index 930c20955..a4b34163c 100644
--- a/backend/src/websockets/middleware.ts
+++ b/backend/src/websockets/middleware.ts
@@ -46,7 +46,7 @@ export class WebSocketMiddleware {
     for (const middleware of this.middlewares) {
       const actionOrContinue = await middleware(action, clientSessionId, ws)
       if (actionOrContinue) {
-        logger.error('Middleware execution halted:', actionOrContinue)
+        logger.warn('Middleware execution halted:', actionOrContinue)
         sendAction(ws, actionOrContinue)
         return false
       }
@@ -112,7 +112,7 @@ protec.use(async (action, clientSessionId, ws) => {
         }
 
         if (quota.quotaExceeded) {
-          logger.error(`Quota exceeded for user ${quota.userId}`)
+          logger.warn(`Quota exceeded for user ${quota.userId}`)
           return genUsageResponse(clientSessionId, fingerprintId, quota.userId)
         }
         return
@@ -161,7 +161,7 @@ protec.use(async (action, clientSessionId, ws) => {
         }
 
         if (quota.quotaExceeded) {
-          logger.error(`Quota exceeded for fingerprint ${fingerprintId}`)
+          logger.warn(`Quota exceeded for fingerprint ${fingerprintId}`)
           return genUsageResponse(clientSessionId, fingerprintId)
         }
 
