diff --git a/backend/src/generate-diffs-prompt.ts b/backend/src/generate-diffs-prompt.ts
index 7623b0df8..f9a113be4 100644
--- a/backend/src/generate-diffs-prompt.ts
+++ b/backend/src/generate-diffs-prompt.ts
@@ -130,15 +130,24 @@ export const tryToDoStringReplacementWithExtraIndentation = (params: {
   return null
 }
 
-export async function retryDiffBlocksPrompt(
-  filePath: string,
-  oldContent: string,
-  clientSessionId: string,
-  fingerprintId: string,
-  userInputId: string,
-  userId: string | undefined,
-  diffBlocksThatDidntMatch: { searchContent: string; replaceContent: string }[],
-) {
+export async function retryDiffBlocksPrompt(params: {
+  filePath: string
+  oldContent: string
+  clientSessionId: string
+  fingerprintId: string
+  userInputId: string
+  userId: string | undefined
+  diffBlocksThatDidntMatch: { searchContent: string; replaceContent: string }[]
+}) {
+  const {
+    filePath,
+    oldContent,
+    clientSessionId,
+    fingerprintId,
+    userInputId,
+    userId,
+    diffBlocksThatDidntMatch,
+  } = params
   const newPrompt =
     `The assistant failed to find a match for the following changes. Please help the assistant understand what the changes should be.
 
diff --git a/backend/src/process-file-block.ts b/backend/src/process-file-block.ts
index 9822462ce..5200b658a 100644
--- a/backend/src/process-file-block.ts
+++ b/backend/src/process-file-block.ts
@@ -311,15 +311,15 @@ Please output just the SEARCH/REPLACE blocks like this:
     )
 
     const { newDiffBlocks, newDiffBlocksThatDidntMatch } =
-      await retryDiffBlocksPrompt(
+      await retryDiffBlocksPrompt({
         filePath,
-        updatedContent,
+        oldContent: updatedContent,
         clientSessionId,
         fingerprintId,
         userInputId,
         userId,
         diffBlocksThatDidntMatch,
-      )
+      })
 
     if (newDiffBlocksThatDidntMatch.length > 0) {
       logger.error(
