diff --git a/npm-app/package.json b/npm-app/package.json
index 6525339a8..fc0401bdb 100644
--- a/npm-app/package.json
+++ b/npm-app/package.json
@@ -41,7 +41,6 @@
     "@types/diff": "5.2.1",
     "@vscode/ripgrep": "1.15.9",
     "axios": "1.7.4",
-    "cheerio": "1.0.0",
     "diff": "5.2.0",
     "env-cmd": "^10.1.0",
     "ignore": "5.3.2",
diff --git a/npm-app/src/web-scraper.ts b/npm-app/src/web-scraper.ts
index 85d498000..bacfe64ba 100644
--- a/npm-app/src/web-scraper.ts
+++ b/npm-app/src/web-scraper.ts
@@ -1,5 +1,4 @@
 import axios from 'axios'
-import * as cheerio from 'cheerio'
 
 // Global cache for scraped web pages
 const scrapedPagesCache: Record<string, string> = {}
@@ -11,15 +10,14 @@ export async function scrapeWebPage(url: string) {
   }
 
   try {
-    // const response = await axios.get(url)
-    const response = await axios.get(`https://r.jina.ai/${url}`)
-    const content = response.data
-    // const html = response.data
-    // const $ = cheerio.load(html)
-
-    // // Extract the main content (you may need to adjust this selector based on the target websites)
-    // const content = $('body').text()
-
+    let content = ''
+    if (url.startsWith('https://raw.githubusercontent.com/')) {
+      const response = await axios.get(url)
+      content = response.data
+    } else {
+      const response = await axios.get(`https://r.jina.ai/${url}`)
+      content = response.data
+    }
     // Store the scraped content in the cache
     scrapedPagesCache[url] = content
 
