diff --git a/packages/internal/src/env.ts b/packages/internal/src/env.ts
index 30dd37ab5..c79eae0b9 100644
--- a/packages/internal/src/env.ts
+++ b/packages/internal/src/env.ts
@@ -14,6 +14,7 @@ const envSchema = {
     HELICONE_API_KEY: z.string().min(1),
     OPEN_AI_KEY: z.string().min(1),
     GEMINI_API_KEY: z.string().min(1),
+    GOOGLE_GENERATIVE_AI_API_KEY: z.string().min(1),
     DEEPSEEK_API_KEY: z.string().min(1),
     OPEN_ROUTER_API_KEY: z.string().min(1),
     RELACE_API_KEY: z.string().min(1),
@@ -58,6 +59,7 @@ const envSchema = {
     HELICONE_API_KEY: process.env.HELICONE_API_KEY,
     OPEN_AI_KEY: process.env.OPEN_AI_KEY,
     GEMINI_API_KEY: process.env.GEMINI_API_KEY,
+    GOOGLE_GENERATIVE_AI_API_KEY: process.env.GOOGLE_GENERATIVE_AI_API_KEY,
     DEEPSEEK_API_KEY: process.env.DEEPSEEK_API_KEY,
     OPEN_ROUTER_API_KEY: process.env.OPEN_ROUTER_API_KEY,
     RELACE_API_KEY: process.env.RELACE_API_KEY,
