diff --git a/common/src/constants/analytics-events.ts b/common/src/constants/analytics-events.ts
index 2150acc4b..b8a35a87c 100644
--- a/common/src/constants/analytics-events.ts
+++ b/common/src/constants/analytics-events.ts
@@ -9,6 +9,7 @@ export enum AnalyticsEvent {
   BACKGROUND_PROCESS_LEFTOVER_DETECTED = 'cli.background_process_leftover_detected',
   BACKGROUND_PROCESS_START = 'cli.background_process_start',
   LOGIN = 'cli.login',
+  CHECKPOINT_COMMAND_USED = 'cli.checkpoint_command_used',
 
   // Backend
   PROMPT_SENT = 'backend.prompt_sent',
diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 6327d03ca..ac6eabf4d 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -5,6 +5,7 @@ import * as readline from 'readline'
 
 import { type ApiKeyType } from 'common/api-keys/constants'
 import type { CostMode } from 'common/constants'
+import { AnalyticsEvent } from 'common/constants/analytics-events'
 import { Message } from 'common/types/message'
 import { ProjectFileContext } from 'common/util/file'
 import { pluralize } from 'common/util/string'
@@ -36,7 +37,7 @@ import { disableSquashNewlines, enableSquashNewlines } from './display'
 import { displayGreeting, displayMenu } from './menu'
 import { getProjectRoot, isDir } from './project-files'
 import { CliOptions, GitCommand } from './types'
-import { flushAnalytics } from './utils/analytics'
+import { flushAnalytics, trackEvent } from './utils/analytics'
 import { Spinner } from './utils/spinner'
 import {
   isCommandRunning,
@@ -350,6 +351,9 @@ export class CLI {
 
     // Checkpoint commands
     if (isCheckpointCommand(userInput)) {
+      trackEvent(AnalyticsEvent.CHECKPOINT_COMMAND_USED, {
+        command: userInput,
+      })
       if (isCheckpointCommand(userInput, 'undo')) {
         await saveCheckpoint(userInput, this.client, this.readyPromise)
         const toRestore = await handleUndo(this.client, this.rl)
diff --git a/npm-app/src/client.ts b/npm-app/src/client.ts
index 5bc1528a2..b33a92ca9 100644
--- a/npm-app/src/client.ts
+++ b/npm-app/src/client.ts
@@ -6,6 +6,7 @@ import {
   unlinkSync,
   writeFileSync,
 } from 'fs'
+import os from 'os'
 import path from 'path'
 import { Interface } from 'readline'
 
@@ -54,6 +55,7 @@ import {
 import { match, P } from 'ts-pattern'
 import { z } from 'zod'
 
+import packageJson from '../package.json'
 import { getBackgroundProcessUpdates } from './background-process-manager'
 import { activeBrowserRunner } from './browser-runner'
 import { setMessages } from './chat-storage'
@@ -185,7 +187,15 @@ export class Client {
     this.freshPrompt = freshPrompt
     this.reconnectWhenNextIdle = reconnectWhenNextIdle
     this.rl = rl
-    logger.info({ eventId: AnalyticsEvent.APP_LAUNCHED }, 'App launched')
+    logger.info(
+      {
+        eventId: AnalyticsEvent.APP_LAUNCHED,
+        platform: os.platform(),
+        costMode: this.costMode,
+        model: this.model,
+      },
+      'App launched'
+    )
   }
 
   async exit() {
@@ -218,6 +228,8 @@ export class Client {
         email: user.email,
         name: user.name,
         fingerprintId: this.fingerprintId,
+        platform: os.platform(),
+        version: packageJson.version,
       })
       loggerContext.userId = user.id
       loggerContext.userEmail = user.email
@@ -495,6 +507,8 @@ export class Client {
               email: user.email,
               name: user.name,
               fingerprintId: fingerprintId,
+              platform: os.platform(),
+              version: packageJson.version,
             })
             loggerContext.userId = user.id
             loggerContext.userEmail = user.email
diff --git a/npm-app/src/utils/analytics.ts b/npm-app/src/utils/analytics.ts
index 2c0ce9e8d..61ae818b3 100644
--- a/npm-app/src/utils/analytics.ts
+++ b/npm-app/src/utils/analytics.ts
@@ -1,6 +1,11 @@
 import { AnalyticsEvent } from 'common/constants/analytics-events'
 import { PostHog } from 'posthog-node'
 
+// Prints the events to console
+// It's very noisy, so recommended you set this to true
+// only when you're actively adding new analytics
+let DEBUG_DEV_EVENTS = false
+
 // Store the identified user ID
 let currentUserId: string | undefined
 let client: PostHog | undefined
@@ -42,6 +47,12 @@ export function trackEvent(
   }
 
   if (process.env.NEXT_PUBLIC_CB_ENVIRONMENT !== 'production') {
+    if (DEBUG_DEV_EVENTS) {
+      console.log('Analytics event sent', {
+        event,
+        properties,
+      })
+    }
     return
   }
 
@@ -61,6 +72,12 @@ export function identifyUser(userId: string, properties?: Record<string, any>) {
   }
 
   if (process.env.NEXT_PUBLIC_CB_ENVIRONMENT !== 'production') {
+    if (DEBUG_DEV_EVENTS) {
+      console.log('Identify event sent', {
+        userId,
+        properties,
+      })
+    }
     return
   }
 
