diff --git a/web/src/app/page.tsx b/web/src/app/page.tsx
index fe051dc08..a6041d87d 100644
--- a/web/src/app/page.tsx
+++ b/web/src/app/page.tsx
@@ -70,7 +70,7 @@ const Home = () => {
         </main>
 
         <section className="py-10 px-4 relative z-10 space-y-8">
-          <h2 className="text-3xl md:text-4xl font-bold text-center">
+          <h2 className="text-3xl md:text-4xl font-bold text-center px-4 md:px-0">
             Watch a Demo
           </h2>
           <div className="max-w-3xl mx-auto">
@@ -137,7 +137,7 @@ const Home = () => {
         </section>
 
         <section id="features" className="px-4 py-20 relative z-10 space-y-16">
-          <h2 className="text-3xl md:text-4xl font-bold mb-12 text-center">
+          <h2 className="text-3xl md:text-4xl font-bold mb-12 text-center px-4 md:px-0">
             Revolutionize Your Coding Workflow
           </h2>
           <div className="grid md:grid-cols-3 gap-8">
@@ -175,7 +175,7 @@ const Home = () => {
         </section>
 
         <section className="py-20 relative z-10">
-          <h2 className="text-3xl md:text-4xl font-bold text-center">
+          <h2 className="text-3xl md:text-4xl font-bold text-center px-4 md:px-0">
             What Developers Are Saying
           </h2>
           <h6 className="text-center text-gray-700 dark:text-gray-300 text-sm mt-2 mb-12">
@@ -226,7 +226,7 @@ const Home = () => {
         </section>
 
         <section className="py-20 relative z-10">
-          <h2 className="text-3xl md:text-4xl font-bold text-center mb-12">
+          <h2 className="text-3xl md:text-4xl font-bold text-center mb-12 px-4 md:px-0">
             FAQ
           </h2>
           <div className="max-w-3xl mx-auto space-y-4">
@@ -255,7 +255,7 @@ const Home = () => {
         </section>
 
         <section className="py-20 relative z-10">
-          <h2 className="text-3xl md:text-4xl font-bold text-center mb-12">
+          <h2 className="text-3xl md:text-4xl font-bold text-center mb-12 px-4 md:px-0">
             Ready to experience magic?
           </h2>
           <div className="text-center">
diff --git a/web/src/components/footer.tsx b/web/src/components/footer.tsx
index 725cb09c1..2a58ca703 100644
--- a/web/src/components/footer.tsx
+++ b/web/src/components/footer.tsx
@@ -6,7 +6,7 @@ import { siteConfig } from '@/lib/constant'
 export const Footer = () => {
   return (
     <footer className="w-full border-t z-10">
-      <div className="container mx-auto px-4 py-8">
+      <div className="container mx-auto px-4 py-4">
         <div className="grid grid-cols-1 md:grid-cols-4 gap-8">
           {/* Logo and company name */}
           <div className="flex items-center space-x-2">
@@ -32,6 +32,13 @@ export const Footer = () => {
               >
                 Home
               </Link>
+              <Link
+                href="https://news.codebuff.com"
+                target="_blank"
+                className="text-muted-foreground hover:text-primary"
+              >
+                News
+              </Link>
               <Link
                 href="/pricing"
                 className="text-muted-foreground hover:text-primary"
@@ -83,7 +90,7 @@ export const Footer = () => {
 
         <Separator className="my-4" />
 
-        <div className="text-center text-sm text-muted-foreground">
+        <div className="text-center text-xs text-muted-foreground">
           © {new Date().getFullYear()} {siteConfig.title}. All rights reserved.
         </div>
       </div>
diff --git a/web/src/components/navbar/navbar.tsx b/web/src/components/navbar/navbar.tsx
index 6b8723689..3f80884dd 100644
--- a/web/src/components/navbar/navbar.tsx
+++ b/web/src/components/navbar/navbar.tsx
@@ -30,6 +30,13 @@ export const Navbar = async () => {
         <span className="font-mono text-2xl font-bold">Codebuff</span>
       </Link>
       <nav className="hidden md:flex space-x-4">
+        <Link
+          href="https://news.codebuff.com"
+          target="_blank"
+          className="hover:text-blue-400 transition-colors"
+        >
+          News
+        </Link>
         <Link href="/pricing" className="hover:text-blue-400 transition-colors">
           Pricing
         </Link>
diff --git a/web/src/components/ui/banner.tsx b/web/src/components/ui/banner.tsx
index 218dad7b9..21effa0fe 100644
--- a/web/src/components/ui/banner.tsx
+++ b/web/src/components/ui/banner.tsx
@@ -12,14 +12,14 @@ export function Banner() {
   if (!isVisible) return <></>
 
   return (
-    <div className="w-full bg-blue-900 text-white px-4 py-2 relative z-20">
+    <div className="w-full bg-blue-900 text-white px-4 py-2 md:py-0 relative z-20">
       <div className="md:container mx-auto flex items-center justify-between">
         <div className="w-8" />
         <div className="flex items-center gap-2 text-center">
           <Gift className="hidden md:block h-4 w-4" />
           <p className="text-sm">
-            Refer a friend, and earn {CREDITS_REFERRAL_BONUS} credits per month for both
-            of you!{' '}
+            Refer a friend, and earn {CREDITS_REFERRAL_BONUS} credits per month
+            for both of you!{' '}
             <Link href="/referrals" className="underline hover:text-blue-200">
               Learn more
             </Link>
diff --git a/web/src/lib/faq.tsx b/web/src/lib/faq.tsx
index ad8e07d66..307b17aee 100644
--- a/web/src/lib/faq.tsx
+++ b/web/src/lib/faq.tsx
@@ -9,29 +9,29 @@ export const faqs: faq[] = [
   {
     question: 'Do you store my data?',
     answer: (
-      <>
+      <p>
         We don't store your codebase. Mostly our server is a thin client that
         passes data along to OpenAI and Anthropic. We store logs of your chats
         temporarily to help us debug. We also store portions of your chat
         history in our database. Soon, we plan to support a Privacy Mode where
         no data at all is stored.
-      </>
+      </p>
     ),
   },
   {
     question: 'What models do you use?',
     answer: (
-      <>
+      <p>
         We primarily use Claude 3.5 Sonnet for the coding, and Claude 3.5 Haiku
         to find relevant files. We also use GPT-4o-mini as a fallback to rewrite
         files with an intended edit (using their fancy Predicted Outputs API).
-      </>
+      </p>
     ),
   },
   {
     question: 'Can I specify custom instructions for Codebuff?',
     answer: (
-      <>
+      <p>
         Yes! We recommend you create knowledge.md files to give Codebuff more
         context about your codebase, like you're introducing it to another
         engineer. All file names ending in knowledge.md are loaded into context
@@ -40,42 +40,42 @@ export const faqs: faq[] = [
         Codebuff will not write one, but it will update existing knowledge.md
         files autonomously. The Codebuff codebase currently has a knowledge.md
         in almost every directory.
-      </>
+      </p>
     ),
   },
   {
     question: 'Can I tell Codebuff to ignore certain files?',
     answer: (
-      <>
+      <p>
         Codebuff by default will not read files that are specified in your
         .gitignore. You can also create a '.codebuffignore' file to specify
         additional files or folders to ignore.
-      </>
+      </p>
     ),
   },
   {
     question: 'Help! Codebuff made a bad change, how I undo it?',
     answer: (
-      <>
+      <p>
         Type 'undo' to remove the edits! For other cool features, type 'help' in
         the terminal when Codebuff is running.
-      </>
+      </p>
     ),
   },
   {
     question: 'Can I see the changes that Codebuff has made as a diff?',
     answer: (
-      <>
+      <p>
         Yes! Type 'diff' after Codebuff has made a change to see the edits! For
         other cool features, type 'help' in the terminal when Codebuff is
         running.
-      </>
+      </p>
     ),
   },
   {
     question: 'How does Codebuff actually work?',
     answer: (
-      <>
+      <p>
         You invoke it in your terminal with 'codebuff' and it starts by running
         through the source files in that directory and subdirectories and
         parsing out all the function and class names (or equivalents in 11
@@ -86,13 +86,13 @@ export const faqs: faq[] = [
         websockets to ferry data back and forth to clients. Claude 3.5 Haiku
         picks the relevant files, and we load them into context and Claude 3.5
         Sonnet responds with the right edit.
-      </>
+      </p>
     ),
   },
   {
     question: 'What have other users made with Codebuff?',
     answer: (
-      <>
+      <p>
         Many users do real work professionally on large codebases with Codebuff.
         You can use Codebuff to write features faster, create and run unit tests
         until they pass, or add integrations to new API's. Codebuff is great for
@@ -101,13 +101,13 @@ export const faqs: faq[] = [
         scratch over the weekend for their teams and personal use. At the end of
         the day, with Codebuff you can spend more of your time thinking about
         architecture and design, instead of implementation details.
-      </>
+      </p>
     ),
   },
   {
     question: 'Can I integrate Codebuff into my app/product/system?',
     answer: (
-      <>
+      <p>
         We currently have an alpha SDK that exposes the same natural language
         interface for your apps to call and receive code edits.{' '}
         <a
@@ -116,24 +116,24 @@ export const faqs: faq[] = [
         >
           Sign up here for early access!
         </a>
-      </>
+      </p>
     ),
   },
   {
     question: 'Why is Codebuff so expensive?',
     answer: (
-      <>
+      <p>
         We realize it costs a bit more than alternatives, but in exchange,
         Codebuff does more LLM calls using more examples from your codebase,
         leading to better code edits.
-      </>
+      </p>
     ),
   },
   {
     question: 'I have more questions!',
     answer: (
-      <>
-        Contact us at{' '}
+      <p>
+        We're happy to answer! Contact us at{' '}
         <a
           href="mailto:support@codebuff.com"
           className="no-underline hover:underline"
@@ -148,7 +148,7 @@ export const faqs: faq[] = [
           join our Discord
         </a>
         !
-      </>
+      </p>
     ),
   },
 ]
