diff --git a/common/src/templates/initial-agents-dir/types/agent-definition.ts b/common/src/templates/initial-agents-dir/types/agent-definition.ts
index 6029ca070..5cd9de223 100644
--- a/common/src/templates/initial-agents-dir/types/agent-definition.ts
+++ b/common/src/templates/initial-agents-dir/types/agent-definition.ts
@@ -15,6 +15,7 @@
  */
 
 import type { Message } from './codebuff-message'
+import { ToolResultOutput } from './content-part'
 import type * as Tools from './tools'
 type ToolName = Tools.ToolName
 
@@ -184,7 +185,7 @@ export interface AgentDefinition {
     void,
     {
       agentState: AgentState
-      toolResult: string | undefined
+      toolResult: ToolResultOutput[] | undefined
       stepsComplete: boolean
     }
   >
