diff --git a/web/src/components/footer.tsx b/web/src/components/footer.tsx
index d9945d6d4..b7d6e4534 100644
--- a/web/src/components/footer.tsx
+++ b/web/src/components/footer.tsx
@@ -4,10 +4,10 @@ import Link from 'next/link'
 export const Footer = () => {
   return (
     <footer className="text-muted-foreground w-full text-center text-sm">
-      © {new Date().getFullYear()} coded with 🤖{' '}
+      © {new Date().getFullYear()} {' '}
       <Button variant="link" className="p-0" asChild>
         <Link href="https://discord.gg/mcWTGjgTj3" target="_blank">
-          (Keep in touch!)
+          Join our Discord!
         </Link>
       </Button>
     </footer>
