diff --git a/web/src/app/referrals/[code]/page.tsx b/web/src/app/referrals/[code]/page.tsx
index ee744fb9c..26f7bebe2 100644
--- a/web/src/app/referrals/[code]/page.tsx
+++ b/web/src/app/referrals/[code]/page.tsx
@@ -1,6 +1,16 @@
 'use client'
 
+import { useQuery } from '@tanstack/react-query'
+import { sleep } from 'common/util/promise'
+import { CheckIcon, CopyIcon, GiftIcon } from 'lucide-react'
 import { useSession } from 'next-auth/react'
+import { useSearchParams } from 'next/navigation'
+import posthog from 'posthog-js'
+import { useEffect, useState } from 'react'
+
+import type { ReferralCodeResponse } from '@/app/api/referrals/[code]/route'
+import CardWithBeams from '@/components/card-with-beams'
+import { Button } from '@/components/ui/button'
 import {
   Card,
   CardContent,
@@ -8,19 +18,11 @@ import {
   CardHeader,
   CardTitle,
 } from '@/components/ui/card'
-import { Skeleton } from '@/components/ui/skeleton'
-import { useEffect, useState } from 'react'
 import { Input } from '@/components/ui/input'
-import { sleep } from 'common/util/promise'
-import { CopyIcon, CheckIcon, GiftIcon } from 'lucide-react'
-import { useQuery, useQueryClient } from '@tanstack/react-query'
-import type { ReferralCodeResponse } from '@/app/api/referrals/[code]/route'
-import { Button } from '@/components/ui/button'
+import { Skeleton } from '@/components/ui/skeleton'
 import { env } from '@/env.mjs'
-import CardWithBeams from '@/components/card-with-beams'
-import { useSearchParams } from 'next/navigation'
 import { storeSearchParams } from '@/lib/trackConversions'
-import posthog from 'posthog-js'
+
 
 const InputWithCopyButton = ({ text }: { text: string }) => {
   const [copied, setCopied] = useState(false)
@@ -153,7 +155,7 @@ export default function RedeemPage({ params }: { params: { code: string } }) {
               </li>
               <li>
                 Run Codebuff in Terminal
-                <InputWithCopyButton text={'codebuff'} />
+                <InputWithCopyButton text={'cd /my/project/dir && codebuff'} />
               </li>
               {!data?.status.reason && (
                 <li>
