diff --git a/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts b/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts
index be1974033..2f6205d8b 100644
--- a/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts
+++ b/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts
@@ -94,11 +94,17 @@ export const promptAiSdkStream = async function* (
     if (chunk.type === 'error') {
       logger.error({ chunk }, 'Error from AI SDK')
       if (process.env.ENVIRONMENT !== 'prod') {
-        throw new Error(
-          typeof chunk.error === 'string'
-            ? chunk.error
-            : JSON.stringify(chunk.error)
-        )
+        throw chunk.error instanceof Error
+          ? new Error(`Error from AI SDK: ${chunk.error.message}`, {
+              cause: chunk.error,
+            })
+          : new Error(
+              `Error from AI SDK: ${
+                typeof chunk.error === 'string'
+                  ? chunk.error
+                  : JSON.stringify(chunk.error)
+              }`
+            )
       }
     }
     if (chunk.type === 'reasoning') {
