diff --git a/.agents/base.ts b/.agents/base.ts
index 0aa6e00d7..c9f39480f 100644
--- a/.agents/base.ts
+++ b/.agents/base.ts
@@ -6,7 +6,7 @@ import type { SecretAgentDefinition } from './types/secret-agent-definition'
 const definition: SecretAgentDefinition = {
   id: 'base',
   publisher,
-  ...base('anthropic/claude-sonnet-4'),
+  ...base('anthropic/claude-4-sonnet-20250522'),
 }
 
 export default definition
diff --git a/.agents/factory/base.ts b/.agents/factory/base.ts
index 33f4183b5..bc39e9345 100644
--- a/.agents/factory/base.ts
+++ b/.agents/factory/base.ts
@@ -8,10 +8,10 @@ import {
 import { AgentTemplateTypes } from '../types/secret-agent-definition'
 
 import type { SecretAgentDefinition } from '../types/secret-agent-definition'
-import type { Model } from '@codebuff/common/constants'
+import { ModelName } from 'types/agent-definition'
 
 export const base = (
-  model: Model,
+  model: ModelName,
   allAvailableAgents?: string[],
 ): Omit<SecretAgentDefinition, 'id'> => ({
   model,
diff --git a/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts b/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts
index e81bd7ce5..4876da55d 100644
--- a/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts
+++ b/backend/src/llm-apis/vercel-ai-sdk/ai-sdk.ts
@@ -180,9 +180,12 @@ export const promptAiSdkStream = async function* (
       cacheReadInputTokens =
         openrouterUsage.promptTokensDetails?.cachedTokens ?? 0
       inputTokens = openrouterUsage.promptTokens - cacheReadInputTokens
-      costOverrideDollars =
-        (openrouterUsage.cost ?? 0) +
-        (openrouterUsage.costDetails?.upstreamInferenceCost ?? 0)
+
+      // Note (James): This seemed to be forcing 0 cost openrouter models where we use keys
+      // so I disabled the cost override for now.
+      costOverrideDollars = undefined
+      // (openrouterUsage.cost ?? 0) +
+      // (openrouterUsage.costDetails?.upstreamInferenceCost ?? 0)
     }
   }
 
