diff --git a/web/src/content/help/faq.mdx b/web/src/content/help/faq.mdx
index f65405d1a..a89cfb107 100644
--- a/web/src/content/help/faq.mdx
+++ b/web/src/content/help/faq.mdx
@@ -37,7 +37,7 @@ We don't store your codebase. Our server is a thin client that passes data along
 
 ## Can I trust Codebuff with full access to my terminal?
 
-None of our users have had significant issues so far. However, if you prefer an extra layer of safety, we made a [Dockerfile](https://github.com/CodebuffAI/codebuff-docker) that you can use. It'll set up Codebuff in an environment that's isolated from your filesystem and only has the code you want to share.
+None of our users have had significant issues so far. However, if you prefer an extra layer of safety, we made a [Dockerfile](https://github.com/CodebuffAI/codebuff-community/tree/main/utils/docker) that you can use. It'll set up Codebuff in an environment that's isolated from your filesystem and only has the code you want to share.
 
 ## Can I specify custom instructions for Codebuff?
 
