diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index cfc3ffaf8..65c010e2d 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -712,6 +712,7 @@ async function getFileReadingUpdates(
   const previousFilePaths = uniq(Object.keys(previousFiles))
 
   const editedFilePaths = messages
+    .filter(({ role }) => role === 'assistant')
     .map(toContentString)
     .filter((content) => content.includes('<write_file'))
     .flatMap((content) => Object.keys(parseFileBlocks(content)))
