diff --git a/web/src/app/page.tsx b/web/src/app/page.tsx
index d30ab7fa7..46db247f4 100644
--- a/web/src/app/page.tsx
+++ b/web/src/app/page.tsx
@@ -1,6 +1,6 @@
 'use client'
 import { Button } from '@/components/ui/button'
-import { FolderCodeIcon, TerminalIcon, Copy, ZapIcon, Play } from 'lucide-react'
+import { FolderCodeIcon, TerminalIcon, Copy, ZapIcon, Play, ChevronDown } from 'lucide-react'
 import Link from 'next/link'
 import { Dialog, DialogContent } from '@/components/ui/dialog'
 import { BackgroundBeams } from '@/components/ui/background-beams'
@@ -9,6 +9,7 @@ import { useToast } from '@/components/ui/use-toast'
 import { Card, CardContent } from '@/components/ui/card'
 import Marquee from '@/components/ui/marquee'
 import { testimonials } from '@/lib/testimonials'
+import { faqs } from '@/lib/faq'
 import { cn } from '@/lib/utils'
 import { useState } from 'react'
 
@@ -215,6 +216,25 @@ const Home = () => {
           ))}
         </section>
 
+        <section className="py-20 relative z-10">
+          <h2 className="text-3xl md:text-4xl font-bold text-center mb-12">
+            FAQ
+          </h2>
+          <div className="max-w-3xl mx-auto space-y-4">
+            {faqs.map((item, index) => (
+              <details key={index}>
+                <summary className="flex cursor-pointer font-semibold w-full text-left p-4 text-gray-700 dark:text-gray-300 justify-between items-center marker:[font-size:0px]">
+                  {item.question}
+                  <ChevronDown className="h-4 w-4" />
+                </summary>
+                <div className="text-gray-700 dark:text-gray-300 px-4 pb-4 items-center">
+                  {item.answer}
+                </div>
+              </details>
+            ))}
+          </div>
+        </section>
+
         <section className="py-20 relative z-10">
           <h2 className="text-3xl md:text-4xl font-bold text-center mb-12">
             Ready to experience magic?
diff --git a/web/src/lib/faq.tsx b/web/src/lib/faq.tsx
new file mode 100644
index 000000000..1d1ddcf31
--- /dev/null
+++ b/web/src/lib/faq.tsx
@@ -0,0 +1,72 @@
+import { ReactNode } from 'react'
+
+export type faq = {
+  question: string
+  answer: ReactNode
+}
+
+export const faqs: faq[] = [
+  {
+    question: "Do you store my data?",
+    answer: "We don't store your codebase. Mostly our server is a thin client that passes data along to LLM providers like OpenAI and Anthropic. We store logs of your chats temporarily to help us debug. We also store portions of chat in our database. Soon, we plan to support a Privacy Mode where no data at all is stored."
+  },
+  {
+    question: "What models do you use?",
+    answer: "We primarily use Claude 3.5 Sonnet for the coding, and Claude 3.5 Haiku to find relvant files. We also use GPT-4o-mini as a fallback to rewrite files with an intended edit."
+  },
+  {
+    question: "Can I specify custom instructions for Codebuff?",
+    answer: "Yes! We recommend you create knowledge.md files to give Codebuff more context about your codebase, like you're introducing it to another engineer. All file names ending in knowledge.md are loaded into context automatically, and you can use the files to do your own prompt engineering in them too. We currently have a knowledge.md in almost every directory"
+  },
+  {
+    question: "Can I tell Codebuff to ignore certain files?",
+    answer: "Yes! Use '.codebuffignore' to tell Codebuff to ignore these files or folders. Codebuff also does not read 'gitignore'd files."
+  },
+  {
+    question: "Help! Codebuff made a bad change, and how I undo it?",
+    answer: "Type 'undo' after Codebuff has made a change to remove the edits."
+  },
+  {
+    question: "Can I see the changes that Codebuff has made as a diff?",
+    answer: "Yes! Type 'diff' after Codebuff has made a change to see the edits! For all the cool features that Codebuff has, type 'help' in the terminal when Codebuff is running."
+  },
+  {
+    question: "How does Codebuff actually work?",
+    answer: "You invoke it in your terminal with 'codebuff' and it starts by running through the source files in that directory and subdirectories and parsing out all the function and class names (or equivalents in 11 languages) with a tree-sitter library. Then, it fires off a request to Claude Haiku 3.5 to cache this codebase context so user inputs can be responded to with lower latency (Prompt caching is OP!). We have a stateless server that passes messages along to Anthropic or OpenAI and websockets to ferry data back and forth to clients. Claude 3.5 Haiku picks the relevant files, and we load them into context and Claude 3.5 Sonnet responds with the right edit."
+  },
+  {
+    question: "What have other users made with Codebuff?",
+    answer: "Many users built real apps over a weekend for their teams and personal use. Others also frequently use Codebuff to write unit tests. They would build a feature in parallel with unit tests and have Codebuff do loops to fix up the code until the tests pass. They would also ask it to do drudge work like set up Oauth flows or API scaffolding. At the end of the day, you can spend more of their time thinking about architecture and design, instead of implementation details."
+  },
+  {
+    question: "Can I integrate Codebuff into my app/product/system?",
+    answer: (
+      <>
+        We currently have an alpha SDK that exposes the same natural language interface for your apps to call and receive code edits.{' '}
+        <a href="https://codebuff.retool.com/form/c8b15919-52d0-4572-aca5-533317403dde" className="underline hover:text-blue-500">
+          Sign up here for early access!
+        </a>
+      </>
+    )
+  },
+  {
+    question: "Why is Codebuff so expensive?",
+    answer: "We realize this is a lot more than competitors, but in exchange, we do more LLM calls with more examples native to your codebase that leads to more accurate identification of relevant files and better code edits."
+  },
+  {
+    question: "I have more questions!",
+    answer: (
+      <>
+        Contact us at{' '}
+        <a href="mailto:support@codebuff.com" className="underline hover:text-blue-500">
+          support@codebuff.com
+        </a>
+        {' '}or{' '}
+        <a href="https://discord.gg/mcWTGjgTj3" className="underline hover:text-blue-500">
+          join our Discord
+        </a>
+        !
+      </>
+    )
+  }
+]
\ No newline at end of file
