diff --git a/web/src/app/referrals/page.tsx b/web/src/app/referrals/page.tsx
index 776e7e337..443f42c94 100644
--- a/web/src/app/referrals/page.tsx
+++ b/web/src/app/referrals/page.tsx
@@ -136,7 +136,7 @@ const ReferralsPage = () => {
                     <p>
                       {data.limitReached
                         ? `Dang, you've got a lot of friends! You have reached your referral limit.`
-                        : `Send this link to your friend. We'll take it from there!`}
+                        : `Refer a friend and earn ${CREDITS_REFERRAL_BONUS} credits per month! Share this link with them:`}
                     </p>
                     <div className="relative">
                       {loading ? (
