diff --git a/packages/internal/src/env.ts b/packages/internal/src/env.ts
index 01c79cdc9..b3498c31b 100644
--- a/packages/internal/src/env.ts
+++ b/packages/internal/src/env.ts
@@ -15,7 +15,6 @@ const envSchema = {
     HELICONE_API_KEY: z.string().min(1),
     OPEN_AI_KEY: z.string().min(1),
     GEMINI_API_KEY: z.string().min(1),
-    GOOGLE_GENERATIVE_AI_API_KEY: z.string().min(1),
     DEEPSEEK_API_KEY: z.string().min(1),
     OPEN_ROUTER_API_KEY: z.string().min(1),
     RELACE_API_KEY: z.string().min(1),
@@ -63,7 +62,6 @@ const envSchema = {
     HELICONE_API_KEY: process.env.HELICONE_API_KEY,
     OPEN_AI_KEY: process.env.OPEN_AI_KEY,
     GEMINI_API_KEY: process.env.GEMINI_API_KEY,
-    GOOGLE_GENERATIVE_AI_API_KEY: process.env.GOOGLE_GENERATIVE_AI_API_KEY,
     DEEPSEEK_API_KEY: process.env.DEEPSEEK_API_KEY,
     OPEN_ROUTER_API_KEY: process.env.OPEN_ROUTER_API_KEY,
     RELACE_API_KEY: process.env.RELACE_API_KEY,
