diff --git a/cli/src/hooks/use-send-message.ts b/cli/src/hooks/use-send-message.ts
index 8fd8d16c4..0e4be4e64 100644
--- a/cli/src/hooks/use-send-message.ts
+++ b/cli/src/hooks/use-send-message.ts
@@ -643,7 +643,6 @@ export const useSendMessage = ({
           maxAgentSteps: 40,
 
           handleStreamChunk: (event) => {
-            logger.info({ event }, 'asdf event')
             if (typeof event === 'string' || event.type === 'reasoning_chunk') {
               const eventObj:
                 | { type: 'text'; text: string }
