diff --git a/npm-app/src/utils/analytics.ts b/npm-app/src/utils/analytics.ts
index 5de0a8bd9..d16606a37 100644
--- a/npm-app/src/utils/analytics.ts
+++ b/npm-app/src/utils/analytics.ts
@@ -51,7 +51,10 @@ export function trackEvent(
     return
   }
   if (!client) {
-    throw new Error('Analytics client not initialized')
+    if (process.env.NEXT_PUBLIC_CB_ENVIRONMENT === 'production') {
+      throw new Error('Analytics client not initialized')
+    }
+    return
   }
 
   if (process.env.NEXT_PUBLIC_CB_ENVIRONMENT !== 'production') {
