diff --git a/web/src/app/login/page.tsx b/web/src/app/login/page.tsx
index d063d3612..2eb71c746 100644
--- a/web/src/app/login/page.tsx
+++ b/web/src/app/login/page.tsx
@@ -44,15 +44,16 @@ const Home = () => {
           {authCode ? (
             <Card>
               <CardHeader>
-                <CardTitle>Confirm cli login</CardTitle>
+                <CardTitle className="mb-2">Login</CardTitle>
                 <CardDescription>
-                  If you just logged into Manicode from the command line, please
-                  select an OAuth provider below to continue.
+                  Continue to sign in to the Manicode CLI.
+                  {/* If you just logged into Manicode from the command line, please
+                  select an OAuth provider below to continue. */}
                 </CardDescription>
-                <CardDescription>
+                {/* <CardDescription>
                   (Otherwise, you can just close this window. Phishing attack
                   averted, phew!)
-                </CardDescription>
+                </CardDescription> */}
               </CardHeader>
               <SignInCardFooter />
             </Card>
diff --git a/web/src/components/sign-in/sign-in-button.tsx b/web/src/components/sign-in/sign-in-button.tsx
index 1bd531450..2c20216be 100644
--- a/web/src/components/sign-in/sign-in-button.tsx
+++ b/web/src/components/sign-in/sign-in-button.tsx
@@ -42,7 +42,9 @@ export const SignInButton = ({
         className="rounded-full"
       />
       Continue with{' '}
-      {providerName.charAt(0).toUpperCase() + providerName.slice(1)}
+      {providerName === 'github'
+        ? 'GitHub'
+        : providerName.charAt(0).toUpperCase() + providerName.slice(1)}
     </Button>
   )
 }
