diff --git a/backend/Dockerfile b/backend/Dockerfile
index adeb118a5..d4a43fef6 100644
--- a/backend/Dockerfile
+++ b/backend/Dockerfile
@@ -27,6 +27,7 @@ COPY . .
 FROM base AS release
 COPY --from=install /temp/prod/node_modules node_modules
 COPY src .
+COPY .env .
 COPY package.json .
 COPY tsconfig.json .
 
diff --git a/backend/src/websockets/websocket-action.ts b/backend/src/websockets/websocket-action.ts
index 7fd9f55e3..bde36708a 100644
--- a/backend/src/websockets/websocket-action.ts
+++ b/backend/src/websockets/websocket-action.ts
@@ -87,9 +87,15 @@ const onCheckNpmVersion = async (
   { version }: Extract<ClientAction, { type: 'check-npm-version' }>,
   ws: WebSocket
 ) => {
-  const backendPackageJsonPath = path.join(__dirname, '..', '..', 'package.json')
-  const backendPackageJson = JSON.parse(fs.readFileSync(backendPackageJsonPath, 'utf-8'))
-  const latestVersion = backendPackageJson.version
+  let latestVersion = version
+
+  if (process.env.NODE_ENV === 'production') {
+    const backendPackageJsonPath = path.join(__dirname, '..', 'package.json')
+    const backendPackageJson = JSON.parse(
+      fs.readFileSync(backendPackageJsonPath, 'utf-8')
+    )
+    latestVersion = backendPackageJson.version
+  }
 
   const isUpToDate = version === latestVersion
 
diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 45cf32e8a..ea07123e6 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -60,6 +60,7 @@ export class CLI {
       if (key.name === 'escape') {
         this.handleEscKey()
       }
+      this.detectPasting()
     })
   }
 
