diff --git a/common/src/websockets/websocket-client.ts b/common/src/websockets/websocket-client.ts
index 0aa48fcf1..df69086a4 100644
--- a/common/src/websockets/websocket-client.ts
+++ b/common/src/websockets/websocket-client.ts
@@ -45,7 +45,7 @@ export function formatState(state: ReadyState) {
 type OutstandingTxn = {
   resolve: () => void
   reject: (err: Error) => void
-  timeout?: NodeJS.Timer
+  timeout?: any
 }
 
 /** Client for the API websocket realtime server. Automatically manages reconnection
@@ -59,8 +59,8 @@ export class APIRealtimeClient {
   txid: number
   // all txns that are in flight, with no ack/error/timeout
   txns: Map<number, OutstandingTxn>
-  connectTimeout?: NodeJS.Timer
-  heartbeat?: NodeJS.Timer
+  connectTimeout?: any
+  heartbeat?: any
   hadError = false
   onError: () => void
 
