diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 1b2413c10..47055ef23 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -195,7 +195,11 @@ export class CLI {
   }
 
   private async handleUserInput(userInput: string) {
-    if (!userInput) return
+    this.rl.setPrompt('')
+    if (!userInput) {
+      this.freshPrompt()
+      return
+    }
     userInput = userInput.trim()
     if (await this.processCommand(userInput)) {
       return
