diff --git a/web/src/lib/testimonials.ts b/web/src/lib/testimonials.ts
index ae5499560..5142e4ad9 100644
--- a/web/src/lib/testimonials.ts
+++ b/web/src/lib/testimonials.ts
@@ -80,7 +80,7 @@ export const testimonials: Testimonial[][] = [
       quote:
         "Im basically hiring an engineer for 100/month, that's how I see it",
       author: 'Shardool Patel',
-      title: 'Founder, CTO',
+      title: 'Founder & CTO',
       avatar: '/testimonials/shardool-patel.jpg',
       link: '/testimonials/proof/shardool-patel.png',
     },
