diff --git a/agents/base2/base2.ts b/agents/base2/base2.ts
index 8f2781f67..31ffa8943 100644
--- a/agents/base2/base2.ts
+++ b/agents/base2/base2.ts
@@ -63,8 +63,8 @@ export function createBase2(
       !isFast && !noAskUser && 'suggest_followups',
       'str_replace',
       'write_file',
-      'propose_str_replace',
-      'propose_write_file',
+      !isFree && 'propose_str_replace',
+      !isFree && 'propose_write_file',
       !noAskUser && 'ask_user',
       'skill',
       'set_output',
