diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 8479f8ec8..2df0fe5cf 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -8,7 +8,9 @@ import {
   ONE_TIME_LABELS,
   type CostMode,
 } from 'common/constants'
+import { AnalyticsEvent } from 'common/constants/analytics-events'
 import { getToolCallString } from 'common/constants/tools'
+import { trackEvent } from 'common/src/analytics'
 import { AgentState, ToolResult } from 'common/types/agent-state'
 import { Message } from 'common/types/message'
 import { buildArray } from 'common/util/array'
@@ -622,6 +624,10 @@ export const mainPrompt = async (
     }
 
     const { name, parameters } = toolCall
+    trackEvent(AnalyticsEvent.TOOL_USE, userId ?? '', {
+      tool: name,
+      parameters,
+    })
     if (name === 'write_file') {
       // write_file tool calls are handled as they are streamed in.
     } else if (name === 'add_subgoal' || name === 'update_subgoal') {
diff --git a/common/src/analytics.ts b/common/src/analytics.ts
index 28d4dfccf..b71361ef8 100644
--- a/common/src/analytics.ts
+++ b/common/src/analytics.ts
@@ -17,7 +17,7 @@ export function initAnalytics() {
   client = new PostHog(env.NEXT_PUBLIC_POSTHOG_API_KEY, {
     host: env.NEXT_PUBLIC_POSTHOG_HOST_URL,
     flushAt: 1,
-    flushInterval: 1000,
+    flushInterval: 0,
   })
 }
 export async function flushAnalytics() {
@@ -37,7 +37,7 @@ export function trackEvent(
   }
 
   if (env.NEXT_PUBLIC_CB_ENVIRONMENT !== 'production') {
-    logger.info({ event }, 'Analytics event tracked')
+    logger.info({ payload: { event, properties } }, 'Analytics event tracked')
     return
   }
 
diff --git a/common/src/constants/analytics-events.ts b/common/src/constants/analytics-events.ts
index 4976e412f..df33d95be 100644
--- a/common/src/constants/analytics-events.ts
+++ b/common/src/constants/analytics-events.ts
@@ -13,6 +13,7 @@ export enum AnalyticsEvent {
   // Backend
   PROMPT_SENT = 'backend.prompt_sent',
   CREDIT_GRANT = 'backend.credit_grant',
+  TOOL_USE = 'backend.tool_use',
 
   // Web
   SIGNUP = 'web.signup',
