diff --git a/backend/src/process-file-block.ts b/backend/src/process-file-block.ts
index 1faa63510..d6fe3f92b 100644
--- a/backend/src/process-file-block.ts
+++ b/backend/src/process-file-block.ts
@@ -91,22 +91,6 @@ export async function processFileBlock(
     countTokens(normalizedInitialContent) + countTokens(normalizedEditSnippet)
 
   if (tokenCount > LARGE_FILE_TOKEN_LIMIT) {
-    // Delete this later vvv
-    if (tokenCount < 64_000) {
-      // Temporary: send to relace
-      const messageId = generateCompactId('cb-')
-      // no need to await, just send the request and continue
-      sendToRelaceLongContext(normalizedInitialContent, normalizedEditSnippet, {
-        clientSessionId,
-        fingerprintId,
-        userInputId,
-        userId,
-        messageId,
-        userMessage: lastUserPrompt,
-      })
-    }
-    // Delete this later ^^^
-
     const largeFileContent = await handleLargeFile(
       normalizedInitialContent,
       normalizedEditSnippet,
@@ -211,7 +195,7 @@ export async function processFileBlock(
   }
 }
 
-const LARGE_FILE_TOKEN_LIMIT = 16_000
+const LARGE_FILE_TOKEN_LIMIT = 64_000
 
 export async function handleLargeFile(
   oldContent: string,
