diff --git a/sdk/CHANGELOG.md b/sdk/CHANGELOG.md
index 1f52bfb44..b82360aae 100644
--- a/sdk/CHANGELOG.md
+++ b/sdk/CHANGELOG.md
@@ -8,6 +8,10 @@ All notable changes to the @codebuff/sdk package will be documented in this file
 
 - `closeConnection` method in `CodebuffClient`
 
+### Changed
+
+- Automatic parsing of `knowledgeFiles` if not provided
+
 ## [0.1.8] - 2025-08-13
 
 ### Added
diff --git a/sdk/src/run-state.ts b/sdk/src/run-state.ts
index 3534925ca..c7d5e2d2d 100644
--- a/sdk/src/run-state.ts
+++ b/sdk/src/run-state.ts
@@ -22,7 +22,26 @@ export function initialSessionState(
     maxAgentSteps?: number
   },
 ) {
-  const { knowledgeFiles = {}, agentDefinitions = [] } = options
+  const { projectFiles = {}, agentDefinitions = [] } = options
+  let { knowledgeFiles } = options
+
+  if (knowledgeFiles === undefined) {
+    knowledgeFiles = {}
+    for (const [filePath, fileContents] of Object.entries(projectFiles)) {
+      if (filePath in projectFiles) {
+        continue
+      }
+      const lowercasePathName = filePath.toLowerCase()
+      if (
+        !lowercasePathName.endsWith('knowledge.md') &&
+        !lowercasePathName.endsWith('claude.md')
+      ) {
+        continue
+      }
+
+      knowledgeFiles[filePath] = fileContents
+    }
+  }
 
   // Process agentDefinitions array and convert handleSteps functions to strings
   const processedAgentTemplates: Record<string, any> = {}
