diff --git a/packages/agent-runtime/src/prompt-agent-stream.ts b/packages/agent-runtime/src/prompt-agent-stream.ts
index ab0918690..ac8a60b40 100644
--- a/packages/agent-runtime/src/prompt-agent-stream.ts
+++ b/packages/agent-runtime/src/prompt-agent-stream.ts
@@ -57,7 +57,7 @@ export const getAgentStreamFromTemplate = (params: {
     return {
       getStream: async function* stream(): ReturnType<PromptAiSdkStreamFn> {
         yield { type: 'text', text: textOverride!, agentId }
-        return null
+        return crypto.randomUUID()
       },
     }
   }
