diff --git a/common/src/constants.ts b/common/src/constants.ts
index 4296034b1..71b9eb207 100644
--- a/common/src/constants.ts
+++ b/common/src/constants.ts
@@ -46,18 +46,20 @@ export const SKIPPED_TERMINAL_COMMANDS = [
 
 export const MAX_DATE = new Date(86399999999999)
 
-export type UsageLimits = 'ANON' | 'FREE' | 'PAID'
+export type UsageLimits = 'ANON' | 'FREE' | 'PAID' | 'PRO_PLUS'
 export const CREDITS_USAGE_LIMITS: Record<UsageLimits, number> =
   process.env.NEXT_PUBLIC_ENVIRONMENT === 'local'
     ? {
         ANON: 1_000_000,
         FREE: 2_500_000,
         PAID: 10_000_000,
+        PRO_PLUS: 55_000_000,
       }
     : {
         ANON: 750,
         FREE: 1_500,
         PAID: 10_000,
+        PRO_PLUS: 55_000,
       }
 export const CREDITS_REFERRAL_BONUS = 500
 
diff --git a/web/src/app/pricing/page.tsx b/web/src/app/pricing/page.tsx
index bdb1e6e82..0d44166d1 100644
--- a/web/src/app/pricing/page.tsx
+++ b/web/src/app/pricing/page.tsx
@@ -120,6 +120,30 @@ const PricingPage = () => {
         </div>
       ),
     },
+    {
+      name: 'Pro Plus',
+      price: '$500/month',
+      credits: CREDITS_USAGE_LIMITS.PRO_PLUS,
+      features: [
+        '$0.95 per 100 credits afterwards',
+        'Priority support over email or Discord',
+        'Dedicated account manager',
+        'Custom integrations available',
+      ],
+      cardFooterChildren: (
+        <div className="w-full flex flex-col items-center text-center justify-center space-y-2">
+          <Button
+            className="w-full bg-blue-600 hover:bg-blue-700 text-white transition-colors"
+            onClick={() => handleCreateCheckoutSession()}
+            disabled={isPending}
+          >
+            {isPending && <Icons.loader className="mr-2 size-4 animate-spin" />}
+            Contact Sales
+          </Button>
+        </div>
+      ),
+    },
+
     {
       name: 'Enterprise',
       price: 'Custom',
