diff --git a/.agents/base/base-max.ts b/.agents/base/base-max.ts
index d709d1b66..dd5edb913 100644
--- a/.agents/base/base-max.ts
+++ b/.agents/base/base-max.ts
@@ -12,7 +12,7 @@ const definition: SecretAgentDefinition = {
     'find-all-referencer',
     'researcher-web-sonnet',
     'researcher-docs-sonnet',
-    'decomposing-planner',
+    'generate-plan',
     'decomposing-thinker',
     'reviewer',
     'context-pruner',
