diff --git a/manicode/package.json b/manicode/package.json
index 8ba53080e..0a1ffc6e7 100644
--- a/manicode/package.json
+++ b/manicode/package.json
@@ -1,6 +1,6 @@
 {
   "name": "manicode",
-  "version": "1.0.9",
+  "version": "1.0.11",
   "description": "AI dev assistant",
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
@@ -38,4 +38,4 @@
     "@types/lodash": "*",
     "typescript": "*"
   }
-}
+}
\ No newline at end of file
diff --git a/manicode/src/chat-storage.ts b/manicode/src/chat-storage.ts
index 545fa1a94..e0c36e856 100644
--- a/manicode/src/chat-storage.ts
+++ b/manicode/src/chat-storage.ts
@@ -1,6 +1,7 @@
 import * as fs from 'fs'
 import * as path from 'path'
 import { Message } from 'common/actions'
+import { projectRoot } from './config'
 
 const MANICODE_DIR = '.manicode'
 const CHATS_DIR = 'chats'
@@ -23,7 +24,7 @@ export class ChatStorage {
   private currentChat: Chat
   private currentVersionIndex: number
 
-  constructor(projectRoot: string) {
+  constructor() {
     this.baseDir = path.join(projectRoot, MANICODE_DIR, CHATS_DIR)
     this.ensureDirectoryExists()
     this.currentChat = this.createChat()
diff --git a/manicode/src/config.ts b/manicode/src/config.ts
index 8620fcccc..4452dbe00 100644
--- a/manicode/src/config.ts
+++ b/manicode/src/config.ts
@@ -1,7 +1,5 @@
-import * as path from 'path'
-
 const isProduction = process.env.NODE_ENV === 'production'
 export const websocketUrl = isProduction
   ? 'ws://manicode.ai:4242/ws'
   : 'ws://localhost:4242/ws'
-export const projectRoot = path.resolve(__dirname, '..')
+export const projectRoot = process.cwd()
diff --git a/manicode/src/index.ts b/manicode/src/index.ts
index 679acdff1..916685d2f 100644
--- a/manicode/src/index.ts
+++ b/manicode/src/index.ts
@@ -1,6 +1,6 @@
 #!/usr/bin/env node
 
-import { projectRoot, websocketUrl } from './config'
+import { websocketUrl } from './config'
 import { ChatClient } from './chat-client'
 import { ChatStorage } from './chat-storage'
 import { CLI } from './cli'
@@ -12,7 +12,7 @@ async function manicode(userPrompt: string | undefined) {
   // Preload.
   getProjectFileContext()
 
-  const chatStorage = new ChatStorage(projectRoot)
+  const chatStorage = new ChatStorage()
   const client = new ChatClient(websocketUrl, chatStorage)
 
   const cli = new CLI(chatStorage, client)
diff --git a/manicode/src/project-files.ts b/manicode/src/project-files.ts
index aa13032f1..785bfac1d 100644
--- a/manicode/src/project-files.ts
+++ b/manicode/src/project-files.ts
@@ -2,12 +2,11 @@ import fs from 'fs'
 import path from 'path'
 import ts from 'typescript'
 
-import { createFileBlock, ProjectFileContext } from 'common/util/file'
+import { createFileBlock } from 'common/util/file'
 import { FileChanges } from 'common/actions'
 import { filterObject } from 'common/util/object'
 import { scrapeWebPage, parseUrlsFromContent } from './web-scraper'
-
-const projectRoot = path.normalize(path.resolve(__dirname, '../..'))
+import { projectRoot } from './config'
 
 export const applyChanges = (changes: FileChanges) => {
   const changesSuceeded = []
