diff --git a/packages/agent-runtime/src/run-agent-step.ts b/packages/agent-runtime/src/run-agent-step.ts
index 1a17fe593..3f457c6eb 100644
--- a/packages/agent-runtime/src/run-agent-step.ts
+++ b/packages/agent-runtime/src/run-agent-step.ts
@@ -109,10 +109,6 @@ export const runAgentStep = async (
       typeof getMCPToolData,
       'toolNames' | 'mcpServers' | 'writeTo'
     > &
-    ParamsExcluding<
-      typeof getAgentStreamFromTemplate,
-      'agentId' | 'template' | 'onCostCalculated' | 'includeCacheControl'
-    > &
     ParamsExcluding<
       PromptAiSdkFn,
       'messages' | 'model' | 'onCostCalculated' | 'n'
