diff --git a/npm-app/src/utils/terminal.ts b/npm-app/src/utils/terminal.ts
index 3c705cabd..f82f89a6b 100644
--- a/npm-app/src/utils/terminal.ts
+++ b/npm-app/src/utils/terminal.ts
@@ -193,6 +193,14 @@ export const runTerminalCommand = async (
       }, MAX_EXECUTION_TIME)
 
       const dataDisposable = ptyProcess.onData((data: string) => {
+        // Trim first line if it's the prompt identifier
+        if (
+          commandOutput.trim() === '' &&
+          data.trimStart().startsWith(promptIdentifier)
+        ) {
+          data = data.trimStart().slice(promptIdentifier.length)
+        }
+
         const prefix = commandOutput + data
 
         // Skip the first line of the output, because it's the command being printed.
