diff --git a/npm-app/package.json b/npm-app/package.json
index 0dbe7de6d..106474537 100644
--- a/npm-app/package.json
+++ b/npm-app/package.json
@@ -16,8 +16,8 @@
     "prepublishOnly": "yarn build && node clean-package.js",
     "postpublish": "node -e \"const fs=require('fs');fs.copyFileSync('temp.package.json','package.json');fs.unlinkSync('temp.package.json');\"",
     "format": "prettier --write \"**/*.{ts,tsx,json,md}\"",
-    "start-js": "node dist/index.js",
-    "start": "ts-node src/index.ts",
+    "start-js": "node dist/index.js ..",
+    "start": "ts-node src/index.ts ..",
     "start-prod": "NODE_ENV=production ts-node src/index.ts",
     "ssh": "gcloud compute ssh manicode-backend --zone=us-east4-a"
   },
diff --git a/npm-app/src/cli.ts b/npm-app/src/cli.ts
index 50cf8f584..3f4e7af48 100644
--- a/npm-app/src/cli.ts
+++ b/npm-app/src/cli.ts
@@ -115,7 +115,7 @@ export class CLI {
     return {}
   }
 
-  public async handleUserInput(userInput: string) {
+  private async handleUserInput(userInput: string) {
     if (!userInput) return
 
     // Handle "undo" and "redo" commands
diff --git a/npm-app/src/index.ts b/npm-app/src/index.ts
index b695d9988..fe672ba4a 100644
--- a/npm-app/src/index.ts
+++ b/npm-app/src/index.ts
@@ -4,9 +4,11 @@ import { websocketUrl } from './config'
 import { Client } from './client'
 import { ChatStorage } from './chat-storage'
 import { CLI } from './cli'
-import { getProjectFileContext } from './project-files'
+import { getProjectFileContext, initProjectRoot } from './project-files'
+
+async function manicode(projectDir: string | undefined) {
+  initProjectRoot(projectDir)
 
-async function manicode(userPrompt: string | undefined) {
   // Preload.
   getProjectFileContext()
 
@@ -16,15 +18,10 @@ async function manicode(userPrompt: string | undefined) {
   await client.connect()
 
   const cli = new CLI(client, chatStorage)
-
-  if (userPrompt) {
-    await cli.handleUserInput(userPrompt)
-  } else {
-    cli.printInitialPrompt()
-  }
+  cli.printInitialPrompt()
 }
 
 if (require.main === module) {
-  const userPrompt = process.argv[2] || undefined
-  manicode(userPrompt)
+  const projectDir = process.argv[2]
+  manicode(projectDir)
 }
diff --git a/npm-app/src/project-files.ts b/npm-app/src/project-files.ts
index 58b48faeb..34e51b88d 100644
--- a/npm-app/src/project-files.ts
+++ b/npm-app/src/project-files.ts
@@ -6,7 +6,12 @@ import { createFileBlock, ProjectFileContext } from 'common/util/file'
 import { FileChanges } from 'common/actions'
 import { filterObject } from 'common/util/object'
 import { scrapeWebPage, parseUrlsFromContent } from './web-scraper'
-import { projectRoot } from './config'
+
+let projectRoot: string
+
+export function initProjectRoot(dir: string | undefined) {
+  projectRoot = path.resolve(dir || process.cwd())
+}
 
 export const applyChanges = (changes: FileChanges) => {
   const changesSuceeded = []
