diff --git a/backend/src/env.mjs b/backend/src/env.mjs
index ca89b4dd2..30d7f3447 100644
--- a/backend/src/env.mjs
+++ b/backend/src/env.mjs
@@ -8,7 +8,7 @@ if (!process.env.ENVIRONMENT) {
   process.exit(1)
 }
 
-const DOTENV_PATH = process.env.RENDER === 'true' ? '..' : '/etc/secrets'
+const DOTENV_PATH = process.env.RENDER === 'true' ? '/etc/secrets' : '..'
 const path = `${DOTENV_PATH}/.env.${process.env.ENVIRONMENT}`
 console.log(`Using environment: ${process.env.ENVIRONMENT} (path: ${path})`)
 dotenv.config({ path })
diff --git a/web/src/env.mjs b/web/src/env.mjs
index 8fe0fe724..7de9e2de0 100644
--- a/web/src/env.mjs
+++ b/web/src/env.mjs
@@ -8,7 +8,7 @@ if (!process.env.NEXT_PUBLIC_ENVIRONMENT) {
     'NEXT_PUBLIC_ENVIRONMENT is not set, please check `stack.env`'
   )
 }
-const DOTENV_PATH = process.env.RENDER === 'true' ? '..' : '/etc/secrets'
+const DOTENV_PATH = process.env.RENDER === 'true' ? '/etc/secrets' : '..'
 const path = `${DOTENV_PATH}/.env.${process.env.ENVIRONMENT}`
 dotenv.config({ path })
 
