diff --git a/backend/package.json b/backend/package.json
index a3d40b902..467a457fa 100644
--- a/backend/package.json
+++ b/backend/package.json
@@ -24,7 +24,7 @@
     "express": "4.19.2",
     "ignore": "5.3.2",
     "lodash": "*",
-    "openai": "^4.56.0",
+    "openai": "^4.67.1",
     "postgres": "3.4.4",
     "tiktoken": "1.0.15",
     "ts-pattern": "5.3.1",
diff --git a/backend/src/openai-api.ts b/backend/src/openai-api.ts
index 834a6f065..c147dba01 100644
--- a/backend/src/openai-api.ts
+++ b/backend/src/openai-api.ts
@@ -42,6 +42,7 @@ export async function promptOpenAI(
         model,
         messages,
         temperature: 0,
+        store: true
       }),
       timeoutPromise(200000) as Promise<OpenAI.Chat.ChatCompletion>,
     ])
