diff --git a/web/.eslintrc.js b/web/.eslintrc.js
index 53418e768..2d6cdeb58 100644
--- a/web/.eslintrc.js
+++ b/web/.eslintrc.js
@@ -1,4 +1,4 @@
-module.exports = {
+export default {
   parser: '@typescript-eslint/parser',
   extends: [
     'next/core-web-vitals',
diff --git a/web/.prettierrc.js b/web/.prettierrc.js
index efb0e6c92..8843be279 100644
--- a/web/.prettierrc.js
+++ b/web/.prettierrc.js
@@ -1,4 +1,4 @@
-module.exports = {
+export default {
   endOfLine: 'auto',
   semi: false,
   singleQuote: true,
