diff --git a/web/src/app/page.tsx b/web/src/app/page.tsx
index b002a664b..45e99dc54 100644
--- a/web/src/app/page.tsx
+++ b/web/src/app/page.tsx
@@ -131,7 +131,7 @@ const Home = () => {
             </section>
 
             <section className="relative w-full flex items-center justify-center rounded-xl">
-              <div className="relative w-full aspect-[13/9] rounded-xl overflow-hidden shadow-2xl border border-gray-200/20">
+              <div className="relative w-full aspect-[13.5/9] rounded-xl overflow-hidden shadow-2xl border border-gray-200/20">
                 <video
                   className="absolute inset-0 w-full h-full object-cover"
                   autoPlay
