diff --git a/backend/src/request-files-prompt.ts b/backend/src/request-files-prompt.ts
index ce589bde4..554049601 100644
--- a/backend/src/request-files-prompt.ts
+++ b/backend/src/request-files-prompt.ts
@@ -3,7 +3,7 @@ import {
   ProjectFileContext,
   printFileTree,
   createFileBlock,
-} from 'common/src/util/file'
+} from 'common/util/file'
 import { models, promptClaude } from './claude'
 import { debugLog } from './debug'
 
