diff --git a/backend/src/env.mjs b/backend/src/env.mjs
index 74932c004..30d7f3447 100644
--- a/backend/src/env.mjs
+++ b/backend/src/env.mjs
@@ -1,7 +1,6 @@
 import { createEnv } from '@t3-oss/env-core'
 import { z } from 'zod'
 import dotenv from 'dotenv'
-import { logger } from './util/logger'
 
 dotenv.config({ path: '../stack.env' })
 if (!process.env.ENVIRONMENT) {
@@ -11,7 +10,7 @@ if (!process.env.ENVIRONMENT) {
 
 const DOTENV_PATH = process.env.RENDER === 'true' ? '/etc/secrets' : '..'
 const path = `${DOTENV_PATH}/.env.${process.env.ENVIRONMENT}`
-logger.info(`Using environment: ${process.env.ENVIRONMENT} (path: ${path})`)
+console.log(`Using environment: ${process.env.ENVIRONMENT} (path: ${path})`)
 dotenv.config({ path })
 
 export const env = createEnv({
