diff --git a/npm-app/src/tool-handlers.ts b/npm-app/src/tool-handlers.ts
index d09495021..9a7a14a14 100644
--- a/npm-app/src/tool-handlers.ts
+++ b/npm-app/src/tool-handlers.ts
@@ -371,6 +371,7 @@ const isNotACommand = (output: string) => {
     output.includes(': not found') ||
     // Common
     output.includes('syntax error:') ||
+    output.includes('syntax error near unexpected token') ||
     // Linux
     output.includes('Syntax error:') ||
     // Windows
