diff --git a/backend/src/generate-diffs-prompt.ts b/backend/src/generate-diffs-prompt.ts
index d44728dcb..c8fc310ad 100644
--- a/backend/src/generate-diffs-prompt.ts
+++ b/backend/src/generate-diffs-prompt.ts
@@ -15,6 +15,7 @@ export async function generateExpandedFileWithDiffBlocks(
   const diffBlocks = await generateDiffBlocks(
     userId,
     messageHistory,
+    fullResponse,
     filePath,
     oldContent,
     newContent
@@ -52,6 +53,7 @@ export async function generateExpandedFileWithDiffBlocks(
 export async function generateDiffBlocks(
   userId: string,
   messageHistory: Message[],
+  fullResponse: string,
   filePath: string,
   oldContent: string,
   newContent: string
@@ -62,7 +64,16 @@ export async function generateDiffBlocks(
   debugLog('Old content:', oldContent)
   debugLog('New content:', newContent)
 
-  const prompt = `I have a new version of a file with placeholder comments like "// ... existing code ...", and I want to change the old file into the new file without the placeholder comments.
+  const prompt = `
+The following is a conversation with a user leading up to your task:
+  
+<message_history>${messageHistory.map((msg) => `<${msg.role}>${msg.content}</${msg.role}>`).join('\n')}</message_history>
+
+<assistant_message_partial_response>${fullResponse}</assistant_message_partial_response>
+  
+Your task: I have a new version of a file with placeholder comments like "// ... existing code ..." or "# ... existing code ...", and I want to change the old file into the expanded new file without the placeholder comments.
+
+Consider the intent of the user: if only one function or code block is shown, don't delete everything else that was not shown.
   
 I need to generate <search> and <replace> blocks to represent the exact line-by-line differences so I can string replace the old content to the new content.
 
@@ -446,7 +457,8 @@ Your Response:`
     console.log('No diff blocks generated', filePath)
     debugLog('Warning: No diff blocks generated', filePath)
   } else if (diffBlocksThatDidntMatch.length > 0) {
-    const newPrompt = `The assistant failed to find a match for the following changes in the file ${filePath}. Please help the assistant understand what the changes should be.
+    const newPrompt =
+      `The assistant failed to find a match for the following changes in the file ${filePath}. Please help the assistant understand what the changes should be.
 
 Here is the previous prompt that the assistant was given:
 <prompt>
@@ -468,7 +480,7 @@ You should:
 
 Please make sure to end your response with the following string:
 ${STOP_MARKER}
-`
+`.trim()
     console.log('Trying a second prompt for getDiffBlocks', filePath)
     debugLog('Trying a second prompt for getDiffBlocks', filePath)
     const response = await promptOpenAIWithContinuation(
diff --git a/backend/src/generate-diffs-via-expansion.ts b/backend/src/generate-diffs-via-expansion.ts
index 1a00d39b4..6f0e1b71c 100644
--- a/backend/src/generate-diffs-via-expansion.ts
+++ b/backend/src/generate-diffs-via-expansion.ts
@@ -11,9 +11,16 @@ export async function expandNewContent(
   messageHistory: Message[],
   fullResponse: string
 ): Promise<string> {
-  const prompt = `<assistant_message_context>${fullResponse}</assistant_message_context>
+  const prompt = `
+The following is a conversation with a user leading up to your task:
+  
+<message_history>${messageHistory.map((msg) => `<${msg.role}>${msg.content}</${msg.role}>`).join('\n')}</message_history>
 
-You are an expert programmer tasked with expanding a shortened version of a file into its full content. The shortened version uses comments like "// ... existing code ..." or "// ... rest of the function ..." or "// keep existing code ..." to indicate unchanged sections. Your task is to replace these comments with the actual code from the old version of the file.
+<assistant_message_partial_response>${fullResponse}</assistant_message_partial_response>
+
+Your task: You are an expert programmer tasked with expanding a shortened version of a file into its full content. The shortened file to be expanded will be provided at the end of this message. This shortened version uses comments like "// ... existing code ..." or "# ... rest of the function ..." or "// keep existing code ..." to indicate unchanged sections. Your task is to replace these comments with the actual code from the old version of the file.
+
+Consider the intent of the user: if only one function or code block is shown, don't delete everything else that was not shown.
 
 Your response should follow the following format:
 A. Please discuss the changes in the new file content compared to the old file content in a <discussion> block.
@@ -27,7 +34,7 @@ This requires you to compose the resulting file with exact lines from the old fi
 
 Output the full content of the new file within a <file> block, using the provided file path as an attribute.
 
-If comments are being added that describe the change that is being made, such as "// Add this import" or "// Add this function" or "// Update this log", then please ommit these lines from the new file.
+If comments are being added that describe the change that is being made, such as "# Add this import" or "// Add this function" or "// Update this log", then please ommit these lines from the new file.
 
 Here are four examples to illustrate the task:
 
@@ -662,7 +669,8 @@ Old file content:
 ${createFileBlock(filePath, oldContent)}
 
 New file content (with placeholders):
-${createFileBlock(filePath, newContent)}`
+${createFileBlock(filePath, newContent)}
+`.trim()
   const expandedContentResponse = await promptOpenAI(
     userId,
     [
@@ -674,11 +682,7 @@ ${createFileBlock(filePath, newContent)}`
     'gpt-4o-2024-08-06'
   )
 
-  debugLog(
-    'New file (unexpanded) for filePath',
-    filePath,
-    newContent
-  )
+  debugLog('New file (unexpanded) for filePath', filePath, newContent)
   debugLog(
     'Expanded content response for filePath',
     filePath,
