diff --git a/backend/src/index.ts b/backend/src/index.ts
index 1f1848b81..15533c8d0 100644
--- a/backend/src/index.ts
+++ b/backend/src/index.ts
@@ -170,15 +170,4 @@ process.on('uncaughtException', (err, origin) => {
     },
     'uncaught exception detected',
   )
-
-  server.close(() => {
-    process.exit(1)
-  })
-
-  // If a graceful shutdown is not achieved after 1 second,
-  // shut down the process completely
-  setTimeout(() => {
-    process.abort() // exit immediately and generate a core dump file
-  }, 1000).unref()
-  process.exit(1)
 })
