diff --git a/packages/code-map/parse.ts b/packages/code-map/parse.ts
index fa833a782..e5c5198cb 100644
--- a/packages/code-map/parse.ts
+++ b/packages/code-map/parse.ts
@@ -89,6 +89,11 @@ export async function getFileTokenScores(
         continue
       }
 
+      // Skip token names in default objects, e.g. toString, hasOwnProperty
+      if (call in {}) {
+        continue
+      }
+
       if (!tokenCallers[definingFile]) {
         tokenCallers[definingFile] = {}
       }
