diff --git a/packages/internal/src/env.ts b/packages/internal/src/env.ts
index d99483322..3c3f60ce8 100644
--- a/packages/internal/src/env.ts
+++ b/packages/internal/src/env.ts
@@ -17,6 +17,7 @@ if (isCI) {
   ensureEnvDefault('OPENAI_API_KEY', 'test')
   ensureEnvDefault('ANTHROPIC_API_KEY', 'test')
   ensureEnvDefault('FIREWORKS_API_KEY', 'test')
+  ensureEnvDefault('CANOPYWAVE_API_KEY', 'test')
   ensureEnvDefault('LINKUP_API_KEY', 'test')
   ensureEnvDefault('GRAVITY_API_KEY', 'test')
   ensureEnvDefault('IPINFO_TOKEN', 'test')
