diff --git a/backend/src/run-agent-step.ts b/backend/src/run-agent-step.ts
index f97095062..a2629ce29 100644
--- a/backend/src/run-agent-step.ts
+++ b/backend/src/run-agent-step.ts
@@ -402,6 +402,7 @@ export const runAgentStep = async (
     toolResults: newToolResults,
     state,
     fullResponse: fullResponseAfterStream,
+    fullResponseChunks,
   } = await processStreamWithTools({
     stream,
     ws,
@@ -470,6 +471,7 @@ export const runAgentStep = async (
       agentId: agentState.agentId,
       prompt,
       fullResponse,
+      fullResponseChunks,
       toolCalls,
       toolResults,
       agentContext: newAgentContext,
diff --git a/backend/src/tools/stream-parser.ts b/backend/src/tools/stream-parser.ts
index 5f0dc80d8..e0cce2652 100644
--- a/backend/src/tools/stream-parser.ts
+++ b/backend/src/tools/stream-parser.ts
@@ -55,7 +55,7 @@ export async function processStreamWithTools<T extends string>(options: {
     agentState,
     onResponseChunk,
   } = options
-  let fullResponse = options.fullResponse
+  const fullResponseChunks: string[] = [options.fullResponse]
 
   const messages = [...options.messages]
 
@@ -105,7 +105,7 @@ export async function processStreamWithTools<T extends string>(options: {
           agentStepId,
           clientSessionId,
           userInputId,
-          fullResponse,
+          fullResponse: fullResponseChunks.join(''),
           onResponseChunk,
           state,
           userId,
@@ -135,19 +135,25 @@ export async function processStreamWithTools<T extends string>(options: {
 
   for await (const chunk of streamWithTags) {
     onResponseChunk(chunk)
-    fullResponse += chunk
+    fullResponseChunks.push(chunk)
   }
 
   state.messages = buildArray<CodebuffMessage>([
     ...expireMessages(state.messages, 'agentStep'),
-    fullResponse && {
+    fullResponseChunks.length > 0 && {
       role: 'assistant' as const,
-      content: fullResponse,
+      content: fullResponseChunks.join(''),
     },
   ])
 
   resolveStreamDonePromise()
   await previousToolCallFinished
 
-  return { toolCalls, toolResults, state, fullResponse }
+  return {
+    toolCalls,
+    toolResults,
+    state,
+    fullResponse: fullResponseChunks.join(''),
+    fullResponseChunks,
+  }
 }
diff --git a/backend/src/xml-stream-parser.ts b/backend/src/xml-stream-parser.ts
index 2ff4b9464..c7b636c77 100644
--- a/backend/src/xml-stream-parser.ts
+++ b/backend/src/xml-stream-parser.ts
@@ -51,15 +51,12 @@ export async function* processStreamWithTags(
 
   function extractToolCalls(): string[] {
     const matches: string[] = []
-    let lastIndex = -1
+    let lastIndex = 0
     for (const match of buffer.matchAll(toolExtractionPattern)) {
       lastIndex = match.index + match[0].length
       matches.push(match[1])
     }
 
-    if (lastIndex === -1) {
-      lastIndex = 0
-    }
     buffer = removeProcessedToolCalls(buffer.slice(lastIndex))
     return matches
   }
