diff --git a/common/src/types/agent-template.ts b/common/src/types/agent-template.ts
index 59839120b..60c236dfa 100644
--- a/common/src/types/agent-template.ts
+++ b/common/src/types/agent-template.ts
@@ -12,7 +12,7 @@ export type AgentTemplate<
   id: AgentTemplateType
   displayName: string
   model: Model
-  reasoningOptions: OpenRouterProviderOptions['reasoning']
+  reasoningOptions?: OpenRouterProviderOptions['reasoning']
 
   toolNames: ToolName[]
   spawnableAgents: AgentTemplateType[]
