diff --git a/.agents/researcher.ts b/.agents/researcher.ts
index 794754646..23ac43b5b 100644
--- a/.agents/researcher.ts
+++ b/.agents/researcher.ts
@@ -31,7 +31,7 @@ const config: AgentConfig = {
   handleSteps: function* ({ agentState, prompt, params }) {
     yield {
       toolName: 'web_search',
-      args: { prompt },
+      args: { query: prompt },
     }
     yield 'STEP_ALL'
   },
diff --git a/backend/src/templates/agents/researcher.ts b/backend/src/templates/agents/researcher.ts
index 1f6d95f1e..2bc2ca5c0 100644
--- a/backend/src/templates/agents/researcher.ts
+++ b/backend/src/templates/agents/researcher.ts
@@ -48,7 +48,7 @@ Always end your response with the end_turn tool.\\n\\n` +
   handleSteps: function* ({ agentState, prompt, params }) {
     yield {
       toolName: 'web_search',
-      args: { prompt },
+      args: { query: prompt },
     }
     yield 'STEP_ALL'
   },
