diff --git a/backend/src/tools/constants.ts b/backend/src/tools/constants.ts
index 2005793cb..4dbc28308 100644
--- a/backend/src/tools/constants.ts
+++ b/backend/src/tools/constants.ts
@@ -107,14 +107,14 @@ type PresentOrAbsent<K extends PropertyKey, V> =
 
 export type CodebuffToolHandlerFunction<T extends NonWIPTool = NonWIPTool> = (
   params: {
-    previousToolCallResult: Promise<any>
+    previousToolCallFinished: Promise<void>
     toolCall: CodebuffToolCall<T>
     state: { [K in string]?: any }
   } & PresentOrAbsent<
     'requestClientToolCall',
     (toolCall: ClientToolCall<T>) => Promise<string>
   >
-) => Promise<{ result: string; state: Record<string, any> }>
+) => { result: Promise<string>; state: Record<string, any> }
 
 /**
  * Each value in this record that:
diff --git a/backend/src/tools/handlers/add-subgoal.ts b/backend/src/tools/handlers/add-subgoal.ts
index 82498922c..1b448cdb0 100644
--- a/backend/src/tools/handlers/add-subgoal.ts
+++ b/backend/src/tools/handlers/add-subgoal.ts
@@ -8,20 +8,19 @@ export type Subgoal = {
   logs: string[]
 }
 
-export const handleAddSubgoal = (async (params: {
-  previousToolCallResult: Promise<any>
+export const handleAddSubgoal = ((params: {
+  previousToolCallFinished: Promise<void>
   toolCall: CodebuffToolCall<'add_subgoal'>
   state: { agentContext?: Record<string, Subgoal> }
-}): Promise<{
-  result: string
+}): {
+  result: Promise<string>
   state: { agentContext: Record<string, Subgoal> }
-}> => {
-  const { previousToolCallResult, toolCall, state } = params
+} => {
+  const { previousToolCallFinished, toolCall, state } = params
   const { agentContext } = state
 
-  await previousToolCallResult
   return {
-    result: 'Successfully added subgoal.',
+    result: previousToolCallFinished.then(() => 'Successfully added subgoal'),
     state: {
       agentContext: {
         ...agentContext,
diff --git a/backend/src/tools/handlers/browser-logs.ts b/backend/src/tools/handlers/browser-logs.ts
index 1d5ee9ca4..e392f4258 100644
--- a/backend/src/tools/handlers/browser-logs.ts
+++ b/backend/src/tools/handlers/browser-logs.ts
@@ -4,15 +4,19 @@ import {
   CodebuffToolHandlerFunction,
 } from '../constants'
 
-export const handleBrowserLogs = (async (params: {
-  previousToolCallResult: Promise<any>
+export const handleBrowserLogs = ((params: {
+  previousToolCallFinished: Promise<void>
   toolCall: CodebuffToolCall<'browser_logs'>
   requestClientToolCall: (
     toolCall: ClientToolCall<'browser_logs'>
   ) => Promise<string>
-}): Promise<{ result: string; state: {} }> => {
-  const { previousToolCallResult, toolCall, requestClientToolCall } = params
+}): { result: Promise<string>; state: {} } => {
+  const { previousToolCallFinished, toolCall, requestClientToolCall } = params
 
-  await previousToolCallResult
-  return { result: await requestClientToolCall(toolCall), state: {} }
+  return {
+    result: previousToolCallFinished.then(() =>
+      requestClientToolCall(toolCall)
+    ),
+    state: {},
+  }
 }) satisfies CodebuffToolHandlerFunction<'browser_logs'>
diff --git a/backend/src/tools/handlers/code-search.ts b/backend/src/tools/handlers/code-search.ts
index 9b0903506..3a5340478 100644
--- a/backend/src/tools/handlers/code-search.ts
+++ b/backend/src/tools/handlers/code-search.ts
@@ -4,15 +4,19 @@ import {
   CodebuffToolHandlerFunction,
 } from '../constants'
 
-export const handleCodeSearch = (async (params: {
-  previousToolCallResult: Promise<any>
+export const handleCodeSearch = ((params: {
+  previousToolCallFinished: Promise<void>
   toolCall: CodebuffToolCall<'code_search'>
   requestClientToolCall: (
     toolCall: ClientToolCall<'code_search'>
   ) => Promise<string>
-}): Promise<{ result: string; state: {} }> => {
-  const { previousToolCallResult, toolCall, requestClientToolCall } = params
+}): { result: Promise<string>; state: {} } => {
+  const { previousToolCallFinished, toolCall, requestClientToolCall } = params
 
-  await previousToolCallResult
-  return { result: await requestClientToolCall(toolCall), state: {} }
+  return {
+    result: previousToolCallFinished.then(() =>
+      requestClientToolCall(toolCall)
+    ),
+    state: {},
+  }
 }) satisfies CodebuffToolHandlerFunction<'code_search'>
diff --git a/backend/src/tools/handlers/end-turn.ts b/backend/src/tools/handlers/end-turn.ts
index c34c606c9..de7a6d4ed 100644
--- a/backend/src/tools/handlers/end-turn.ts
+++ b/backend/src/tools/handlers/end-turn.ts
@@ -1,9 +1,8 @@
 import { CodebuffToolCall, CodebuffToolHandlerFunction } from '../constants'
 
-export const handleEndTurn = (async (params: {
-  previousToolCallResult: Promise<any>
+export const handleEndTurn = ((params: {
+  previousToolCallFinished: Promise<any>
   toolCall: CodebuffToolCall<'end_turn'>
-}): Promise<{ result: ''; state: {} }> => {
-  await params.previousToolCallResult
-  return { result: '', state: {} }
+}): { result: Promise<string>; state: {} } => {
+  return { result: params.previousToolCallFinished.then(() => ''), state: {} }
 }) satisfies CodebuffToolHandlerFunction<'end_turn'>
diff --git a/backend/src/tools/handlers/run-file-change-hooks.ts b/backend/src/tools/handlers/run-file-change-hooks.ts
index abebed079..d8fd15792 100644
--- a/backend/src/tools/handlers/run-file-change-hooks.ts
+++ b/backend/src/tools/handlers/run-file-change-hooks.ts
@@ -4,15 +4,19 @@ import {
   CodebuffToolHandlerFunction,
 } from '../constants'
 
-export const handleRunFileChangeHooks = (async (params: {
-  previousToolCallResult: Promise<any>
+export const handleRunFileChangeHooks = ((params: {
+  previousToolCallFinished: Promise<void>
   toolCall: CodebuffToolCall<'run_file_change_hooks'>
   requestClientToolCall: (
     toolCall: ClientToolCall<'run_file_change_hooks'>
   ) => Promise<string>
-}): Promise<{ result: string; state: {} }> => {
-  const { previousToolCallResult, toolCall, requestClientToolCall } = params
+}): { result: Promise<string>; state: {} } => {
+  const { previousToolCallFinished, toolCall, requestClientToolCall } = params
 
-  await previousToolCallResult
-  return { result: await requestClientToolCall(toolCall), state: {} }
+  return {
+    result: previousToolCallFinished.then(() =>
+      requestClientToolCall(toolCall)
+    ),
+    state: {},
+  }
 }) satisfies CodebuffToolHandlerFunction<'run_file_change_hooks'>
diff --git a/backend/src/tools/handlers/run-terminal-command.ts b/backend/src/tools/handlers/run-terminal-command.ts
index 5b38a81ce..35a52df5b 100644
--- a/backend/src/tools/handlers/run-terminal-command.ts
+++ b/backend/src/tools/handlers/run-terminal-command.ts
@@ -4,16 +4,15 @@ import {
   CodebuffToolHandlerFunction,
 } from '../constants'
 
-export const handleRunTerminalCommand = (async (params: {
-  previousToolCallResult: Promise<any>
+export const handleRunTerminalCommand = ((params: {
+  previousToolCallFinished: Promise<void>
   toolCall: CodebuffToolCall<'run_terminal_command'>
   requestClientToolCall: (
     toolCall: ClientToolCall<'run_terminal_command'>
   ) => Promise<string>
-}): Promise<{ result: string; state: {} }> => {
-  const { previousToolCallResult, toolCall, requestClientToolCall } = params
+}): { result: Promise<string>; state: {} } => {
+  const { previousToolCallFinished, toolCall, requestClientToolCall } = params
 
-  await previousToolCallResult
   const clientToolCall: ClientToolCall<'run_terminal_command'> = {
     toolName: 'run_terminal_command',
     toolCallId: toolCall.toolCallId,
@@ -25,5 +24,10 @@ export const handleRunTerminalCommand = (async (params: {
       cwd: toolCall.args.cwd,
     },
   }
-  return { result: await requestClientToolCall(clientToolCall), state: {} }
+  return {
+    result: previousToolCallFinished.then(() =>
+      requestClientToolCall(clientToolCall)
+    ),
+    state: {},
+  }
 }) satisfies CodebuffToolHandlerFunction<'run_terminal_command'>
diff --git a/backend/src/tools/handlers/update-subgoal.ts b/backend/src/tools/handlers/update-subgoal.ts
index 8f491f246..a1bccdfe1 100644
--- a/backend/src/tools/handlers/update-subgoal.ts
+++ b/backend/src/tools/handlers/update-subgoal.ts
@@ -1,19 +1,17 @@
 import { CodebuffToolCall, CodebuffToolHandlerFunction } from '../constants'
 import { Subgoal } from './add-subgoal'
 
-export const handleUpdateSubgoal = (async (params: {
-  previousToolCallResult: Promise<any>
+export const handleUpdateSubgoal = ((params: {
+  previousToolCallFinished: Promise<void>
   toolCall: CodebuffToolCall<'update_subgoal'>
   state: { agentContext?: Record<string, Subgoal> }
-}): Promise<{
-  result: string
+}): {
+  result: Promise<string>
   state: { agentContext: Record<string, Subgoal> }
-}> => {
-  const { previousToolCallResult, toolCall, state } = params
+} => {
+  const { previousToolCallFinished, toolCall, state } = params
   const { agentContext } = state
 
-  await previousToolCallResult
-
   let messages: string[] = []
   const newAgentContext = { ...agentContext }
   if (!newAgentContext[toolCall.args.id]) {
@@ -38,7 +36,7 @@ export const handleUpdateSubgoal = (async (params: {
   }
   messages.push('Successfully updated subgoal.')
   return {
-    result: messages.join('\n\n'),
+    result: previousToolCallFinished.then(() => messages.join('\n\n')),
     state: { agentContext: newAgentContext },
   }
 }) satisfies CodebuffToolHandlerFunction<'update_subgoal'>
