diff --git a/npm-app/src/project-files.ts b/npm-app/src/project-files.ts
index 52ce47455..3493e06b4 100644
--- a/npm-app/src/project-files.ts
+++ b/npm-app/src/project-files.ts
@@ -199,8 +199,13 @@ export function getFiles(filePaths: string[]) {
       result[relativePath] = '[FILE_OUTSIDE_PROJECT]'
       continue
     }
-    if (ig.ignores(relativePath)) {
-      result[relativePath] = '[FILE_IGNORED]'
+    try {
+      if (ig.ignores(relativePath)) {
+        result[relativePath] = '[FILE_IGNORED]'
+        continue
+      }
+    } catch (error) {
+      result[relativePath] = '[ERROR_LOADING_FILE]'
       continue
     }
     try {
