diff --git a/web/next.config.mjs b/web/next.config.mjs
index de954c77a..fce0f5658 100644
--- a/web/next.config.mjs
+++ b/web/next.config.mjs
@@ -153,6 +153,11 @@ const nextConfig = {
           'https://github.com/CodebuffAI/codebuff-community/releases',
         permanent: false,
       },
+      {
+        source: '/b/:hash',
+        destination: 'https://go.trybeluga.ai/:hash',
+        permanent: false,
+      },
     ]
   },
   images: {
