diff --git a/backend/src/index.ts b/backend/src/index.ts
index 75cdf0eb1..ea4f53150 100644
--- a/backend/src/index.ts
+++ b/backend/src/index.ts
@@ -41,3 +41,17 @@ webSocketListen(server, '/ws')
 process.on('SIGINT', () => {
   process.exit()
 })
+
+process.on('uncaughtException', (err) => {
+  logger.fatal(err, 'uncaught exception detected')
+  server.close(() => {
+    process.exit(1)
+  })
+
+  // If a graceful shutdown is not achieved after 1 second,
+  // shut down the process completely
+  setTimeout(() => {
+    process.abort() // exit immediately and generate a core dump file
+  }, 1000).unref()
+  process.exit(1)
+})
