diff --git a/.agents/editor/editor.ts b/.agents/editor/editor.ts
index 80d4c8ed7..5c6377037 100644
--- a/.agents/editor/editor.ts
+++ b/.agents/editor/editor.ts
@@ -8,7 +8,7 @@ import {
 const editor: SecretAgentDefinition = {
   id: 'editor',
   publisher,
-  model: 'openai/gpt-5-chat',
+  model: 'anthropic/claude-sonnet-4.5',
   displayName: 'Code Editor',
   spawnerPrompt:
     'Expert code editor with access to tools to find and edit files, run terminal commands, and search the web. Can handle small to medium sized tasks, or work off of a plan for more complex tasks. For easy tasks, you can spawn this agent directly rather than invoking a scout or planner first.',
diff --git a/.agents/orchestrator/orchestrator.ts b/.agents/orchestrator/orchestrator.ts
index db170da34..cc3df69ac 100644
--- a/.agents/orchestrator/orchestrator.ts
+++ b/.agents/orchestrator/orchestrator.ts
@@ -31,9 +31,11 @@ const definition: SecretAgentDefinition = {
   toolNames: ['spawn_agents', 'read_files', 'str_replace', 'write_file'],
   spawnableAgents: [
     'read-only-commander',
-    'researcher-grok-4-fast',
-    'decomposing-planner-lite',
-    'editor-lite',
+    'researcher-file-explorer',
+    'researcher-web',
+    'researcher-docs',
+    'decomposing-planner',
+    'editor',
     'reviewer-lite',
     'context-pruner',
   ],
diff --git a/evals/subagents/eval-planner.ts b/evals/subagents/eval-planner.ts
index fd2c13cad..f339cfb0b 100644
--- a/evals/subagents/eval-planner.ts
+++ b/evals/subagents/eval-planner.ts
@@ -124,7 +124,20 @@ Evaluate how well the implementation plan matches the real commit changes. Consi
     },
   })
   if (judgeResult.output.type !== 'structuredOutput') {
-    throw new Error('Error running judge agent')
+    console.log(
+      'Error running judge agent -- not structured output',
+      JSON.stringify(judgeResult.output, null, 2),
+    )
+    // throw new Error('Error running judge agent')
+    return {
+      judgingResults: {
+        reasoning: 'Error running judge agent -- not structured output',
+        pros: '',
+        cons: '',
+        overallScore: 0,
+      },
+      agentOutput: outputString,
+    }
   }
   const { output: judgeOutput } = judgeResult
   const judgingResults = (judgeOutput.value ?? {}) as {
