diff --git a/packages/internal/src/openrouter-ai-sdk/types/openrouter-chat-settings.ts b/packages/internal/src/openrouter-ai-sdk/types/openrouter-chat-settings.ts
index 4bc7bb3df..ecdc9810f 100644
--- a/packages/internal/src/openrouter-ai-sdk/types/openrouter-chat-settings.ts
+++ b/packages/internal/src/openrouter-ai-sdk/types/openrouter-chat-settings.ts
@@ -15,7 +15,7 @@ The exact effect will vary per model, but values between -1 and 1 should
 decrease or increase likelihood of selection; values like -100 or 100
 should result in a ban or exclusive selection of the relevant token.
 
-As an example, you can pass {"50256": -100} to prevent the <|endoftext|>
+As an example, you can pass {"50256": -100} to prevent the <|{'endoftext'}|>
 token from being generated.
 */
   logitBias?: Record<number, number>
