diff --git a/npm-app/src/client.ts b/npm-app/src/client.ts
index 1f5cd36f1..268afa8df 100644
--- a/npm-app/src/client.ts
+++ b/npm-app/src/client.ts
@@ -151,6 +151,9 @@ export class Client {
             'Content-Type': 'application/json',
             Cookie: `next-auth.session-token=${this.user.authToken}`,
           },
+          body: JSON.stringify({
+            authToken: this.user.authToken,
+          }),
         }
       )
 
@@ -184,7 +187,11 @@ export class Client {
             'Content-Type': 'application/json',
             Cookie: `next-auth.session-token=${this.user.authToken}`,
           },
-          body: JSON.stringify({ keyType, apiKey }),
+          body: JSON.stringify({
+            keyType,
+            apiKey,
+            authToken: this.user.authToken,
+          }),
         }
       )
 
@@ -831,6 +838,6 @@ export class Client {
       fileContext,
     })
 
-    this.fetchStoredApiKeyTypes()
+    await this.fetchStoredApiKeyTypes()
   }
 }
diff --git a/web/src/app/api/api-keys/route.ts b/web/src/app/api/api-keys/route.ts
index ca67a6e49..0d57ccaca 100644
--- a/web/src/app/api/api-keys/route.ts
+++ b/web/src/app/api/api-keys/route.ts
@@ -1,31 +1,40 @@
 import db from 'common/db'
+import * as schema from 'common/db/schema'
 import { ApiKeyType } from 'common/src/api-keys/constants'
 import { encryptAndStoreApiKey } from 'common/src/api-keys/crypto'
 import { apiKeyTypeEnum, encryptedApiKeys } from 'common/src/db/schema'
 import { eq } from 'drizzle-orm'
-import { getServerSession } from 'next-auth'
 import { NextRequest, NextResponse } from 'next/server'
 import { z } from 'zod'
 
-import { authOptions } from '@/app/api/auth/[...nextauth]/auth-options'
 import { logger } from '@/util/logger'
 
-// Define the schema for the request body
-const ApiKeySchema = z.object({
-  keyType: z.enum(apiKeyTypeEnum.enumValues),
-  apiKey: z.string().min(1, 'API key cannot be empty'),
-})
+export async function GET(request: NextRequest) {
+  const reqJson = await request.json()
+  const parsedJson = z
+    .object({
+      authToken: z.string(),
+    })
+    .safeParse(reqJson)
 
-export async function GET(req: NextRequest) {
-  const session = await getServerSession(authOptions)
-  console.log({ session })
+  if (!parsedJson.success) {
+    return NextResponse.json({ error: 'Invalid request body' }, { status: 400 })
+  }
+
+  const { authToken } = parsedJson.data
+  const user = await db.query.session.findFirst({
+    where: eq(schema.session.sessionToken, authToken),
+    columns: {
+      userId: true,
+    },
+  })
+
+  const userId = user?.userId
 
-  if (!session?.user?.id) {
+  if (!userId) {
     return NextResponse.json({ error: 'Unauthorized' }, { status: 401 })
   }
 
-  const userId = session.user.id
-
   try {
     const storedKeys = await db
       .select({ type: encryptedApiKeys.type })
@@ -48,34 +57,33 @@ export async function GET(req: NextRequest) {
   }
 }
 
-export async function POST(req: NextRequest) {
-  const session = await getServerSession(authOptions)
+export async function POST(request: NextRequest) {
+  const reqJson = await request.json()
+  const parsedJson = z
+    .object({
+      keyType: z.enum(apiKeyTypeEnum.enumValues),
+      apiKey: z.string().min(1, 'API key cannot be empty'),
+      authToken: z.string(),
+    })
+    .safeParse(reqJson)
 
-  if (!session?.user?.id) {
-    return NextResponse.json({ error: 'Unauthorized' }, { status: 401 })
+  if (!parsedJson.success) {
+    return NextResponse.json({ error: 'Invalid request body' }, { status: 400 })
   }
 
-  const userId = session.user.id
+  const { keyType, apiKey, authToken } = parsedJson.data
+  const user = await db.query.session.findFirst({
+    where: eq(schema.session.sessionToken, authToken),
+    columns: {
+      userId: true,
+    },
+  })
 
-  let reqBody: z.infer<typeof ApiKeySchema>
-  try {
-    reqBody = await req.json()
-    ApiKeySchema.parse(reqBody) // Validate the request body
-  } catch (error) {
-    logger.error({ error, userId }, 'Invalid request body for adding API key')
-    if (error instanceof z.ZodError) {
-      return NextResponse.json(
-        { error: 'Invalid request body', details: error.errors },
-        { status: 400 }
-      )
-    }
-    return NextResponse.json(
-      { error: 'Failed to parse request body' },
-      { status: 400 }
-    )
-  }
+  const userId = user?.userId
 
-  const { keyType, apiKey } = reqBody
+  if (!userId) {
+    return NextResponse.json({ error: 'Unauthorized' }, { status: 401 })
+  }
 
   try {
     await encryptAndStoreApiKey(userId, keyType, apiKey)
