diff --git a/npm-app/src/dev-process-manager.ts b/npm-app/src/dev-process-manager.ts
index bcd161c4e..954829291 100644
--- a/npm-app/src/dev-process-manager.ts
+++ b/npm-app/src/dev-process-manager.ts
@@ -59,7 +59,7 @@ export function startDevProcesses(
         stderrFile,
       },
       (result) => {
-        if (has(result, 'processId')) {
+        if (has(result[0].value, 'processId')) {
           console.log(yellow(`- ${name}: ${command}`))
         } else {
           console.log(yellow(`- ${name}: ${command} — failed to start`))
