diff --git a/backend/src/tools.ts b/backend/src/tools.ts
index 4bce8d13a..688c7daff 100644
--- a/backend/src/tools.ts
+++ b/backend/src/tools.ts
@@ -4,7 +4,7 @@ export const getTools = (): Tool[] => {
   return [
     {
       name: 'update_file_context',
-      description: `Updates the context with a new set of files. Another assistant will try to choose files that will be helpful based on the message history. You should also provide a prompt that describes in natural language what files to add or remove from the context.`,
+      description: `Updates the context with a the set of existing files you want to read. Another assistant will try to choose files that will be helpful based on the message history. You should also provide a prompt that describes in natural language what files to add or remove from the context. Do not use this tool to create a new file, only to read existing files.`,
       input_schema: {
         type: 'object',
         properties: {
