diff --git a/web/next.config.mjs b/web/next.config.mjs
index 7c5802a68..b05a1eadf 100644
--- a/web/next.config.mjs
+++ b/web/next.config.mjs
@@ -58,6 +58,11 @@ const nextConfig = {
         permanent: false,
         destination: `${env.NEXT_PUBLIC_APP_URL}/:path*`,
       },
+      {
+        source: '/discord',
+        destination: 'https://discord.gg/mcWTGjgTj3',
+        permanent: false,
+      },
     ]
   },
   images: {
