diff --git a/backend/src/websockets/middleware.ts b/backend/src/websockets/middleware.ts
index 8cb7fc5fc..4222256f8 100644
--- a/backend/src/websockets/middleware.ts
+++ b/backend/src/websockets/middleware.ts
@@ -45,6 +45,10 @@ export class WebSocketMiddleware {
       const res = await middleware(action, clientSessionId, ws)
       if (res) {
         console.error('Middleware execution halted:', res)
+        sendAction(ws, {
+          type: 'error',
+          message: res.message,
+        })
         return false
       }
     }
@@ -118,6 +122,14 @@ protec.use(async (action, _clientSessionId, ws) => {
         fingerprintId: P.string,
       },
       async ({ fingerprintId }) => {
+        // Create a new fingerprint if it doesn't exist
+        await db
+          .insert(schema.fingerprint)
+          .values({
+            id: fingerprintId,
+          })
+          .onConflictDoNothing()
+
         const quotas = await db
           .select({
             fingerprintId: schema.fingerprint.id,
diff --git a/backend/src/websockets/websocket-action.ts b/backend/src/websockets/websocket-action.ts
index 4a6358bcf..926ea5463 100644
--- a/backend/src/websockets/websocket-action.ts
+++ b/backend/src/websockets/websocket-action.ts
@@ -348,7 +348,7 @@ export const onWebsocketAction = async (
 }
 
 subscribeToAction('user-input', protec.run(onUserInput))
-subscribeToAction('init', () => onInit)
+subscribeToAction('init', () => protec.run(onInit))
 
 subscribeToAction('clear-auth-token', onClearAuthTokenRequest)
 subscribeToAction('login-code-request', onLoginCodeRequest)
diff --git a/common/src/actions.ts b/common/src/actions.ts
index c994e8750..deae9d47c 100644
--- a/common/src/actions.ts
+++ b/common/src/actions.ts
@@ -159,5 +159,9 @@ export const SERVER_ACTION_SCHEMA = z.discriminatedUnion('type', [
     type: z.literal('quota-exceeded'),
     nextQuotaReset: z.date(),
   }),
+  z.object({
+    type: z.literal('error'),
+    message: z.string(),
+  }),
 ])
 export type ServerAction = z.infer<typeof SERVER_ACTION_SCHEMA>
