diff --git a/npm-app/src/menu.ts b/npm-app/src/menu.ts
index 3409438e4..74bf9a15f 100644
--- a/npm-app/src/menu.ts
+++ b/npm-app/src/menu.ts
@@ -40,7 +40,7 @@ export function displayGreeting(costMode: CostMode, username: string | null) {
   }
 
   console.log(
-    `\nWelcome${username ? ` back ${username}` : ''}! What would you like to do?`
+    `\nWelcome${username ? ` back ${username}` : ''}! What would you like to do today? 🥧`
   )
 }
 
