diff --git a/backend/src/run-programmatic-step.ts b/backend/src/run-programmatic-step.ts
index 57a8c46f4..d7789a66f 100644
--- a/backend/src/run-programmatic-step.ts
+++ b/backend/src/run-programmatic-step.ts
@@ -232,6 +232,7 @@ export async function runProgrammaticStep(
         state,
         userId,
         autoInsertEndStepParam: true,
+        validateInputSchema: false,
       })
 
       // TODO: Remove messages from state and always use agentState.messageHistory.
diff --git a/backend/src/tools/tool-executor.ts b/backend/src/tools/tool-executor.ts
index c35d8e00b..8fb53f219 100644
--- a/backend/src/tools/tool-executor.ts
+++ b/backend/src/tools/tool-executor.ts
@@ -126,6 +126,7 @@ export interface ExecuteToolCallParams<T extends string = ToolName> {
   state: Record<string, any>
   userId: string | undefined
   autoInsertEndStepParam?: boolean
+  validateInputSchema?: boolean
 }
 
 export function executeToolCall<T extends ToolName>({
@@ -145,15 +146,16 @@ export function executeToolCall<T extends ToolName>({
   state,
   userId,
   autoInsertEndStepParam = false,
+  validateInputSchema = true
 }: ExecuteToolCallParams<T>): Promise<void> {
-  const toolCall: CodebuffToolCall<T> | ToolCallError = parseRawToolCall<T>(
+  const toolCall: CodebuffToolCall<T> | ToolCallError = validateInputSchema ? parseRawToolCall<T>(
     {
       toolName,
       toolCallId: generateCompactId(),
       input,
     },
     autoInsertEndStepParam,
-  )
+  ) : {toolName, toolCallId: generateCompactId(), input} as CodebuffToolCall<T>
   if ('error' in toolCall) {
     toolResults.push({
       type: 'tool-result',
diff --git a/common/src/tools/params/tool/set-messages.ts b/common/src/tools/params/tool/set-messages.ts
index c7f461b08..becbf1335 100644
--- a/common/src/tools/params/tool/set-messages.ts
+++ b/common/src/tools/params/tool/set-messages.ts
@@ -14,14 +14,7 @@ export const setMessagesParams = {
           // Make sure to pass through any additional properties!
           .looseObject({
             role: z.enum(['user', 'assistant']),
-            content: z.string().or(
-              z.array(
-                z.object({
-                  type: z.enum(['text']),
-                  text: z.string(),
-                }),
-              ),
-            ),
+            content: z.string()
           }),
       ),
     })
