diff --git a/common/package.json b/common/package.json
index 0700c35a9..a2092f685 100644
--- a/common/package.json
+++ b/common/package.json
@@ -11,18 +11,19 @@
   "dependencies": {
     "ignore": "5.3.2",
     "lodash": "*",
-    "zod": "3.23.8",
     "tree-sitter": "0.21.1",
-    "tree-sitter-typescript": "0.23.0",
-    "tree-sitter-javascript": "0.23.0",
-    "tree-sitter-python": "0.23.2",
-    "tree-sitter-ruby": "0.23.0",
-    "tree-sitter-go": "0.23.1",
-    "tree-sitter-java": "0.23.2",
+    "tree-sitter-c": "0.23.0",
     "tree-sitter-c-sharp": "0.23.0",
     "tree-sitter-cpp": "0.23.0",
+    "tree-sitter-go": "0.23.1",
+    "tree-sitter-java": "0.23.2",
+    "tree-sitter-javascript": "0.23.0",
     "tree-sitter-php": "0.23.2",
-    "tree-sitter-rust": "0.23.0"
+    "tree-sitter-python": "0.23.2",
+    "tree-sitter-ruby": "0.23.0",
+    "tree-sitter-rust": "0.23.0",
+    "tree-sitter-typescript": "0.23.0",
+    "zod": "3.23.8"
   },
   "peerDependencies": {
     "@types/node": "*",
diff --git a/common/src/codemap/languages.ts b/common/src/codemap/languages.ts
index 98d7540a8..70c3db8e2 100644
--- a/common/src/codemap/languages.ts
+++ b/common/src/codemap/languages.ts
@@ -3,13 +3,14 @@ import * as path from 'path'
 import TypeScriptLanguage from 'tree-sitter-typescript'
 import JavaScriptLanguage from 'tree-sitter-javascript'
 import PythonLanguage from 'tree-sitter-python'
+import JavaLanguage from 'tree-sitter-java'
+import CSharpLanguage from 'tree-sitter-c-sharp'
+import CPPLanguage from 'tree-sitter-cpp'
+import CLanguage from 'tree-sitter-c'
+import RustLanguage from 'tree-sitter-rust'
 // import RubyLanguage from 'tree-sitter-ruby'
 // import GoLanguage from 'tree-sitter-go'
-import JavaLanguage from 'tree-sitter-java'
-// import CSharpLanguage from 'tree-sitter-c-sharp'
-// import CPPLanguage from 'tree-sitter-cpp'
 // import PHPLanguage from 'tree-sitter-php'
-import RustLanguage from 'tree-sitter-rust'
 import Parser from 'tree-sitter'
 import { Query } from 'tree-sitter'
 import { DEBUG_PARSING } from './parse'
@@ -43,6 +44,31 @@ const languageConfigs: Omit<LanguageConfig, 'parser' | 'query'>[] = [
     extensions: ['.py'],
     queryFile: 'tree-sitter-python-tags.scm',
   },
+  {
+    language: JavaLanguage,
+    extensions: ['.java'],
+    queryFile: 'tree-sitter-java-tags.scm',
+  },
+  {
+    language: CSharpLanguage,
+    extensions: ['.cs'],
+    queryFile: 'tree-sitter-c_sharp-tags.scm',
+  },
+  {
+    language: CLanguage,
+    extensions: ['.c', '.h'],
+    queryFile: 'tree-sitter-cpp-tags.scm',
+  },
+  {
+    language: CPPLanguage,
+    extensions: ['.cpp', '.hpp'],
+    queryFile: 'tree-sitter-cpp-tags.scm',
+  },
+  {
+    language: RustLanguage,
+    extensions: ['.rs'],
+    queryFile: 'tree-sitter-rust-tags.scm',
+  },
   // {
   //   language: RubyLanguage,
   //   extensions: ['.rb'],
@@ -53,31 +79,11 @@ const languageConfigs: Omit<LanguageConfig, 'parser' | 'query'>[] = [
   //   extensions: ['.go'],
   //   queryFile: 'tree-sitter-go-tags.scm',
   // },
-  {
-    language: JavaLanguage,
-    extensions: ['.java'],
-    queryFile: 'tree-sitter-java-tags.scm',
-  },
-  // {
-  //   language: CSharpLanguage,
-  //   extensions: ['.cs'],
-  //   queryFile: 'tree-sitter-c_sharp-tags.scm',
-  // },
-  // {
-  //   language: CPPLanguage,
-  //   extensions: ['.cpp', '.hpp', '.c', '.h'],
-  //   queryFile: 'tree-sitter-cpp-tags.scm',
-  // },
   // {
   //   language: PHPLanguage,
   //   extensions: ['.php'],
   //   queryFile: 'tree-sitter-php-tags.scm',
   // },
-  {
-    language: RustLanguage,
-    extensions: ['.rs'],
-    queryFile: 'tree-sitter-rust-tags.scm',
-  },
 ]
 
 export function getLanguageConfig(
diff --git a/common/src/codemap/tree-sitter-queries/tree-sitter-c-tags.scm b/common/src/codemap/tree-sitter-queries/tree-sitter-c-tags.scm
index 1035aa224..323eb1565 100644
--- a/common/src/codemap/tree-sitter-queries/tree-sitter-c-tags.scm
+++ b/common/src/codemap/tree-sitter-queries/tree-sitter-c-tags.scm
@@ -1,9 +1,11 @@
-(struct_specifier name: (type_identifier) @name.definition.class body:(_)) @definition.class
+(struct_specifier name: (type_identifier) @identifier)
 
-(declaration type: (union_specifier name: (type_identifier) @name.definition.class)) @definition.class
+(declaration type: (union_specifier name: (type_identifier) @identifier))
 
-(function_declarator declarator: (identifier) @name.definition.function) @definition.function
+(function_declarator declarator: (identifier) @identifier)
 
-(type_definition declarator: (type_identifier) @name.definition.type) @definition.type
+(type_definition declarator: (type_identifier) @identifier)
 
-(enum_specifier name: (type_identifier) @name.definition.type) @definition.type
+(enum_specifier name: (type_identifier) @identifier)
+
+; TODO add calls
diff --git a/common/src/codemap/tree-sitter-queries/tree-sitter-c_sharp-tags.scm b/common/src/codemap/tree-sitter-queries/tree-sitter-c_sharp-tags.scm
index 7ee4b8a43..de6286d2e 100644
--- a/common/src/codemap/tree-sitter-queries/tree-sitter-c_sharp-tags.scm
+++ b/common/src/codemap/tree-sitter-queries/tree-sitter-c_sharp-tags.scm
@@ -1,11 +1,13 @@
 (class_declaration
- name: (identifier) @className
+ name: (identifier) @identifier
  )
 
 (interface_declaration
- name: (identifier) @interfaceName
+ name: (identifier) @identifier
 )
 
 (method_declaration
- name: (identifier) @methodName
+ name: (identifier) @identifier
  )
+
+; TODO add calls
\ No newline at end of file
diff --git a/common/src/codemap/tree-sitter-queries/tree-sitter-cpp-tags.scm b/common/src/codemap/tree-sitter-queries/tree-sitter-cpp-tags.scm
index 7a7ad0b99..5d1eea1f3 100644
--- a/common/src/codemap/tree-sitter-queries/tree-sitter-cpp-tags.scm
+++ b/common/src/codemap/tree-sitter-queries/tree-sitter-cpp-tags.scm
@@ -1,15 +1,17 @@
-(struct_specifier name: (type_identifier) @name.definition.class body:(_)) @definition.class
+(struct_specifier name: (type_identifier) @identifier)
 
-(declaration type: (union_specifier name: (type_identifier) @name.definition.class)) @definition.class
+(declaration type: (union_specifier name: (type_identifier) @identifier))
 
-(function_declarator declarator: (identifier) @name.definition.function) @definition.function
+(function_declarator declarator: (identifier) @identifier)
 
-(function_declarator declarator: (field_identifier) @name.definition.function) @definition.function
+(function_declarator declarator: (field_identifier) @identifier)
 
-(function_declarator declarator: (qualified_identifier scope: (namespace_identifier) @scope name: (identifier) @name.definition.method)) @definition.method
+(function_declarator declarator: (qualified_identifier scope: (namespace_identifier) name: (identifier) @identifier))
 
-(type_definition declarator: (type_identifier) @name.definition.type) @definition.type
+(type_definition declarator: (type_identifier) @identifier)
 
-(enum_specifier name: (type_identifier) @name.definition.type) @definition.type
+(enum_specifier name: (type_identifier) @identifier)
 
-(class_specifier name: (type_identifier) @name.definition.class) @definition.class
+(class_specifier name: (type_identifier) @identifier)
+
+; TODO add calls
\ No newline at end of file
diff --git a/common/src/codemap/tree-sitter-queries/tree-sitter-elisp-tags.scm b/common/src/codemap/tree-sitter-queries/tree-sitter-elisp-tags.scm
index 743c8d8a8..743eff0c0 100644
--- a/common/src/codemap/tree-sitter-queries/tree-sitter-elisp-tags.scm
+++ b/common/src/codemap/tree-sitter-queries/tree-sitter-elisp-tags.scm
@@ -1,8 +1,5 @@
-;; defun/defsubst
-(function_definition name: (symbol) @name.definition.function) @definition.function
+(function_definition name: (symbol) @identifier)
+(macro_definition name: (symbol) @identifier)
+(list (symbol) @identifier)
 
-;; Treat macros as function definitions for the sake of TAGS.
-(macro_definition name: (symbol) @name.definition.function) @definition.function
-
-;; Match function calls
-(list (symbol) @name.reference.function) @reference.function
+; TODO add calls
\ No newline at end of file
diff --git a/npm-app/package.json b/npm-app/package.json
index e4d7ebf9b..f72c635bd 100644
--- a/npm-app/package.json
+++ b/npm-app/package.json
@@ -40,19 +40,20 @@
     "lodash": "*",
     "nanoid": "5.0.7",
     "systeminformation": "5.23.4",
-    "ws": "8.18.0",
-    "zod": "3.23.8",
     "tree-sitter": "0.21.1",
-    "tree-sitter-typescript": "0.23.0",
-    "tree-sitter-javascript": "0.23.0",
-    "tree-sitter-python": "0.23.2",
-    "tree-sitter-ruby": "0.23.0",
-    "tree-sitter-go": "0.23.1",
-    "tree-sitter-java": "0.23.2",
+    "tree-sitter-c": "0.23.0",
     "tree-sitter-c-sharp": "0.23.0",
     "tree-sitter-cpp": "0.23.0",
+    "tree-sitter-go": "0.23.1",
+    "tree-sitter-java": "0.23.2",
+    "tree-sitter-javascript": "0.23.0",
     "tree-sitter-php": "0.23.2",
-    "tree-sitter-rust": "0.23.0"
+    "tree-sitter-python": "0.23.2",
+    "tree-sitter-ruby": "0.23.0",
+    "tree-sitter-rust": "0.23.0",
+    "tree-sitter-typescript": "0.23.0",
+    "ws": "8.18.0",
+    "zod": "3.23.8"
   },
   "devDependencies": {
     "@types/lodash": "^4.17.7",
