diff --git a/npm-app/src/tool-handlers.ts b/npm-app/src/tool-handlers.ts
index 9445d4b8b..02e80915d 100644
--- a/npm-app/src/tool-handlers.ts
+++ b/npm-app/src/tool-handlers.ts
@@ -131,6 +131,9 @@ export const handleRunTerminalCommand = async (
           data.startsWith('Der Befehl') ||
           data.includes(
             'ist entweder falsch geschrieben oder konnte nicht gefunden werden'
+          ) ||
+          data.includes(
+            'wurde nicht als Name eines Cmdlet, einer Funktion, einer Skriptdatei oder eines ausführbaren'
           ))
       ) {
         clearTimeout(timer)
