diff --git a/common/src/constants/analytics-events.ts b/common/src/constants/analytics-events.ts
index 10f83f94c..e0dc3c58a 100644
--- a/common/src/constants/analytics-events.ts
+++ b/common/src/constants/analytics-events.ts
@@ -11,16 +11,17 @@ export enum AnalyticsEvent {
   BACKGROUND_PROCESS_START = 'cli.background_process_start',
   CHANGE_DIRECTORY = 'cli.change_directory',
   CHECKPOINT_COMMAND_USED = 'cli.checkpoint_command_used',
+  INVALID_COMMAND = 'cli.invalid_command',
   KNOWLEDGE_FILE_UPDATED = 'cli.knowledge_file_updated',
   LOGIN = 'cli.login',
-  TERMINAL_COMMAND_COMPLETED = 'cli.terminal_command_completed',
   MALFORMED_PROMPT_RESPONSE = 'cli.malformed_prompt_response',
-  USER_INPUT_COMPLETE = 'cli.user_input_complete',
-  UPDATE_CODEBUFF_FAILED = 'cli.update_codebuff_failed',
+  RAGE = 'cli.rage',
+  SHELL_RECREATED = 'cli.shell_recreated',
   SLASH_MENU_ACTIVATED = 'cli.slash_menu_activated',
   SLASH_COMMAND_USED = 'cli.slash_command_used',
-  INVALID_COMMAND = 'cli.invalid_command',
-  RAGE = 'cli.rage',
+  TERMINAL_COMMAND_COMPLETED = 'cli.terminal_command_completed',
+  USER_INPUT_COMPLETE = 'cli.user_input_complete',
+  UPDATE_CODEBUFF_FAILED = 'cli.update_codebuff_failed',
 
   // Backend
   USER_INPUT = 'backend.user_input',
diff --git a/npm-app/src/terminal/base.ts b/npm-app/src/terminal/base.ts
index 295950179..384bb2dca 100644
--- a/npm-app/src/terminal/base.ts
+++ b/npm-app/src/terminal/base.ts
@@ -207,6 +207,7 @@ export const isCommandRunning = () => {
 
 export const recreateShell = async (cwd: string, forceChildProcess = false) => {
   persistentProcess = await createPersistantProcess(cwd, forceChildProcess)
+  trackEvent(AnalyticsEvent.SHELL_RECREATED, { persistentProcess })
 }
 
 export const resetShell = async (cwd: string) => {
