diff --git a/backend/src/websockets/middleware.ts b/backend/src/websockets/middleware.ts
index 3ce2f99cc..06003b398 100644
--- a/backend/src/websockets/middleware.ts
+++ b/backend/src/websockets/middleware.ts
@@ -82,6 +82,9 @@ protec.use(async (action, _clientSessionId, _) => {
   logger.debug(`Protecting action of type: '${action.type}'`)
 })
 protec.use(async (action, _clientSessionId, ws) => {
+  if (env.ENVIRONMENT === 'local') {
+    return
+  }
   return match(action)
     .with(
       {
