diff --git a/backend/src/claude.ts b/backend/src/claude.ts
index 4b1ead471..c89da7103 100644
--- a/backend/src/claude.ts
+++ b/backend/src/claude.ts
@@ -4,6 +4,7 @@ import { removeUndefinedProps } from 'common/util/object'
 import { Message, ToolCall } from 'common/actions'
 import { STOP_MARKER } from 'common/constants'
 import { debugLog } from './util/debug'
+import { RATE_LIMIT_POLICY } from './constants'
 
 export const models = {
   sonnet: 'claude-3-5-sonnet-20240620' as const,
@@ -14,7 +15,12 @@ export type model_types = (typeof models)[keyof typeof models]
 
 export const promptClaudeStream = async function* (
   messages: Message[],
-  options: { system?: string; tools?: Tool[]; model?: model_types; userId: string }
+  options: {
+    system?: string
+    tools?: Tool[]
+    model?: model_types
+    userId: string
+  }
 ): AsyncGenerator<string | ToolCall, void, unknown> {
   const { model = models.sonnet, system, tools, userId } = options
 
@@ -30,6 +36,7 @@ export const promptClaudeStream = async function* (
     defaultHeaders: {
       'Helicone-Auth': `Bearer ${process.env.HELICONE_API_KEY}`,
       'Helicone-User-Id': userId,
+      'Helicone-RateLimit-Policy': RATE_LIMIT_POLICY,
     },
   })
 
@@ -84,7 +91,12 @@ export const promptClaudeStream = async function* (
 
 export const promptClaude = async (
   prompt: string,
-  options: { system?: string; tools?: Tool[]; model?: model_types; userId: string }
+  options: {
+    system?: string
+    tools?: Tool[]
+    model?: model_types
+    userId: string
+  }
 ) => {
   let fullResponse = ''
   for await (const chunk of promptClaudeStream(
diff --git a/backend/src/constants.ts b/backend/src/constants.ts
new file mode 100644
index 000000000..b936aa44c
--- /dev/null
+++ b/backend/src/constants.ts
@@ -0,0 +1 @@
+export const RATE_LIMIT_POLICY = '2000;w=86400;u=cents;s=user'
diff --git a/backend/src/openai-api.ts b/backend/src/openai-api.ts
index 092e12069..9aac5bea1 100644
--- a/backend/src/openai-api.ts
+++ b/backend/src/openai-api.ts
@@ -1,4 +1,5 @@
 import OpenAI from 'openai'
+import { RATE_LIMIT_POLICY } from './constants'
 
 let openai: OpenAI | null = null
 
@@ -10,6 +11,7 @@ const getOpenAI = (userId: string) => {
       defaultHeaders: {
         'Helicone-Auth': `Bearer ${process.env.HELICONE_API_KEY}`,
         'Helicone-User-Id': userId,
+        'Helicone-RateLimit-Policy': RATE_LIMIT_POLICY,
       },
     })
   }
@@ -20,7 +22,7 @@ const getOpenAI = (userId: string) => {
 export async function promptOpenAI(
   userId: string,
   messages: OpenAI.Chat.ChatCompletionMessageParam[],
-  model: string,
+  model: string
 ) {
   const openai = getOpenAI(userId)
   try {
