diff --git a/web/public/testimonials/proof/shardool-patel.png b/web/public/testimonials/proof/shardool-patel.png
new file mode 100644
index 000000000..284fc43a3
Binary files /dev/null and b/web/public/testimonials/proof/shardool-patel.png differ
diff --git a/web/public/testimonials/shardool-patel.jpg b/web/public/testimonials/shardool-patel.jpg
new file mode 100644
index 000000000..c6f488fb1
Binary files /dev/null and b/web/public/testimonials/shardool-patel.jpg differ
diff --git a/web/src/lib/testimonials.ts b/web/src/lib/testimonials.ts
index 790fddb1d..9398d2af0 100644
--- a/web/src/lib/testimonials.ts
+++ b/web/src/lib/testimonials.ts
@@ -68,5 +68,13 @@ export const testimonials: Testimonial[][] = [
       avatar: '/testimonials/janna-lu.jpg',
       link: '/testimonials/proof/janna-lu.png',
     },
+    {
+      quote:
+        "Im basically hiring an engineer for 100/month, that's how I see it",
+      author: 'Shardool Patel',
+      title: 'Founder, CTO',
+      avatar: '/testimonials/shardool-patel.jpg',
+      link: '/testimonials/proof/shardool-patel.png',
+    },
   ],
 ]
