diff --git a/common/src/util/__tests__/string.test.ts b/common/src/util/__tests__/string.test.ts
index 0791de004..8c2d153ad 100644
--- a/common/src/util/__tests__/string.test.ts
+++ b/common/src/util/__tests__/string.test.ts
@@ -1,3 +1,4 @@
+import { EXISTING_CODE_MARKER } from 'src/constants'
 import { replaceNonStandardPlaceholderComments } from '../string'
 
 describe('replaceNonStandardPlaceholderComments', () => {
@@ -11,12 +12,14 @@ function example() {
 `
     const expected = `
 function example() {
-  // ... existing code ...
+  ${EXISTING_CODE_MARKER}
   console.log('Hello');
-  // ... existing code ...
+  ${EXISTING_CODE_MARKER}
 }
 `
-    expect(replaceNonStandardPlaceholderComments(input)).toBe(expected)
+    expect(
+      replaceNonStandardPlaceholderComments(input, EXISTING_CODE_MARKER)
+    ).toBe(expected)
   })
 
   it('should replace multi-line C-style comments', () => {
@@ -29,12 +32,14 @@ function example() {
 `
     const expected = `
 function example() {
-  // ... existing code ...
+  ${EXISTING_CODE_MARKER}
   console.log('Hello');
-  // ... existing code ...
+  ${EXISTING_CODE_MARKER}
 }
 `
-    expect(replaceNonStandardPlaceholderComments(input)).toBe(expected)
+    expect(
+      replaceNonStandardPlaceholderComments(input, EXISTING_CODE_MARKER)
+    ).toBe(expected)
   })
 
   it('should replace Python-style comments', () => {
@@ -46,10 +51,40 @@ def example():
 `
     const expected = `
 def example():
-    // ... existing code ...
+    ${EXISTING_CODE_MARKER}
     print('Hello')
-    // ... existing code ...
+    ${EXISTING_CODE_MARKER}
 `
-    expect(replaceNonStandardPlaceholderComments(input)).toBe(expected)
+    expect(
+      replaceNonStandardPlaceholderComments(input, EXISTING_CODE_MARKER)
+    ).toBe(expected)
+  })
+
+  it('should replace JSX comments', () => {
+    const input = `
+function Example() {
+  return (
+    <div>
+      {/* ... existing code ... */}
+      <p>Hello, World!</p>
+      {/* ...rest of component... */}
+    </div>
+  );
+}
+`
+    const expected = `
+function Example() {
+  return (
+    <div>
+      ${EXISTING_CODE_MARKER}
+      <p>Hello, World!</p>
+      ${EXISTING_CODE_MARKER}
+    </div>
+  );
+}
+`
+    expect(
+      replaceNonStandardPlaceholderComments(input, EXISTING_CODE_MARKER)
+    ).toBe(expected)
   })
 })
diff --git a/common/src/util/string.ts b/common/src/util/string.ts
index b4ed7da52..089513f4f 100644
--- a/common/src/util/string.ts
+++ b/common/src/util/string.ts
@@ -12,6 +12,12 @@ export const replaceNonStandardPlaceholderComments = (
   replacement: string
 ): string => {
   const commentPatterns = [
+    // JSX comments (match this first)
+    {
+      regex:
+        /{\s*\/\*\s*\.{3}.*(?:rest|unchanged|keep|file|existing|some).*(?:\s*\.{3})?\s*\*\/\s*}/gi,
+      placeholder: replacement,
+    },
     // C-style comments (C, C++, Java, JavaScript, TypeScript, etc.)
     {
       regex:
