diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index e9a803b7c..dc8ef6545 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -36,7 +36,6 @@ export async function mainPrompt(
   )
 
   const relevantFiles = await requestRelevantFiles(messages, fileContext)
-  debugLog('Relevant files:', relevantFiles)
 
   // Load relevant files content
   const fileContents = await requestFiles(ws, relevantFiles)
@@ -64,6 +63,8 @@ export async function mainPrompt(
 <relevant_files>
 Here are some relevant files to aid in the user request, ordered by most important first:
 ${fileBlocks}
+
+You can read more files with the read_files tool.
 </relevant_files>
 
 <additional_instruction>
diff --git a/backend/src/request-files-prompt.ts b/backend/src/request-files-prompt.ts
index d5c53bcd4..46c634224 100644
--- a/backend/src/request-files-prompt.ts
+++ b/backend/src/request-files-prompt.ts
@@ -1,3 +1,5 @@
+import { uniq } from 'lodash'
+
 import { Message } from 'common/actions'
 import {
   ProjectFileContext,
@@ -24,18 +26,10 @@ export async function requestRelevantFiles(
     ),
   ])
 
+  const dedupedFiles = uniq([...keyResult.files, ...comprehensiveResult.files])
+
   debugLog('Key files:', keyResult.files)
-  debugLog('Key request time:', keyResult.duration.toFixed(0), 'ms')
   debugLog('Comprehensive files:', comprehensiveResult.files)
-  debugLog(
-    'Comprehensive request time:',
-    comprehensiveResult.duration.toFixed(0),
-    'ms'
-  )
-
-  const dedupedFiles = Array.from(
-    new Set([...keyResult.files, ...comprehensiveResult.files])
-  )
   debugLog('Deduped files:', dedupedFiles)
 
   return dedupedFiles
@@ -51,11 +45,17 @@ async function getRelevantFiles(
   const end = performance.now()
   const duration = end - start
 
+  debugLog(`${requestType} response time:`, duration.toFixed(0), 'ms')
+  debugLog(`${requestType} response:`, response)
+
   const fileListMatch = response.match(/<file_list>([\s\S]*?)<\/file_list>/)
   if (!fileListMatch) {
     console.error(
       `Failed to extract file list from Claude response for ${requestType} request`
     )
+    debugLog(
+      `Failed to extract file list from Claude response for ${requestType} request`
+    )
     return { files: [], duration }
   }
 
