diff --git a/backend/src/planning.ts b/backend/src/planning.ts
index d1c1ed6b1..35c5f007b 100644
--- a/backend/src/planning.ts
+++ b/backend/src/planning.ts
@@ -6,7 +6,7 @@ import {
   ProjectFileContext,
 } from 'common/util/file'
 import { promptClaude } from './claude'
-import { OpenAIMessage, promptOpenAIStream } from './open-router'
+import { OpenAIMessage, promptOpenAIStream } from './openai-api'
 import { getSearchSystemPrompt } from './system-prompt'
 
 export async function planComplexChange(
diff --git a/common/src/constants.ts b/common/src/constants.ts
index a6d5f8efb..907227a18 100644
--- a/common/src/constants.ts
+++ b/common/src/constants.ts
@@ -116,7 +116,8 @@ export const claudeModels = {
 export const openaiModels = {
   gpt4o: 'gpt-4o-2024-08-06',
   gpt4omini: 'gpt-4o-mini-2024-07-18',
-  o1: 'o1-2024-12-17',
+  o1: 'o1-preview',
+  // o1: 'o1-2024-12-17',
 } as const
 
 export const geminiModels = {
