diff --git a/backend/src/main-prompt.ts b/backend/src/main-prompt.ts
index 38a8c4346..fd4a694e6 100644
--- a/backend/src/main-prompt.ts
+++ b/backend/src/main-prompt.ts
@@ -190,8 +190,6 @@ export async function mainPrompt(
           userId,
         }
       )
-      onResponseChunk('\n\n')
-      fullResponse += '\n\n'
     } else {
       stream = promptClaudeStream(messagesWithContinuedMessage, {
         system,
