diff --git a/web/public/auth-success.jpg b/web/public/auth-success.jpg
deleted file mode 100644
index 261859dc4..000000000
Binary files a/web/public/auth-success.jpg and /dev/null differ
diff --git a/web/public/auth-success.png b/web/public/auth-success.png
new file mode 100644
index 000000000..589fa7042
Binary files /dev/null and b/web/public/auth-success.png differ
diff --git a/web/public/plan-change.png b/web/public/plan-change.png
new file mode 100644
index 000000000..0374e32b9
Binary files /dev/null and b/web/public/plan-change.png differ
diff --git a/web/src/app/onboard/page.tsx b/web/src/app/onboard/page.tsx
index d99fab5a4..a1a6cf48e 100644
--- a/web/src/app/onboard/page.tsx
+++ b/web/src/app/onboard/page.tsx
@@ -175,7 +175,7 @@ const Onboard = async ({ searchParams }: PageProps) => {
       content: (
         <div className="flex flex-col space-y-2">
           <Image
-            src="/auth-success.jpg"
+            src="/auth-success.png"
             alt="Successful authentication"
             width={600}
             height={600}
diff --git a/web/src/app/payment-change/page.tsx b/web/src/app/payment-change/page.tsx
new file mode 100644
index 000000000..f6c551d94
--- /dev/null
+++ b/web/src/app/payment-change/page.tsx
@@ -0,0 +1,77 @@
+'use client'
+
+import Image from 'next/image'
+import { useEffect } from 'react'
+import posthog from 'posthog-js'
+import { PLAN_CONFIGS } from 'common/constants'
+import { useRouter, useSearchParams, usePathname } from 'next/navigation'
+import { useSession } from 'next-auth/react'
+import CardWithBeams from '@/components/card-with-beams'
+import { trackUpgrade } from '@/lib/trackConversions'
+import { useUserPlan } from '@/hooks/use-user-plan'
+import { capitalize } from 'common/util/string'
+
+const PaymentChangePage = () => {
+  const router = useRouter()
+  const pathname = usePathname()
+  const searchParams = useSearchParams()
+  const { data: session } = useSession()
+  const { data: currentPlan } = useUserPlan(session?.user?.stripe_customer_id)
+  const modification = searchParams.get('modification') || 'change'
+
+  useEffect(() => {
+    const params = trackUpgrade(true)
+    const newParams = new URLSearchParams(searchParams)
+    params.forEach((value, key) => newParams.set(key, value))
+    router.replace(`${pathname}?${newParams}`)
+
+    if (session?.user) {
+      posthog.capture('subscription.plan_changed', {
+        plan: currentPlan,
+        modification,
+      })
+    }
+  }, [session, currentPlan, modification])
+
+  if (!session?.user) {
+    return CardWithBeams({
+      title: "You're not logged in.",
+      description: 'How can you change your plan but not be logged in?!',
+      content: (
+        <p>
+          Err this is awkward... Please reach out to support at{' '}
+          <a href="mailto:support@codebuff.com">support@codebuff.com</a> for
+          help finishing your {modification}.
+        </p>
+      ),
+    })
+  }
+
+  if (!currentPlan) {
+    return CardWithBeams({
+      title: 'Something went wrong',
+      description:
+        'We could not find your plan details. Please contact support for assistance.',
+    })
+  }
+
+  const credits = PLAN_CONFIGS[currentPlan].limit
+  const planDisplayName = PLAN_CONFIGS[currentPlan].displayName
+
+  return CardWithBeams({
+    title: `${capitalize(modification)} successful!`,
+    description: `Your plan has been ${modification}d to ${planDisplayName}. You now have ${credits.toLocaleString()} monthly credits.`,
+    content: (
+      <div className="flex flex-col space-y-2">
+        <Image
+          src="/plan-change.png"
+          alt="Successful plan change"
+          width={600}
+          height={600}
+        />
+      </div>
+    ),
+  })
+}
+
+export default PaymentChangePage
diff --git a/web/src/app/subscription/confirm/page.tsx b/web/src/app/subscription/confirm/page.tsx
index 35a8c9142..57e7cd86c 100644
--- a/web/src/app/subscription/confirm/page.tsx
+++ b/web/src/app/subscription/confirm/page.tsx
@@ -70,7 +70,8 @@ const useUpgradeSubscription = (
         return
       }
 
-      router.push('/payment-success')
+      const modification = changeOrUpgrade(currentPlan, targetPlan)
+      router.push('/payment-change?modification=' + modification)
     },
     onError: async (error: any) => {
       console.error('Error upgrading subscription:', error)
