diff --git a/backend/src/claude.ts b/backend/src/claude.ts
index 3e51895b4..b9bc244a6 100644
--- a/backend/src/claude.ts
+++ b/backend/src/claude.ts
@@ -38,7 +38,8 @@ export const promptClaudeStream = async function* (
     ignoreHelicone = false,
   } = options
 
-  const apiKey = env.ANTHROPIC_API_KEY
+  const apiKey =
+    Math.random() > 0.5 ? env.ANTHROPIC_API_KEY : env.ANTHROPIC_API_KEY2
 
   if (!apiKey) {
     throw new Error('Missing ANTHROPIC_API_KEY')
diff --git a/backend/src/env.mjs b/backend/src/env.mjs
index 30d7f3447..547668581 100644
--- a/backend/src/env.mjs
+++ b/backend/src/env.mjs
@@ -16,6 +16,7 @@ dotenv.config({ path })
 export const env = createEnv({
   server: {
     ANTHROPIC_API_KEY: z.string().min(1).startsWith('sk-ant-'),
+    ANTHROPIC_API_KEY2: z.string().min(1).startsWith('sk-ant-'),
     HELICONE_API_KEY: z.string().min(1).startsWith('pk-helicone-'),
     OPEN_AI_KEY: z.string().min(1).startsWith('sk-proj-'),
     STRIPE_SECRET_KEY: z.string().min(1),
