diff --git a/backend/src/run-agent-step.ts b/backend/src/run-agent-step.ts
index 179a76f8b..4e7e5c105 100644
--- a/backend/src/run-agent-step.ts
+++ b/backend/src/run-agent-step.ts
@@ -48,6 +48,7 @@ import type {
 } from '@codebuff/common/types/session-state'
 import type { ProjectFileContext } from '@codebuff/common/util/file'
 import type { WebSocket } from 'ws'
+import { TOOLS_WHICH_WONT_FORCE_NEXT_STEP } from '@codebuff/common/tools/constants'
 
 export interface AgentOptions {
   userId: string | undefined
@@ -437,6 +438,16 @@ export const runAgentStep = async (
     logger.debug({ summary: fullResponse }, 'Compacted messages')
   }
 
+  const hasNoToolResults =
+    toolCalls.filter(
+      (call) => !TOOLS_WHICH_WONT_FORCE_NEXT_STEP.includes(call.toolName),
+    ).length === 0 &&
+    toolResults.filter(
+      (result) => !TOOLS_WHICH_WONT_FORCE_NEXT_STEP.includes(result.toolName),
+    ).length === 0
+  const shouldEndTurn =
+    toolCalls.some((call) => call.toolName === 'end_turn') || hasNoToolResults
+
   logger.debug(
     {
       iteration: iterationNum,
@@ -446,6 +457,7 @@ export const runAgentStep = async (
       fullResponseChunks,
       toolCalls,
       toolResults,
+      shouldEndTurn,
       agentContext: newAgentContext,
       finalMessageHistoryWithToolResults,
       model,
@@ -454,10 +466,6 @@ export const runAgentStep = async (
     },
     `End agent ${agentType} step ${iterationNum} (${userInputId}${prompt ? ` - Prompt: ${prompt.slice(0, 20)}` : ''})`,
   )
-  const shouldEndTurn =
-    toolCalls.some((call) => call.toolName === 'end_turn') ||
-    (toolCalls.length === 0 && toolResults.length === 0)
-
   const newAgentState = {
     ...agentState,
     messageHistory: finalMessageHistoryWithToolResults,
diff --git a/common/src/tools/constants.ts b/common/src/tools/constants.ts
index 738d9171b..28af7dd4e 100644
--- a/common/src/tools/constants.ts
+++ b/common/src/tools/constants.ts
@@ -7,6 +7,8 @@ export const toolXmlName = 'codebuff_tool_call'
 export const startToolTag = `<${toolXmlName}>\n`
 export const endToolTag = `\n</${toolXmlName}>`
 
+export const TOOLS_WHICH_WONT_FORCE_NEXT_STEP = ['think_deeply']
+
 // List of all available tools
 export const toolNames = [
   'add_subgoal',
