diff --git a/backend/src/websockets/websocket-action.ts b/backend/src/websockets/websocket-action.ts
index 528b1f092..eb43e42ae 100644
--- a/backend/src/websockets/websocket-action.ts
+++ b/backend/src/websockets/websocket-action.ts
@@ -23,34 +23,53 @@ const onUserInput = async (
   const lastMessage = messages[messages.length - 1]
   if (typeof lastMessage.content === 'string')
     console.log('Input:', lastMessage)
-  const { toolCall, response, changes } = await mainPrompt(
-    ws,
-    messages,
-    fileContext,
-    (chunk) =>
-      sendAction(ws, {
-        type: 'response-chunk',
-        chunk,
-      })
-  )
 
-  const allChanges = [...previousChanges, ...changes]
+  try {
+    const { toolCall, response, changes } = await mainPrompt(
+      ws,
+      messages,
+      fileContext,
+      (chunk) =>
+        sendAction(ws, {
+          type: 'response-chunk',
+          chunk,
+        })
+    )
+    const allChanges = [...previousChanges, ...changes]
 
-  if (toolCall) {
-    console.log('toolCall', toolCall.name, toolCall.input)
-    sendAction(ws, {
-      type: 'tool-call',
-      response,
-      data: toolCall,
-      changes: allChanges,
-    })
-  } else {
-    console.log('response-complete')
+    if (toolCall) {
+      console.log('toolCall', toolCall.name, toolCall.input)
+      sendAction(ws, {
+        type: 'tool-call',
+        response,
+        data: toolCall,
+        changes: allChanges,
+      })
+    } else {
+      console.log('response-complete')
+      sendAction(ws, {
+        type: 'response-complete',
+        response,
+        changes: allChanges,
+      })
+    }
+  } catch (e) {
+    console.error('Error in mainPrompt', e)
+    const response =
+      e && typeof e === 'object' && 'message' in e
+        ? `\n\nError: ${e.message}`
+        : ''
     sendAction(ws, {
-      type: 'response-complete',
-      response,
-      changes: allChanges,
+      type: 'response-chunk',
+      chunk: response,
     })
+    setTimeout(() => {
+      sendAction(ws, {
+        type: 'response-complete',
+        response,
+        changes: [],
+      })
+    }, 100)
   }
 }
 
diff --git a/manicode/src/cli.ts b/manicode/src/cli.ts
index 3a297b97a..bf78d7d3d 100644
--- a/manicode/src/cli.ts
+++ b/manicode/src/cli.ts
@@ -220,7 +220,9 @@ export class CLI {
         console.log('-', old ? 'Updated' : 'Created', filePath)
       }
     }
-    console.log('Complete!\n')
+    if (changesSuceeded.length > 0) {
+      console.log('Complete!\n')
+    }
 
     this.isReceivingResponse = false
 
